{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "105b52c5-133d-4c01-90e0-49667ed8cc77",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Matplotlib is building the font cache; this may take a moment.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "import protfasta\n",
    "import requests\n",
    "import re\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "import glob\n",
    "\n",
    "\n",
    "from Bio import AlignIO\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "\n",
    "import rich\n",
    "from rich_msa import RichAlignment\n",
    "\n",
    "import logomaker\n",
    "\n",
    "#Import modules\n",
    "import gzip, logomaker, matplotlib.pyplot as plt, numpy as np, os, pandas as pd, protfasta\n",
    "\n",
    "\n",
    "import seaborn as sns\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2a4cf3d5-e75c-4e57-ad28-1dc472e6b519",
   "metadata": {},
   "source": [
    "# 1. Downloading codon alignments for all TFs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "868cdd50-c104-40c7-9008-ea60299579a6",
   "metadata": {},
   "source": [
    "First, using uniprot to get all possible gene names."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "8dd65a30-7178-4d09-b7f2-a478727f1f2c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>ProteinSeq</th>\n",
       "      <th>DBD_coords_merged</th>\n",
       "      <th>uniprotID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sp|A0A087WUV0|ZN892_HUMAN Zinc finger protein ...</td>\n",
       "      <td>MEPEGRGSLFEDSDLLHAGNPKENDVTAVLLTPGSQELMIRDMAEA...</td>\n",
       "      <td>[[221, 243], [249, 271], [277, 299], [305, 327...</td>\n",
       "      <td>A0A087WUV0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sp|A0AVK6|E2F8_HUMAN Transcription factor E2F8...</td>\n",
       "      <td>MENEKENLFCEPHKRGLMKTPLKESTTANIVLAEIQPDFGPLTTPT...</td>\n",
       "      <td>[[114, 182], [262, 347]]</td>\n",
       "      <td>A0AVK6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>sp|A0PJY2|FEZF1_HUMAN Fez family zinc finger p...</td>\n",
       "      <td>MDSSCHNATTKMLATAPARGNMMSTSKPLAFSIERIMARTPEPKAL...</td>\n",
       "      <td>[[260, 282], [288, 310], [316, 338], [344, 366...</td>\n",
       "      <td>A0PJY2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>sp|A1A519|F170A_HUMAN Protein FAM170A OS=Homo ...</td>\n",
       "      <td>MKRRQKRKHLENEESQETAEKGGGMSKSQEDALQPGSTRVAKGWSQ...</td>\n",
       "      <td>[[1, 330]]</td>\n",
       "      <td>A1A519</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>sp|A1YPR0|ZBT7C_HUMAN Zinc finger and BTB doma...</td>\n",
       "      <td>MANDIDELIGIPFPNHSSEVLCSLNEQRHDGLLCDVLLVVQEQEYR...</td>\n",
       "      <td>[[364, 386], [392, 414], [420, 442], [448, 469]]</td>\n",
       "      <td>A1YPR0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1608</th>\n",
       "      <td>sp|Q9Y6Q9|NCOA3_HUMAN Nuclear receptor coactiv...</td>\n",
       "      <td>MSGLGENLDPLASDSRKRKLPCDTPGQGLTCSGEKRRREQESKYIE...</td>\n",
       "      <td>[[31, 83]]</td>\n",
       "      <td>Q9Y6Q9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1609</th>\n",
       "      <td>sp|Q9Y6R6|Z780B_HUMAN Zinc finger protein 780B...</td>\n",
       "      <td>MVHGSVTFRDVAIDFSQEEWECLQPDQRTLYRDVMLENYSHLISLG...</td>\n",
       "      <td>[[165, 187], [193, 215], [221, 243], [249, 271...</td>\n",
       "      <td>Q9Y6R6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1610</th>\n",
       "      <td>sp|Q9Y6X0|SETBP_HUMAN SET-binding protein OS=H...</td>\n",
       "      <td>MESRETLSSSRQRGGESDFLPVSSAKPPAAPGCAGEPLLSTPGPGK...</td>\n",
       "      <td>[[583, 596], [1015, 1027], [1450, 1462]]</td>\n",
       "      <td>Q9Y6X0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1611</th>\n",
       "      <td>sp|Q9Y6X8|ZHX2_HUMAN Zinc fingers and homeobox...</td>\n",
       "      <td>MASKRKSTTPCMVRTSQVVEQDVPEEVDRAKEKGIGTPQPDVAKDS...</td>\n",
       "      <td>[[78, 101], [110, 133], [271, 317], [442, 496]...</td>\n",
       "      <td>Q9Y6X8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1612</th>\n",
       "      <td>sp|Q9Y6Y1|CMTA1_HUMAN Calmodulin-binding trans...</td>\n",
       "      <td>MWRAEGKWLPKTSRKSVSQSVFCGTSTYCVLNTVPPIEDDHGNSNS...</td>\n",
       "      <td>[[69, 182]]</td>\n",
       "      <td>Q9Y6Y1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1613 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     id  \\\n",
       "0     sp|A0A087WUV0|ZN892_HUMAN Zinc finger protein ...   \n",
       "1     sp|A0AVK6|E2F8_HUMAN Transcription factor E2F8...   \n",
       "2     sp|A0PJY2|FEZF1_HUMAN Fez family zinc finger p...   \n",
       "3     sp|A1A519|F170A_HUMAN Protein FAM170A OS=Homo ...   \n",
       "4     sp|A1YPR0|ZBT7C_HUMAN Zinc finger and BTB doma...   \n",
       "...                                                 ...   \n",
       "1608  sp|Q9Y6Q9|NCOA3_HUMAN Nuclear receptor coactiv...   \n",
       "1609  sp|Q9Y6R6|Z780B_HUMAN Zinc finger protein 780B...   \n",
       "1610  sp|Q9Y6X0|SETBP_HUMAN SET-binding protein OS=H...   \n",
       "1611  sp|Q9Y6X8|ZHX2_HUMAN Zinc fingers and homeobox...   \n",
       "1612  sp|Q9Y6Y1|CMTA1_HUMAN Calmodulin-binding trans...   \n",
       "\n",
       "                                             ProteinSeq  \\\n",
       "0     MEPEGRGSLFEDSDLLHAGNPKENDVTAVLLTPGSQELMIRDMAEA...   \n",
       "1     MENEKENLFCEPHKRGLMKTPLKESTTANIVLAEIQPDFGPLTTPT...   \n",
       "2     MDSSCHNATTKMLATAPARGNMMSTSKPLAFSIERIMARTPEPKAL...   \n",
       "3     MKRRQKRKHLENEESQETAEKGGGMSKSQEDALQPGSTRVAKGWSQ...   \n",
       "4     MANDIDELIGIPFPNHSSEVLCSLNEQRHDGLLCDVLLVVQEQEYR...   \n",
       "...                                                 ...   \n",
       "1608  MSGLGENLDPLASDSRKRKLPCDTPGQGLTCSGEKRRREQESKYIE...   \n",
       "1609  MVHGSVTFRDVAIDFSQEEWECLQPDQRTLYRDVMLENYSHLISLG...   \n",
       "1610  MESRETLSSSRQRGGESDFLPVSSAKPPAAPGCAGEPLLSTPGPGK...   \n",
       "1611  MASKRKSTTPCMVRTSQVVEQDVPEEVDRAKEKGIGTPQPDVAKDS...   \n",
       "1612  MWRAEGKWLPKTSRKSVSQSVFCGTSTYCVLNTVPPIEDDHGNSNS...   \n",
       "\n",
       "                                      DBD_coords_merged   uniprotID  \n",
       "0     [[221, 243], [249, 271], [277, 299], [305, 327...  A0A087WUV0  \n",
       "1                              [[114, 182], [262, 347]]      A0AVK6  \n",
       "2     [[260, 282], [288, 310], [316, 338], [344, 366...      A0PJY2  \n",
       "3                                            [[1, 330]]      A1A519  \n",
       "4      [[364, 386], [392, 414], [420, 442], [448, 469]]      A1YPR0  \n",
       "...                                                 ...         ...  \n",
       "1608                                         [[31, 83]]      Q9Y6Q9  \n",
       "1609  [[165, 187], [193, 215], [221, 243], [249, 271...      Q9Y6R6  \n",
       "1610           [[583, 596], [1015, 1027], [1450, 1462]]      Q9Y6X0  \n",
       "1611  [[78, 101], [110, 133], [271, 317], [442, 496]...      Q9Y6X8  \n",
       "1612                                        [[69, 182]]      Q9Y6Y1  \n",
       "\n",
       "[1613 rows x 4 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lambert_TFs = pd.read_csv(\"../output/lambert_TFs_10-21-24_with_DBD_coords.csv\", index_col = 0)\n",
    "lambert_TFs[\"uniprotID\"] = lambert_TFs[\"id\"].str.split(\"|\").str[1]\n",
    "lambert_TFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "8d3496dc-2441-44f1-8e08-1de247d744de",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>From</th>\n",
       "      <th>Entry</th>\n",
       "      <th>Gene Names</th>\n",
       "      <th>Ensembl</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A0A087WUV0</td>\n",
       "      <td>A0A087WUV0</td>\n",
       "      <td>ZNF892</td>\n",
       "      <td>ENST00000425953.6;</td>\n",
       "      <td>ZNF892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A0AVK6</td>\n",
       "      <td>A0AVK6</td>\n",
       "      <td>E2F8</td>\n",
       "      <td>ENST00000250024.9;ENST00000527884.5;ENST000006...</td>\n",
       "      <td>E2F8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A0PJY2-1</td>\n",
       "      <td>A0PJY2</td>\n",
       "      <td>FEZF1 FEZ ZNF312B</td>\n",
       "      <td>ENST00000427185.2 [A0PJY2-2];ENST00000442488.7...</td>\n",
       "      <td>FEZF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A1A519-1</td>\n",
       "      <td>A1A519</td>\n",
       "      <td>FAM170A ZNFD</td>\n",
       "      <td>ENST00000335286.10 [A1A519-2];ENST00000379555....</td>\n",
       "      <td>FAM170A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A1YPR0</td>\n",
       "      <td>A1YPR0</td>\n",
       "      <td>ZBTB7C APM1 ZBTB36 ZNF857C</td>\n",
       "      <td>ENST00000535628.6;ENST00000586438.5;ENST000005...</td>\n",
       "      <td>ZBTB7C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1608</th>\n",
       "      <td>Q9Y6Q9-1</td>\n",
       "      <td>Q9Y6Q9</td>\n",
       "      <td>NCOA3 AIB1 BHLHE42 RAC3 TRAM1</td>\n",
       "      <td>ENST00000371997.3 [Q9Y6Q9-3];ENST00000371998.8...</td>\n",
       "      <td>NCOA3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1609</th>\n",
       "      <td>Q9Y6R6</td>\n",
       "      <td>Q9Y6R6</td>\n",
       "      <td>ZNF780B ZNF779</td>\n",
       "      <td>ENST00000434248.6;ENST00000625684.3;ENST000006...</td>\n",
       "      <td>ZNF780B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1610</th>\n",
       "      <td>Q9Y6X0-1</td>\n",
       "      <td>Q9Y6X0</td>\n",
       "      <td>SETBP1 KIAA0437</td>\n",
       "      <td>ENST00000426838.8 [Q9Y6X0-2];ENST00000649279.2...</td>\n",
       "      <td>SETBP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1611</th>\n",
       "      <td>Q9Y6X8</td>\n",
       "      <td>Q9Y6X8</td>\n",
       "      <td>ZHX2 AFR1 KIAA0854 RAF</td>\n",
       "      <td>ENST00000314393.6;</td>\n",
       "      <td>ZHX2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1612</th>\n",
       "      <td>Q9Y6Y1-1</td>\n",
       "      <td>Q9Y6Y1</td>\n",
       "      <td>CAMTA1 KIAA0833 MSTP023</td>\n",
       "      <td>ENST00000303635.12 [Q9Y6Y1-1];ENST00000473578....</td>\n",
       "      <td>CAMTA1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1613 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            From       Entry                     Gene Names  \\\n",
       "0     A0A087WUV0  A0A087WUV0                         ZNF892   \n",
       "1         A0AVK6      A0AVK6                           E2F8   \n",
       "2       A0PJY2-1      A0PJY2              FEZF1 FEZ ZNF312B   \n",
       "3       A1A519-1      A1A519                   FAM170A ZNFD   \n",
       "4         A1YPR0      A1YPR0     ZBTB7C APM1 ZBTB36 ZNF857C   \n",
       "...          ...         ...                            ...   \n",
       "1608    Q9Y6Q9-1      Q9Y6Q9  NCOA3 AIB1 BHLHE42 RAC3 TRAM1   \n",
       "1609      Q9Y6R6      Q9Y6R6                 ZNF780B ZNF779   \n",
       "1610    Q9Y6X0-1      Q9Y6X0                SETBP1 KIAA0437   \n",
       "1611      Q9Y6X8      Q9Y6X8         ZHX2 AFR1 KIAA0854 RAF   \n",
       "1612    Q9Y6Y1-1      Q9Y6Y1        CAMTA1 KIAA0833 MSTP023   \n",
       "\n",
       "                                                Ensembl     gene  \n",
       "0                                    ENST00000425953.6;   ZNF892  \n",
       "1     ENST00000250024.9;ENST00000527884.5;ENST000006...     E2F8  \n",
       "2     ENST00000427185.2 [A0PJY2-2];ENST00000442488.7...    FEZF1  \n",
       "3     ENST00000335286.10 [A1A519-2];ENST00000379555....  FAM170A  \n",
       "4     ENST00000535628.6;ENST00000586438.5;ENST000005...   ZBTB7C  \n",
       "...                                                 ...      ...  \n",
       "1608  ENST00000371997.3 [Q9Y6Q9-3];ENST00000371998.8...    NCOA3  \n",
       "1609  ENST00000434248.6;ENST00000625684.3;ENST000006...  ZNF780B  \n",
       "1610  ENST00000426838.8 [Q9Y6X0-2];ENST00000649279.2...   SETBP1  \n",
       "1611                                 ENST00000314393.6;     ZHX2  \n",
       "1612  ENST00000303635.12 [Q9Y6Y1-1];ENST00000473578....   CAMTA1  \n",
       "\n",
       "[1613 rows x 5 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Getting all possible gene names from uniprot \n",
    "ensembl_matches = pd.read_csv(\"../data/TF_uniprotIDs_to_ENST_gene.tsv\", sep = \"\\t\")\n",
    "ensembl_matches[\"gene\"] = ensembl_matches[\"Gene Names\"].str.split(\" \").str[0]\n",
    "ensembl_matches"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "d8696e59-f094-4291-9066-667d9c51e68a",
   "metadata": {},
   "outputs": [],
   "source": [
    "files = glob.glob(\"../data/zoonomia_toga_mca/ENST*\")\n",
    "\n",
    "# Converting codon alignment to unaligned protein sequences\n",
    "for file in files:\n",
    "    # Read in as dataframe  \n",
    "    df = protfasta.read_fasta(file, invalid_sequence_action = 'convert') #Convert - to \"\"\n",
    "    df = pd.DataFrame({\"id\" : df.keys(), \"nt_seq\" : df.values()})\n",
    "\n",
    "    # Translate sequences\n",
    "    df[\"prot_seq\"] = [str(Seq(_).translate())[:-1] for _ in df[\"nt_seq\"]]\n",
    "\n",
    "    ENST = file.split(\"/\")[-1].split(\".\")[0]\n",
    "    gene =  file.split(\"/\")[-1].split(\".\")[1]\n",
    "\n",
    "    # Save protein sequences in dictionary\n",
    "    prot_dict = dict(zip(df[\"id\"], df[\"prot_seq\"]))\n",
    "    protfasta.write_fasta(prot_dict, \"../data/zoonomia_toga_mca/prot_fastas/\" + gene)\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "281d6c8b-9f2d-43ef-a542-342b0b401236",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Slow to run\n",
    "genes = [_.split(\".\")[-2] for _ in files]\n",
    "for gene in genes:\n",
    "    os.system(\"mafft --auto --quiet ../data/zoonomia_toga_mca/prot_fastas/\" + gene + \" > ../data/zoonomia_toga_mca/prot_alignments/\" + gene + \".fasta\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09239096-e3a4-46f2-aeaf-f8ddc539c123",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "28699912-7aaa-40da-9dd7-b91d3efced79",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>all_genes</th>\n",
       "      <th>len</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ZNF892</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>E2F8</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FEZF1</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ZNF312B</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>FAM170A</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1611</th>\n",
       "      <td>AFR1</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1611</th>\n",
       "      <td>KIAA0854</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1612</th>\n",
       "      <td>CAMTA1</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1612</th>\n",
       "      <td>KIAA0833</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1612</th>\n",
       "      <td>MSTP023</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3259 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     all_genes  len\n",
       "0       ZNF892    6\n",
       "1         E2F8    4\n",
       "2        FEZF1    5\n",
       "2      ZNF312B    7\n",
       "3      FAM170A    7\n",
       "...        ...  ...\n",
       "1611      AFR1    4\n",
       "1611  KIAA0854    8\n",
       "1612    CAMTA1    6\n",
       "1612  KIAA0833    8\n",
       "1612   MSTP023    7\n",
       "\n",
       "[3259 rows x 2 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# All possible gene names\n",
    "# Uniprot IDs from Lambert\n",
    "# Uniprot -> all gene names possible\n",
    "# Will query all\n",
    "ensembl_matches[\"all_genes\"] = ensembl_matches[\"Gene Names\"].str.split(\" \")\n",
    "all_gene_names = ensembl_matches[[\"all_genes\"]].explode(\"all_genes\")\n",
    "all_gene_names[\"len\"] = all_gene_names[\"all_genes\"].str.len()\n",
    "all_gene_names = all_gene_names[all_gene_names[\"len\"] > 3]\n",
    "all_gene_names"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "750337e4-9298-4d21-8f7c-3e3f79e03ac9",
   "metadata": {},
   "source": [
    "# 2. Scraping zoonomia website to get paths to full protein alignments to donwload"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "0ede5c2a-1182-46f4-afa7-924fd2ecfdee",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "from urllib.parse import urljoin\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "267cce54-546d-46b7-8f6b-46f4185db50a",
   "metadata": {},
   "outputs": [],
   "source": [
    "base_url = \"https://genome.senckenberg.de/download/TOGA/human_hg38_reference/MultipleCodonAlignments/\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "c9ef51e0-3b65-44ad-816b-f5c3e1934feb",
   "metadata": {},
   "outputs": [],
   "source": [
    "response = requests.get(base_url, verify = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "5984a6e5-cef6-41bc-99b3-63cd4ebd81ff",
   "metadata": {},
   "outputs": [],
   "source": [
    "names = []\n",
    "\n",
    "# Check if the request was successful\n",
    "if response.status_code == 200:\n",
    "    # Parse the page content with BeautifulSoup\n",
    "    soup = BeautifulSoup(response.text, \"html.parser\")\n",
    "    \n",
    "    # Find all links (anchor tags with href attributes)\n",
    "    links = soup.find_all(\"a\", href=True)\n",
    "\n",
    "    for link in links:\n",
    "        for gene in all_gene_names[\"all_genes\"]:\n",
    "            if gene in link['href']:\n",
    "                # print(gene)\n",
    "                if str(link['href']).split(\".\")[-3] == gene:\n",
    "                    names.append(link['href'])\n",
    "            \n",
    "    # # Filter the links that contain 'ELP1' in their name\n",
    "    # elp1_files = [link['href'] for link in links if \"ELP1\" in link['href']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "75961310-140c-483e-811c-db0cece375bc",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1214"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "83cdad2e-f611-47da-b625-781fda4616b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ENST00000006015.HOXA11.fasta.gz',\n",
       " 'ENST00000007699.YBX2.fasta.gz',\n",
       " 'ENST00000008391.TFAP2D.fasta.gz']"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "names[:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "b2030cae-95c6-412b-9674-299a7c45841d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check every valid enst.gene mapping\n",
    "\n",
    "with open(\"../output/toga_commands_all_TFs.txt\", \"w\") as file:\n",
    "    for name in names:\n",
    "        \n",
    "        for ENST in uniprotID_ENSTs:\n",
    "            command = \"wget --no-check-certificate https://genome.senckenberg.de/download/TOGA/human_hg38_reference/MultipleCodonAlignments/\" + name + \" -P /Users/sanjanakotha/Desktop/Staller_Lab/SFARI/data/zoonomia_toga_mca_all_TFs/codon_alignments\"\n",
    "            file.write(command + \"\\n\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f98df0bf-2a49-4e38-a3c3-6790703d7499",
   "metadata": {},
   "source": [
    "Ran gunzip *.gz in directory to unzip everything"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "afa3b0ad-6995-4119-b6be-14f10ed6500c",
   "metadata": {},
   "source": [
    "# 3. Convert codon alignment to protein alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "d25d03d9-6442-49a4-824d-5903c60b7edd",
   "metadata": {},
   "outputs": [],
   "source": [
    "! mkdir ../data/zoonomia_toga_mca_all_TFs/prot_fastas/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "16a7ac01-28fa-43d9-ba14-5c390b1af305",
   "metadata": {},
   "outputs": [],
   "source": [
    "files = glob.glob(\"/Users/sanjanakotha/Desktop/Staller_Lab/SFARI/data/zoonomia_toga_mca_all_TFs/codon_alignments/ENST*\")\n",
    "\n",
    "# Converting codon alignment to unaligned protein sequences\n",
    "for file in files:\n",
    "    # Read in as dataframe  \n",
    "    df = protfasta.read_fasta(file, invalid_sequence_action = 'convert') #Convert - to \"\"\n",
    "    df = pd.DataFrame({\"id\" : df.keys(), \"nt_seq\" : df.values()})\n",
    "\n",
    "    # Translate sequences\n",
    "    df[\"prot_seq\"] = [str(Seq(_).translate())[:-1] for _ in df[\"nt_seq\"]]\n",
    "\n",
    "    ENST = file.split(\"/\")[-1].split(\".\")[0]\n",
    "    gene =  file.split(\"/\")[-1].split(\".\")[1]\n",
    "\n",
    "    # Save protein sequences in dictionary\n",
    "    prot_dict = dict(zip(df[\"id\"], df[\"prot_seq\"]))\n",
    "    protfasta.write_fasta(prot_dict, \"../data/zoonomia_toga_mca_all_TFs/prot_fastas/\" + gene)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5ed02d7d-785e-48a6-9c3c-17d8be3e0c96",
   "metadata": {},
   "source": [
    "# 3A. Run MAFFT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "bc1332d5-afee-4e9e-a40b-5fbfa19698ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Slow to run\n",
    "genes = [_.split(\".\")[-2] for _ in files]\n",
    "for gene in genes:\n",
    "    os.system(\"mafft --auto --quiet ../data/zoonomia_toga_mca_all_TFs/prot_fastas/\" + gene + \" > ../data/zoonomia_toga_mca_all_TFs/prot_alignments/\" + gene + \".fasta\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "62f24469-8bfd-44f0-a8bb-cc16153e5d6c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
