{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "ac08a20f-23f8-495c-9ac2-4cdb441ec7e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Matplotlib is building the font cache; this may take a moment.\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "import protfasta\n",
    "import requests\n",
    "import re\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "import glob\n",
    "import collections\n",
    "\n",
    "\n",
    "from Bio import AlignIO\n",
    "import matplotlib.pyplot as plt\n",
    "from Bio.Align import AlignInfo\n",
    "\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "\n",
    "import rich\n",
    "from rich_msa import RichAlignment\n",
    "\n",
    "import logomaker\n",
    "\n",
    "#Import modules\n",
    "import gzip, logomaker, matplotlib.pyplot as plt, numpy as np, os, pandas as pd, protfasta\n",
    "\n",
    "\n",
    "import seaborn as sns\n",
    "import warnings\n",
    "warnings.filterwarnings(\"ignore\")\n",
    "import itertools\n",
    "\n",
    "\n",
    "from scipy import stats"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "659f52f6-8f06-4de2-a325-59046a816659",
   "metadata": {},
   "source": [
    "# Percent identity\n",
    "Saving the percent identity of each position, relative to the lambert TFs sequence.\n",
    "\n",
    "Step 0: Check how many of CC's genes already have the correct isoform."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "7bbcf750-fd19-4689-b47e-0e8ca1b13c58",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>uniprotID</th>\n",
       "      <th>Gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q8IUX7</td>\n",
       "      <td>AEBP1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q8WYP5</td>\n",
       "      <td>AHCTF1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P35869</td>\n",
       "      <td>AHR</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O43918</td>\n",
       "      <td>AIRE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O43823</td>\n",
       "      <td>AKAP8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>441</th>\n",
       "      <td>Q8N1G0</td>\n",
       "      <td>ZNF687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>442</th>\n",
       "      <td>P17040</td>\n",
       "      <td>ZSCAN20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>443</th>\n",
       "      <td>P98168</td>\n",
       "      <td>ZXDA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>444</th>\n",
       "      <td>P98169</td>\n",
       "      <td>ZXDB</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>445</th>\n",
       "      <td>Q2QGD7</td>\n",
       "      <td>ZXDC</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>446 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    uniprotID     Gene\n",
       "0      Q8IUX7    AEBP1\n",
       "1      Q8WYP5   AHCTF1\n",
       "2      P35869      AHR\n",
       "3      O43918     AIRE\n",
       "4      O43823    AKAP8\n",
       "..        ...      ...\n",
       "441    Q8N1G0   ZNF687\n",
       "442    P17040  ZSCAN20\n",
       "443    P98168     ZXDA\n",
       "444    P98169     ZXDB\n",
       "445    Q2QGD7     ZXDC\n",
       "\n",
       "[446 rows x 2 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Loading in info about the ADs\n",
    "cc_names = pd.read_excel(\"../data/gene_names_with_location_and_pmvs.xlsx\")\n",
    "cc_names = cc_names.rename(columns = {\"gene\" : \"Gene\", \"start\" : \"Start\", \"end\" : \"End\"})\n",
    "known_ADs = pd.read_csv(\"../output/known_ADs_considering_isoforms_and_canonical.csv\")\n",
    "known_ADs = known_ADs[known_ADs[\"TileType\"] == \"TF\"]\n",
    "cc_names = pd.merge(known_ADs[[\"uniprotID\", \"Gene\", \"Start\", \"End\"]], cc_names, how = \"left\")\n",
    "# ensts = pd.read_csv(\"../data/SFARI_TFs_with_ENST_corrected.csv\", index_col = 0)\n",
    "# cc_names = pd.merge(cc_names, ensts)\n",
    "# cc_names[\"ENST\"] = cc_names[\"ENST\"].str.split(\".\").str[0]\n",
    "# cc_names[\"len\"] = cc_names[\"End\"] - cc_names[\"Start\"] + 1\n",
    "# cc_names = cc_names.rename(columns = {\"Start\" : \"start\", \"End\" : \"end\"}).drop(columns = \"pMVS #\")\n",
    "# # cc_names.at[9, \"end\"] = 93 # to make consistent with other table\n",
    "# cc_names[\"Gene Name\"] = cc_names[\"Gene Name\"].fillna(\"AD\")\n",
    "cc_TFs = cc_names[[\"uniprotID\"]].drop_duplicates()\n",
    "\n",
    "gene_names = pd.read_csv(\"../data/TF_uniprotIDs_to_gene_names.tsv\", sep = \"\\t\")\n",
    "gene_names[\"uniprotID\"] = gene_names[\"From\"]\n",
    "gene_names[\"Gene\"] = gene_names[\"Gene Names\"].str.split(\" \").str[0]\n",
    "cc_TFs = pd.merge(cc_TFs, gene_names[[\"uniprotID\", \"Gene\"]])\n",
    "cc_TFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "df90696c-3f27-4bcb-b35a-165f349645a4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AHCTF1\n",
      "BMAL1\n",
      "ATF6\n",
      "BMAL2\n",
      "CREB1\n",
      "CXXC1\n",
      "DDIT3\n",
      "DUX4\n",
      "EBF3\n",
      "EGR2\n",
      "ELF5\n",
      "EOMES\n",
      "ERG\n",
      "ESRRA\n",
      "ESRRG\n",
      "ETS1\n",
      "ETS2\n",
      "ETV1\n",
      "ETV2\n",
      "FOXA2\n",
      "FOXM1\n",
      "GATA3\n",
      "GATA4\n",
      "GLIS1\n",
      "GLIS3\n",
      "HIF1A\n",
      "HIVEP3\n",
      "HMGA2\n",
      "IRF3\n",
      "IRF5\n",
      "IRF7\n",
      "IRX4\n",
      "KDM5B\n",
      "KMT2A\n",
      "LHX3\n",
      "MAF\n",
      "MEF2A\n",
      "MEF2C\n",
      "MEIS1\n",
      "MIXL1\n",
      "MYB\n",
      "MYCL\n",
      "MYT1L\n",
      "NANOGP8\n",
      "NFAT5\n",
      "NFATC4\n",
      "NFIA\n",
      "NFIC\n",
      "NFIX\n",
      "NFKB1\n",
      "NPAS3\n",
      "NR1H3\n",
      "NR1I2\n",
      "NR1I3\n",
      "NR3C1\n",
      "NR4A1\n",
      "NR4A3\n",
      "NRF1\n",
      "OTX2\n",
      "PAX3\n",
      "PAX6\n",
      "PAX7\n",
      "PAX8\n",
      "PHF19\n",
      "POU1F1\n",
      "PITX2\n",
      "PLSCR1\n",
      "POU2F1\n",
      "POU2F2\n",
      "PPARA\n",
      "SPI1\n",
      "RARA\n",
      "RORA\n",
      "RUNX1\n",
      "RUNX2\n",
      "RUNX3\n",
      "RXRB\n",
      "SP100\n",
      "SREBF1\n",
      "STAT1\n",
      "STAT3\n",
      "STAT5A\n",
      "STAT6\n",
      "TBX4\n",
      "TBXT\n",
      "TCF12\n",
      "TCF20\n",
      "TCF3\n",
      "TCF4\n",
      "TEAD2\n",
      "TET2\n",
      "TFEB\n",
      "THRB\n",
      "TRERF1\n",
      "USF2\n",
      "VDR\n",
      "ZBTB17\n",
      "ZFX\n",
      "ZFY\n",
      "ZNF561\n",
      "ZNF280D\n",
      "ZNF426\n",
      "ZXDA\n",
      "ZXDB\n"
     ]
    }
   ],
   "source": [
    "# Assuming everything in prot alignments folder is actually correct\n",
    "correc_isoforms = glob.glob(\"../data/zoonomia_toga_mca/prot_alignments/*\")\n",
    "correc_isoform_genes = [_.split(\"/\")[-1].split(\".\")[0] for _ in correc_isoforms]\n",
    "for gene in cc_TFs[\"Gene\"]:\n",
    "    if gene not in correc_isoform_genes:\n",
    "        print(gene)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f9896154-9d37-4282-99a2-56184b81d4df",
   "metadata": {},
   "source": [
    "Plan: For ERG and PAX6, align the correct isoform's sequence to the current protein alignment. Use that to save the percent identity per position."
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0e69eca-9287-46df-949c-d69226b4942a",
   "metadata": {},
   "source": [
    "# Add fragment: Align Lambert TFs sequence to the zoonomia alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "2a5c66d1-3338-4ab8-917c-5a5dffa986c8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>ProteinSeq</th>\n",
       "      <th>DBD_coords_merged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>sp|A0A087WUV0|ZN892_HUMAN Zinc finger protein ...</td>\n",
       "      <td>MEPEGRGSLFEDSDLLHAGNPKENDVTAVLLTPGSQELMIRDMAEA...</td>\n",
       "      <td>[[221, 243], [249, 271], [277, 299], [305, 327...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>sp|A0AVK6|E2F8_HUMAN Transcription factor E2F8...</td>\n",
       "      <td>MENEKENLFCEPHKRGLMKTPLKESTTANIVLAEIQPDFGPLTTPT...</td>\n",
       "      <td>[[114, 182], [262, 347]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>sp|A0PJY2|FEZF1_HUMAN Fez family zinc finger p...</td>\n",
       "      <td>MDSSCHNATTKMLATAPARGNMMSTSKPLAFSIERIMARTPEPKAL...</td>\n",
       "      <td>[[260, 282], [288, 310], [316, 338], [344, 366...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>sp|A1A519|F170A_HUMAN Protein FAM170A OS=Homo ...</td>\n",
       "      <td>MKRRQKRKHLENEESQETAEKGGGMSKSQEDALQPGSTRVAKGWSQ...</td>\n",
       "      <td>[[1, 330]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>sp|A1YPR0|ZBT7C_HUMAN Zinc finger and BTB doma...</td>\n",
       "      <td>MANDIDELIGIPFPNHSSEVLCSLNEQRHDGLLCDVLLVVQEQEYR...</td>\n",
       "      <td>[[364, 386], [392, 414], [420, 442], [448, 469]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1608</th>\n",
       "      <td>sp|Q9Y6Q9|NCOA3_HUMAN Nuclear receptor coactiv...</td>\n",
       "      <td>MSGLGENLDPLASDSRKRKLPCDTPGQGLTCSGEKRRREQESKYIE...</td>\n",
       "      <td>[[31, 83]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1609</th>\n",
       "      <td>sp|Q9Y6R6|Z780B_HUMAN Zinc finger protein 780B...</td>\n",
       "      <td>MVHGSVTFRDVAIDFSQEEWECLQPDQRTLYRDVMLENYSHLISLG...</td>\n",
       "      <td>[[165, 187], [193, 215], [221, 243], [249, 271...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1610</th>\n",
       "      <td>sp|Q9Y6X0|SETBP_HUMAN SET-binding protein OS=H...</td>\n",
       "      <td>MESRETLSSSRQRGGESDFLPVSSAKPPAAPGCAGEPLLSTPGPGK...</td>\n",
       "      <td>[[583, 596], [1015, 1027], [1450, 1462]]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1611</th>\n",
       "      <td>sp|Q9Y6X8|ZHX2_HUMAN Zinc fingers and homeobox...</td>\n",
       "      <td>MASKRKSTTPCMVRTSQVVEQDVPEEVDRAKEKGIGTPQPDVAKDS...</td>\n",
       "      <td>[[78, 101], [110, 133], [271, 317], [442, 496]...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1612</th>\n",
       "      <td>sp|Q9Y6Y1|CMTA1_HUMAN Calmodulin-binding trans...</td>\n",
       "      <td>MWRAEGKWLPKTSRKSVSQSVFCGTSTYCVLNTVPPIEDDHGNSNS...</td>\n",
       "      <td>[[69, 182]]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1613 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     id  \\\n",
       "0     sp|A0A087WUV0|ZN892_HUMAN Zinc finger protein ...   \n",
       "1     sp|A0AVK6|E2F8_HUMAN Transcription factor E2F8...   \n",
       "2     sp|A0PJY2|FEZF1_HUMAN Fez family zinc finger p...   \n",
       "3     sp|A1A519|F170A_HUMAN Protein FAM170A OS=Homo ...   \n",
       "4     sp|A1YPR0|ZBT7C_HUMAN Zinc finger and BTB doma...   \n",
       "...                                                 ...   \n",
       "1608  sp|Q9Y6Q9|NCOA3_HUMAN Nuclear receptor coactiv...   \n",
       "1609  sp|Q9Y6R6|Z780B_HUMAN Zinc finger protein 780B...   \n",
       "1610  sp|Q9Y6X0|SETBP_HUMAN SET-binding protein OS=H...   \n",
       "1611  sp|Q9Y6X8|ZHX2_HUMAN Zinc fingers and homeobox...   \n",
       "1612  sp|Q9Y6Y1|CMTA1_HUMAN Calmodulin-binding trans...   \n",
       "\n",
       "                                             ProteinSeq  \\\n",
       "0     MEPEGRGSLFEDSDLLHAGNPKENDVTAVLLTPGSQELMIRDMAEA...   \n",
       "1     MENEKENLFCEPHKRGLMKTPLKESTTANIVLAEIQPDFGPLTTPT...   \n",
       "2     MDSSCHNATTKMLATAPARGNMMSTSKPLAFSIERIMARTPEPKAL...   \n",
       "3     MKRRQKRKHLENEESQETAEKGGGMSKSQEDALQPGSTRVAKGWSQ...   \n",
       "4     MANDIDELIGIPFPNHSSEVLCSLNEQRHDGLLCDVLLVVQEQEYR...   \n",
       "...                                                 ...   \n",
       "1608  MSGLGENLDPLASDSRKRKLPCDTPGQGLTCSGEKRRREQESKYIE...   \n",
       "1609  MVHGSVTFRDVAIDFSQEEWECLQPDQRTLYRDVMLENYSHLISLG...   \n",
       "1610  MESRETLSSSRQRGGESDFLPVSSAKPPAAPGCAGEPLLSTPGPGK...   \n",
       "1611  MASKRKSTTPCMVRTSQVVEQDVPEEVDRAKEKGIGTPQPDVAKDS...   \n",
       "1612  MWRAEGKWLPKTSRKSVSQSVFCGTSTYCVLNTVPPIEDDHGNSNS...   \n",
       "\n",
       "                                      DBD_coords_merged  \n",
       "0     [[221, 243], [249, 271], [277, 299], [305, 327...  \n",
       "1                              [[114, 182], [262, 347]]  \n",
       "2     [[260, 282], [288, 310], [316, 338], [344, 366...  \n",
       "3                                            [[1, 330]]  \n",
       "4      [[364, 386], [392, 414], [420, 442], [448, 469]]  \n",
       "...                                                 ...  \n",
       "1608                                         [[31, 83]]  \n",
       "1609  [[165, 187], [193, 215], [221, 243], [249, 271...  \n",
       "1610           [[583, 596], [1015, 1027], [1450, 1462]]  \n",
       "1611  [[78, 101], [110, 133], [271, 317], [442, 496]...  \n",
       "1612                                        [[69, 182]]  \n",
       "\n",
       "[1613 rows x 3 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "lambert_TFs = pd.read_csv(\"../output/lambert_TFs_10-21-24_with_DBD_coords.csv\", index_col = 0)\n",
    "lambert_TFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "423dbfdd-9884-406f-bd32-6b1fcb7a69cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ! mkdir ../data/zoonomia_toga_mca/prot_alignments_for_vis/\n",
    "#! mkdir ../data/zoonomia_toga_mca/prot_seqs_for_vis/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "640d7072-5edc-4b96-996c-ee6edeabddc9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['../data/zoonomia_toga_mca/prot_alignments/TP63.fasta',\n",
       " '../data/zoonomia_toga_mca/prot_alignments/RARG.fasta']"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_prot_alignments = glob.glob(\"../data/zoonomia_toga_mca/prot_alignments/*\")\n",
    "full_prot_alignments[:2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "25560f5d-9be3-41f6-a452-add2c9a7cc47",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/BMAL1.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.HPI2KagoW7/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.HPI2KagoW7/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.HPI2KagoW7/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.HPI2KagoW7/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.HPI2KagoW7/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/BMAL2.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.SxZfoAKGex/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.SxZfoAKGex/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.SxZfoAKGex/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.SxZfoAKGex/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.SxZfoAKGex/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/CXXC1.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.l17aaQqIeC/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.l17aaQqIeC/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.l17aaQqIeC/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.l17aaQqIeC/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.l17aaQqIeC/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/DUX4.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.7QhLSVONjI/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.7QhLSVONjI/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.7QhLSVONjI/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.7QhLSVONjI/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.7QhLSVONjI/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/HIVEP3.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.zKVEppinFA/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.zKVEppinFA/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.zKVEppinFA/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.zKVEppinFA/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.zKVEppinFA/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/KMT2A.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.wAH97QSx5K/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.wAH97QSx5K/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.wAH97QSx5K/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.wAH97QSx5K/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.wAH97QSx5K/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/NANOGP8.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.xVq4NXTL56/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.xVq4NXTL56/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.xVq4NXTL56/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.xVq4NXTL56/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.xVq4NXTL56/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/PLSCR1.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.NdzB5wwCYb/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.NdzB5wwCYb/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.NdzB5wwCYb/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.NdzB5wwCYb/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.NdzB5wwCYb/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/SP100.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.1zMetKlc0w/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.1zMetKlc0w/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.1zMetKlc0w/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.1zMetKlc0w/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.1zMetKlc0w/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/STAT5A.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.0340XHnTqu/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.0340XHnTqu/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.0340XHnTqu/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.0340XHnTqu/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.0340XHnTqu/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/TCF20.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.uST9CPOGGR/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.uST9CPOGGR/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.uST9CPOGGR/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.uST9CPOGGR/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.uST9CPOGGR/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/TET2.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.yNZc0VRUJq/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.yNZc0VRUJq/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.yNZc0VRUJq/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.yNZc0VRUJq/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.yNZc0VRUJq/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/ZFX.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.zYUDk7El2p/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.zYUDk7El2p/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.zYUDk7El2p/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.zYUDk7El2p/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.zYUDk7El2p/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/ZFY.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.vFst5CxjvJ/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.vFst5CxjvJ/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.vFst5CxjvJ/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.vFst5CxjvJ/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.vFst5CxjvJ/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/ZNF561.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.mMLfstCMoz/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.mMLfstCMoz/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.mMLfstCMoz/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.mMLfstCMoz/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.mMLfstCMoz/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/ZNF280D.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.BuUcP2oCmY/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.BuUcP2oCmY/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.BuUcP2oCmY/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.BuUcP2oCmY/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.BuUcP2oCmY/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/ZNF426.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.JYnaOtwgOH/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.JYnaOtwgOH/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.JYnaOtwgOH/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.JYnaOtwgOH/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.JYnaOtwgOH/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/ZXDA.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.MoQIqIxikd/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.MoQIqIxikd/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.MoQIqIxikd/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.MoQIqIxikd/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.MoQIqIxikd/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n",
      "/usr/local/bin/mafft: Cannot open ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/ZXDB.fasta.\n",
      "\n",
      "/usr/local/bin/mafft: line 1275: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.yoD648xc9F/_addfile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.yoD648xc9F/addsize\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1277: [: -eq: unary operator expected\n",
      "/usr/local/bin/mafft: line 1291: /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.yoD648xc9F/infile: No such file or directory\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.yoD648xc9F/size\n",
      " source line number 1\n",
      "awk: can't open file /var/folders/hw/xx051vr9457c7lrngf2mypgr0000gn/T/mafft.XXXXXXXXXX.yoD648xc9F/size\n",
      " source line number 1\n",
      "/usr/local/bin/mafft: line 1295: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1300: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1305: [: too many arguments\n",
      "/usr/local/bin/mafft: line 1310: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1315: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1322: [: -lt: unary operator expected\n",
      "/usr/local/bin/mafft: line 1329: [: -lt: unary operator expected\n",
      "------------------------------------------------------------------------------\n",
      "  MAFFT v7.526 (2024/Apr/26)\n",
      "  https://mafft.cbrc.jp/alignment/software/\n",
      "  MBE 30:772-780 (2013), NAR 30:3059-3066 (2002)\n",
      "------------------------------------------------------------------------------\n",
      "High speed:\n",
      "  % mafft in > out\n",
      "  % mafft --retree 1 in > out (fast)\n",
      "\n",
      "High accuracy (for <~200 sequences x <~2,000 aa/nt):\n",
      "  % mafft --maxiterate 1000 --localpair  in > out (% linsi in > out is also ok)\n",
      "  % mafft --maxiterate 1000 --genafpair  in > out (% einsi in > out)\n",
      "  % mafft --maxiterate 1000 --globalpair in > out (% ginsi in > out)\n",
      "\n",
      "If unsure which option to use:\n",
      "  % mafft --auto in > out\n",
      "\n",
      "--op # :         Gap opening penalty, default: 1.53\n",
      "--ep # :         Offset (works like gap extension penalty), default: 0.0\n",
      "--maxiterate # : Maximum number of iterative refinement, default: 0\n",
      "--clustalout :   Output: clustal format, default: fasta\n",
      "--reorder :      Outorder: aligned, default: input order\n",
      "--quiet :        Do not report progress\n",
      "--thread # :     Number of threads (if unsure, --thread -1)\n",
      "--dash :         Add structural information (Rozewicki et al, submitted)\n"
     ]
    }
   ],
   "source": [
    "for gene in cc_TFs[\"Gene\"]:\n",
    "    uniprotID = dict(zip(cc_TFs[\"Gene\"], cc_TFs[\"uniprotID\"]))[gene]\n",
    "    df = lambert_TFs[lambert_TFs[\"id\"].str.contains(uniprotID)]\n",
    "    protfasta.write_fasta([(\"LambertSeq\", df[\"ProteinSeq\"].iloc[0])], \"../data/zoonomia_toga_mca/prot_seqs_for_vis/\" + gene)\n",
    "    #display(df)\n",
    "\n",
    "    if gene in correc_isoform_genes:\n",
    "        command = \"mafft --auto --quiet --add ../data/zoonomia_toga_mca/prot_seqs_for_vis/\" + gene + \" --reorder --thread -1 ../data/zoonomia_toga_mca/prot_alignments/\" + gene + \".fasta > ../data/zoonomia_toga_mca/prot_alignments_for_vis/\" + gene + \".fasta\"\n",
    "    else:\n",
    "        command = \"mafft --auto --quiet --add ../data/zoonomia_toga_mca/prot_seqs_for_vis/\" + gene + \" --reorder --thread -1 ../data/zoonomia_toga_mca/prot_alignments_wrong_isoforms/\" + gene + \".fasta > ../data/zoonomia_toga_mca/prot_alignments_for_vis/\" + gene + \".fasta\"\n",
    "     \n",
    "    os.system(command)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "133ebf29-9151-403b-a9f6-69540db4cf32",
   "metadata": {},
   "source": [
    "# Save percent identities for each"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "f483240b-407f-40ae-a167-2535f53d897b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ZBTB17',\n",
       " 'TP63',\n",
       " 'RARG',\n",
       " 'NCOA2',\n",
       " 'ZNF148',\n",
       " 'SPIB',\n",
       " 'NEUROD6',\n",
       " 'FOXN1',\n",
       " 'HOXA6',\n",
       " 'GRHL1',\n",
       " 'YY2',\n",
       " 'SRF',\n",
       " 'ARID5A',\n",
       " 'SOX2',\n",
       " 'PAX6',\n",
       " 'AR',\n",
       " 'CREBZF',\n",
       " 'ERG',\n",
       " 'ZNF641',\n",
       " 'OTX2',\n",
       " 'ETV4',\n",
       " 'SMAD1',\n",
       " 'TRAFD1',\n",
       " 'MYF6',\n",
       " 'NR1I2',\n",
       " 'SMAD3',\n",
       " 'GCM1',\n",
       " 'KDM5B',\n",
       " 'E2F2',\n",
       " 'MITF',\n",
       " 'FOS',\n",
       " 'MAFA',\n",
       " 'SREBF2',\n",
       " 'KLF1',\n",
       " 'GRHL3',\n",
       " 'MYOG',\n",
       " 'VAX1',\n",
       " 'ZSCAN20',\n",
       " 'MEIS1',\n",
       " 'THRB',\n",
       " 'HBP1',\n",
       " 'SOX18',\n",
       " 'HOXB9',\n",
       " 'AHR',\n",
       " 'TIGD1',\n",
       " 'LEUTX',\n",
       " 'MSX2',\n",
       " 'KLF16',\n",
       " 'MLXIP',\n",
       " 'KLF5',\n",
       " 'NKX6-1',\n",
       " 'FOXF2',\n",
       " 'POU5F1',\n",
       " 'FOXJ2',\n",
       " 'RARA',\n",
       " 'FOXR2',\n",
       " 'MZF1',\n",
       " 'CREM',\n",
       " 'ETV2',\n",
       " 'USF2',\n",
       " 'ZNF687',\n",
       " 'ELK4',\n",
       " 'STAT3',\n",
       " 'TWIST1',\n",
       " 'EOMES',\n",
       " 'MYBL1',\n",
       " 'SOX4',\n",
       " 'PAX2',\n",
       " 'STAT1',\n",
       " 'PRRX1',\n",
       " 'TEF',\n",
       " 'POU2AF1',\n",
       " 'BSX',\n",
       " 'TFEB',\n",
       " 'SMAD5',\n",
       " 'E2F4',\n",
       " 'TIGD7',\n",
       " 'NEUROD2',\n",
       " 'SP9',\n",
       " 'TBX21',\n",
       " 'PHF1',\n",
       " 'ZNF16',\n",
       " 'PITX2',\n",
       " 'KLF7',\n",
       " 'HOXA2',\n",
       " 'STAT6',\n",
       " 'AHCTF1',\n",
       " 'PAX5',\n",
       " 'ZBTB48',\n",
       " 'ELK1',\n",
       " 'SOX1',\n",
       " 'ARNT',\n",
       " 'DMRT1',\n",
       " 'ALX3',\n",
       " 'MYF5',\n",
       " 'ATMIN',\n",
       " 'FOSL1',\n",
       " 'HOXD9',\n",
       " 'IKZF1',\n",
       " 'NR1I3',\n",
       " 'REL',\n",
       " 'JUND',\n",
       " 'OTX1',\n",
       " 'E2F3',\n",
       " 'ZXDC',\n",
       " 'FOXN2',\n",
       " 'CC2D1A',\n",
       " 'EBF3',\n",
       " 'CREB1',\n",
       " 'TBPL2',\n",
       " 'NCOA1',\n",
       " 'YY1',\n",
       " 'KLF13',\n",
       " 'GRHL2',\n",
       " 'HOXD13',\n",
       " 'ATF6B',\n",
       " 'THRA',\n",
       " 'FOSB',\n",
       " 'TFCP2',\n",
       " 'MEIS2',\n",
       " 'HOXA7',\n",
       " 'VAX2',\n",
       " 'KLF2',\n",
       " 'EBF1',\n",
       " 'CREB3',\n",
       " 'SPIC',\n",
       " 'RFX1',\n",
       " 'NCOA3',\n",
       " 'ALX1',\n",
       " 'CASZ1',\n",
       " 'E2F1',\n",
       " 'GCM2',\n",
       " 'POU4F2',\n",
       " 'ETV5',\n",
       " 'SREBF1',\n",
       " 'ELK3',\n",
       " 'PAX7',\n",
       " 'STAT4',\n",
       " 'RBSN',\n",
       " 'MAFB',\n",
       " 'HAND2',\n",
       " 'SOX3',\n",
       " 'ETV1',\n",
       " 'HHEX',\n",
       " 'SMAD4',\n",
       " 'E2F5',\n",
       " 'JUNB',\n",
       " 'ZNF644',\n",
       " 'TFEC',\n",
       " 'NR3C1',\n",
       " 'FOXR1',\n",
       " 'SOX7',\n",
       " 'MYBL2',\n",
       " 'TET1',\n",
       " 'PAX3',\n",
       " 'ARGFX',\n",
       " 'USF1',\n",
       " 'FOXF1',\n",
       " 'KLF6',\n",
       " 'KLF15',\n",
       " 'HNF4A',\n",
       " 'MSX1',\n",
       " 'MIXL1',\n",
       " 'PTF1A',\n",
       " 'PLAG1',\n",
       " 'LEF1',\n",
       " 'NFE2',\n",
       " 'FOXN4',\n",
       " 'FOXJ1',\n",
       " 'NANOG',\n",
       " 'TBX20',\n",
       " 'PRMT3',\n",
       " 'TBX22',\n",
       " 'FOXJ3',\n",
       " 'NEUROD1',\n",
       " 'TIGD4',\n",
       " 'DDIT3',\n",
       " 'PITX1',\n",
       " 'AIRE',\n",
       " 'VDR',\n",
       " 'PRRX2',\n",
       " 'PROP1',\n",
       " 'STAT2',\n",
       " 'PHOX2A',\n",
       " 'PGR',\n",
       " 'ETV3',\n",
       " 'NOBOX',\n",
       " 'MYB',\n",
       " 'SIX4',\n",
       " 'PDX1',\n",
       " 'NR4A2',\n",
       " 'RUNX2',\n",
       " 'HMGA1',\n",
       " 'CDX2',\n",
       " 'NFATC1',\n",
       " 'FOXO3',\n",
       " 'RXRB',\n",
       " 'SOX9',\n",
       " 'TCF3',\n",
       " 'TFDP1',\n",
       " 'GLIS2',\n",
       " 'KLF8',\n",
       " 'NR1H2',\n",
       " 'NFIX',\n",
       " 'STAT5B',\n",
       " 'POU3F1',\n",
       " 'GLI2',\n",
       " 'ZNF143',\n",
       " 'ZNF592',\n",
       " 'TEAD2',\n",
       " 'GATA2',\n",
       " 'SOX11',\n",
       " 'MYCN',\n",
       " 'MBD2',\n",
       " 'NFIC',\n",
       " 'SP6',\n",
       " 'CEBPE',\n",
       " 'ESR1',\n",
       " 'AKAP8',\n",
       " 'MYCL',\n",
       " 'NFIA',\n",
       " 'CREB3L4',\n",
       " 'HOXB2',\n",
       " 'TBX4',\n",
       " 'AEBP1',\n",
       " 'PHF19',\n",
       " 'TCF7L1',\n",
       " 'MYRFL',\n",
       " 'NEUROG2',\n",
       " 'ETS1',\n",
       " 'IRF2',\n",
       " 'DLX3',\n",
       " 'ELF2',\n",
       " 'GMEB2',\n",
       " 'NFATC3',\n",
       " 'FOXO1',\n",
       " 'CRX',\n",
       " 'MYOD1',\n",
       " 'TP73',\n",
       " 'MTERF3',\n",
       " 'JAZF1',\n",
       " 'MGA',\n",
       " 'FOXC1',\n",
       " 'ATF2',\n",
       " 'KMT2B',\n",
       " 'TFAP2B',\n",
       " 'MTF1',\n",
       " 'NFE2L1',\n",
       " 'ZIC2',\n",
       " 'CSRNP1',\n",
       " 'NKX2-2',\n",
       " 'CDX4',\n",
       " 'TAL1',\n",
       " 'HSF2',\n",
       " 'ATF6',\n",
       " 'EGR1',\n",
       " 'DBP',\n",
       " 'SIX2',\n",
       " 'GATA4',\n",
       " 'TEAD4',\n",
       " 'FOXA1',\n",
       " 'PPARD',\n",
       " 'HOXB6',\n",
       " 'SOX17',\n",
       " 'CEBPA',\n",
       " 'TCF12',\n",
       " 'LHX2',\n",
       " 'TBXT',\n",
       " 'MEF2D',\n",
       " 'NR6A1',\n",
       " 'TP53',\n",
       " 'IRF6',\n",
       " 'RORA',\n",
       " 'FOXM1',\n",
       " 'PHF20',\n",
       " 'ZNF513',\n",
       " 'NR5A2',\n",
       " 'ELF4',\n",
       " 'KAT7',\n",
       " 'NPAS4',\n",
       " 'RORC',\n",
       " 'CREB3L2',\n",
       " 'SOX15',\n",
       " 'MYRF',\n",
       " 'ZNF777',\n",
       " 'GATA6',\n",
       " 'HOXA13',\n",
       " 'MAF',\n",
       " 'ATF4',\n",
       " 'TFAP2D',\n",
       " 'EGR3',\n",
       " 'RELA',\n",
       " 'NFAT5',\n",
       " 'PAX9',\n",
       " 'NFE2L3',\n",
       " 'DMTF1',\n",
       " 'PLAGL2',\n",
       " 'MLXIPL',\n",
       " 'CAMTA1',\n",
       " 'POU2F1',\n",
       " 'NRF1',\n",
       " 'IRF3',\n",
       " 'MEF2A',\n",
       " 'ELF3',\n",
       " 'NPAS3',\n",
       " 'FOXI1',\n",
       " 'GLMP',\n",
       " 'NEUROG3',\n",
       " 'GLIS1',\n",
       " 'SP5',\n",
       " 'PPARA',\n",
       " 'TBX5',\n",
       " 'SOX12',\n",
       " 'HOXB3',\n",
       " 'GATA1',\n",
       " 'TEAD1',\n",
       " 'GLI1',\n",
       " 'CEBPD',\n",
       " 'NR1D1',\n",
       " 'FLI1',\n",
       " 'ARNT2',\n",
       " 'HIF3A',\n",
       " 'RUNX1',\n",
       " 'HMGA2',\n",
       " 'ATF3',\n",
       " 'NR4A1',\n",
       " 'TFAP2C',\n",
       " 'EGR4',\n",
       " 'RXRA',\n",
       " 'ONECUT1',\n",
       " 'NFATC2',\n",
       " 'CDX1',\n",
       " 'SRCAP',\n",
       " 'ZNF532',\n",
       " 'TFE3',\n",
       " 'HLF',\n",
       " 'GMEB1',\n",
       " 'SOX8',\n",
       " 'NKX2-5',\n",
       " 'NR4A3',\n",
       " 'HSF5',\n",
       " 'ESRRA',\n",
       " 'RUNX3',\n",
       " 'FOXC2',\n",
       " 'MYC',\n",
       " 'MYT1L',\n",
       " 'HNF1B',\n",
       " 'TEAD3',\n",
       " 'TCF7L2',\n",
       " 'GATA3',\n",
       " 'ZNF473',\n",
       " 'SOX10',\n",
       " 'HOXB1',\n",
       " 'SP7',\n",
       " 'ESR2',\n",
       " 'POU5F1B',\n",
       " 'GLI3',\n",
       " 'ELF1',\n",
       " 'MEF2C',\n",
       " 'ZNF516',\n",
       " 'IRF1',\n",
       " 'NR1H3',\n",
       " 'TRERF1',\n",
       " 'ZFPM2',\n",
       " 'ETS2',\n",
       " 'NEUROG1',\n",
       " 'GLIS3',\n",
       " 'FOXI3',\n",
       " 'ZNF326',\n",
       " 'KLF9',\n",
       " 'CEBPB',\n",
       " 'XBP1',\n",
       " 'HOXB13',\n",
       " 'PPARG',\n",
       " 'TBX3',\n",
       " 'SOX14',\n",
       " 'CREB3L3',\n",
       " 'SP3',\n",
       " 'FOXA2',\n",
       " 'EPAS1',\n",
       " 'IRX4',\n",
       " 'IRF5',\n",
       " 'ELF5',\n",
       " 'JUN',\n",
       " 'LHX1',\n",
       " 'NFATC4',\n",
       " 'FOXO6',\n",
       " 'POU1F1',\n",
       " 'RXRG',\n",
       " 'NKX2-1',\n",
       " 'JRKL',\n",
       " 'PAX8',\n",
       " 'NFE2L2',\n",
       " 'OTP',\n",
       " 'HINFP',\n",
       " 'EGR2',\n",
       " 'HIC2',\n",
       " 'ATF5',\n",
       " 'HSF1',\n",
       " 'HOXD4',\n",
       " 'RELB',\n",
       " 'AKAP8L',\n",
       " 'NRL',\n",
       " 'ESRRG',\n",
       " 'ATF7',\n",
       " 'NFYA',\n",
       " 'POU2F2',\n",
       " 'CAMTA2',\n",
       " 'FOXO4',\n",
       " 'TCF4',\n",
       " 'PLAGL1',\n",
       " 'OVOL2',\n",
       " 'SPI1',\n",
       " 'LMX1A',\n",
       " 'LHX3',\n",
       " 'ZNF654',\n",
       " 'NR5A1',\n",
       " 'IRF7',\n",
       " 'ZNF200',\n",
       " 'HOXA10',\n",
       " 'NFKB1',\n",
       " 'GATA5',\n",
       " 'HIF1A',\n",
       " 'SP1',\n",
       " 'CREB3L1',\n",
       " 'ARX',\n",
       " 'HOXB7']"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aligned_with_lambert = glob.glob(\"../data/zoonomia_toga_mca/prot_alignments_for_vis/*.fasta\")\n",
    "aligned_with_lambert = [_.split(\"/\")[-1].split(\".\")[0] for _ in aligned_with_lambert]\n",
    "aligned_with_lambert"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "299a6041-2457-4f87-a65e-7a63d39e145a",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "AEBP1\n",
      "AHCTF1\n",
      "AHR\n",
      "AIRE\n",
      "AKAP8\n",
      "AKAP8L\n",
      "ALX1\n",
      "ALX3\n",
      "AR\n",
      "ARGFX\n",
      "ARID5A\n",
      "ARNT\n",
      "ARNT2\n",
      "BMAL1\n",
      "ARX\n",
      "ATF2\n",
      "ATF3\n",
      "ATF4\n",
      "ATF5\n",
      "ATF6B\n",
      "ATF6\n",
      "ATF7\n",
      "ATMIN\n",
      "BMAL2\n",
      "BSX\n",
      "CC2D1A\n",
      "CAMTA1\n",
      "CAMTA2\n",
      "CASZ1\n",
      "CDX1\n",
      "CDX2\n",
      "CDX4\n",
      "CEBPA\n",
      "CEBPB\n",
      "CEBPD\n",
      "CEBPE\n",
      "CREB3L1\n",
      "CREB1\n",
      "CREB3\n",
      "CREB3L2\n",
      "CREB3L3\n",
      "CREB3L4\n",
      "CREM\n",
      "CRX\n",
      "CSRNP1\n",
      "CXXC1\n",
      "DBP\n",
      "DDIT3\n",
      "DLX3\n",
      "DMRT1\n",
      "DMTF1\n",
      "DUX4\n",
      "E2F1\n",
      "E2F2\n",
      "E2F3\n",
      "E2F4\n",
      "E2F5\n",
      "EBF1\n",
      "EBF3\n",
      "EGR1\n",
      "EGR2\n",
      "EGR3\n",
      "EGR4\n",
      "ELF1\n",
      "ELF2\n",
      "ELF3\n",
      "ELF4\n",
      "ELF5\n",
      "ELK1\n",
      "ELK3\n",
      "ELK4\n",
      "EOMES\n",
      "EPAS1\n",
      "ERG\n",
      "ESR1\n",
      "ESR2\n",
      "ESRRA\n",
      "ESRRG\n",
      "ETS1\n",
      "ETS2\n",
      "ETV1\n",
      "ETV2\n",
      "ETV3\n",
      "ETV4\n",
      "ETV5\n",
      "FLI1\n",
      "ZFPM2\n",
      "FOS\n",
      "FOSB\n",
      "FOSL1\n",
      "FOXA1\n",
      "FOXA2\n",
      "FOXC1\n",
      "FOXC2\n",
      "FOXF1\n",
      "FOXF2\n",
      "FOXI1\n",
      "FOXI3\n",
      "FOXJ1\n",
      "FOXJ2\n",
      "FOXJ3\n",
      "FOXM1\n",
      "FOXN1\n",
      "FOXN2\n",
      "FOXN4\n",
      "FOXO1\n",
      "FOXO3\n",
      "FOXO4\n",
      "FOXO6\n",
      "FOXR1\n",
      "FOXR2\n",
      "GATA1\n",
      "GATA2\n",
      "GATA3\n",
      "GATA4\n",
      "GATA5\n",
      "GATA6\n",
      "GCM1\n",
      "GCM2\n",
      "GLI1\n",
      "GLI2\n",
      "GLI3\n",
      "GLIS1\n",
      "GLIS2\n",
      "GLIS3\n",
      "GLMP\n",
      "GMEB1\n",
      "GMEB2\n",
      "GRHL1\n",
      "GRHL2\n",
      "GRHL3\n",
      "HAND2\n",
      "HBP1\n",
      "HHEX\n",
      "HIC2\n",
      "HIF1A\n",
      "HIF3A\n",
      "HINFP\n",
      "HIVEP3\n",
      "HLF\n",
      "HMGA1\n",
      "HMGA2\n",
      "HNF1B\n",
      "HNF4A\n",
      "HOXA10\n",
      "HOXA13\n",
      "HOXA2\n",
      "HOXA6\n",
      "HOXA7\n",
      "HOXB1\n",
      "HOXB13\n",
      "HOXB2\n",
      "HOXB3\n",
      "HOXB6\n",
      "HOXB7\n",
      "HOXB9\n",
      "HOXD13\n",
      "HOXD4\n",
      "HOXD9\n",
      "HSF2\n",
      "HSF1\n",
      "HSF5\n",
      "IKZF1\n",
      "IRF1\n",
      "IRF2\n",
      "IRF3\n",
      "IRF5\n",
      "IRF6\n",
      "IRF7\n",
      "IRX4\n",
      "JAZF1\n",
      "JRKL\n",
      "JUN\n",
      "JUNB\n",
      "JUND\n",
      "KAT7\n",
      "KDM5B\n",
      "KLF1\n",
      "KLF13\n",
      "KLF15\n",
      "KLF16\n",
      "KLF2\n",
      "KLF5\n",
      "KLF6\n",
      "KLF7\n",
      "KLF8\n",
      "KLF9\n",
      "KMT2A\n",
      "KMT2B\n",
      "LEF1\n",
      "LEUTX\n",
      "LHX1\n",
      "LHX2\n",
      "LHX3\n",
      "LMX1A\n",
      "MAF\n",
      "MAFA\n",
      "MAFB\n",
      "MBD2\n",
      "MEF2A\n",
      "MEF2C\n",
      "MEF2D\n",
      "MEIS1\n",
      "MEIS2\n",
      "MGA\n",
      "MITF\n",
      "MIXL1\n",
      "MLXIP\n",
      "MLXIPL\n",
      "MSX1\n",
      "MSX2\n",
      "MTERF3\n",
      "MTF1\n",
      "MYB\n",
      "MYBL1\n",
      "MYBL2\n",
      "MYC\n",
      "MYCL\n",
      "MYCN\n",
      "MYF5\n",
      "MYF6\n",
      "MYOD1\n",
      "MYOG\n",
      "MYRF\n",
      "MYRFL\n",
      "MYT1L\n",
      "MZF1\n",
      "NANOG\n",
      "NANOGP8\n",
      "NCOA1\n",
      "NCOA2\n",
      "NCOA3\n",
      "NEUROD1\n",
      "NEUROD2\n",
      "NEUROD6\n",
      "NEUROG1\n",
      "NEUROG2\n",
      "NEUROG3\n",
      "NFAT5\n",
      "NFATC1\n",
      "NFATC2\n",
      "NFATC3\n",
      "NFATC4\n",
      "NFE2\n",
      "NFE2L1\n",
      "NFE2L2\n",
      "NFE2L3\n",
      "NFIA\n",
      "NFIC\n",
      "NFIX\n",
      "NFKB1\n",
      "NFYA\n",
      "NKX2-1\n",
      "NKX2-2\n",
      "NKX2-5\n",
      "NKX6-1\n",
      "NOBOX\n",
      "NPAS3\n",
      "NPAS4\n",
      "NR1D1\n",
      "NR1H2\n",
      "NR1H3\n",
      "NR1I2\n",
      "NR1I3\n",
      "NR3C1\n",
      "NR4A1\n",
      "NR4A2\n",
      "NR4A3\n",
      "NR5A1\n",
      "NR5A2\n",
      "NR6A1\n",
      "NRF1\n",
      "NRL\n",
      "ONECUT1\n",
      "OTP\n",
      "OTX1\n",
      "OTX2\n",
      "OVOL2\n",
      "POU5F1\n",
      "PAX2\n",
      "PAX3\n",
      "PAX5\n",
      "PAX6\n",
      "PAX7\n",
      "PAX8\n",
      "PAX9\n",
      "PDX1\n",
      "PGR\n",
      "PHF1\n",
      "PHF19\n",
      "PHF20\n",
      "PHOX2A\n",
      "POU1F1\n",
      "PITX1\n",
      "PITX2\n",
      "PLAG1\n",
      "PLAGL1\n",
      "PLAGL2\n",
      "PLSCR1\n",
      "POU2AF1\n",
      "POU2F1\n",
      "POU2F2\n",
      "POU3F1\n",
      "POU4F2\n",
      "POU5F1B\n",
      "PPARA\n",
      "PPARD\n",
      "PPARG\n",
      "PRMT3\n",
      "PROP1\n",
      "PRRX1\n",
      "PRRX2\n",
      "PTF1A\n",
      "SPI1\n",
      "RARA\n",
      "RARG\n",
      "RBSN\n",
      "REL\n",
      "RELA\n",
      "RELB\n",
      "RFX1\n",
      "RORA\n",
      "RORC\n",
      "RUNX1\n",
      "RUNX2\n",
      "RUNX3\n",
      "RXRA\n",
      "RXRB\n",
      "RXRG\n",
      "SIX2\n",
      "SIX4\n",
      "SMAD1\n",
      "SMAD3\n",
      "SMAD4\n",
      "SMAD5\n",
      "SOX1\n",
      "SOX10\n",
      "SOX11\n",
      "SOX12\n",
      "SOX14\n",
      "SOX15\n",
      "SOX17\n",
      "SOX18\n",
      "SOX2\n",
      "SOX3\n",
      "SOX4\n",
      "SOX7\n",
      "SOX8\n",
      "SOX9\n",
      "SP1\n",
      "SP100\n",
      "SP3\n",
      "SP5\n",
      "SP6\n",
      "SP7\n",
      "SP9\n",
      "SPIB\n",
      "SPIC\n",
      "SREBF1\n",
      "SREBF2\n",
      "SRCAP\n",
      "SRF\n",
      "STAT1\n",
      "STAT2\n",
      "STAT3\n",
      "STAT4\n",
      "STAT5A\n",
      "STAT5B\n",
      "STAT6\n",
      "TAL1\n",
      "TBPL2\n",
      "TBX20\n",
      "TBX21\n",
      "TBX22\n",
      "TBX3\n",
      "TBX4\n",
      "TBX5\n",
      "TBXT\n",
      "TCF12\n",
      "TCF20\n",
      "TCF3\n",
      "TCF4\n",
      "TCF7L1\n",
      "TCF7L2\n",
      "TEAD1\n",
      "TEAD2\n",
      "TEAD3\n",
      "TEAD4\n",
      "TEF\n",
      "TET1\n",
      "TET2\n",
      "TFAP2B\n",
      "TFAP2C\n",
      "TFAP2D\n",
      "TFCP2\n",
      "TFDP1\n",
      "TFE3\n",
      "TFEB\n",
      "TFEC\n",
      "THRA\n",
      "THRB\n",
      "TIGD1\n",
      "TIGD4\n",
      "TIGD7\n",
      "TP53\n",
      "TP63\n",
      "TP73\n",
      "TRAFD1\n",
      "TRERF1\n",
      "TWIST1\n",
      "USF1\n",
      "USF2\n",
      "VAX1\n",
      "VAX2\n",
      "VDR\n",
      "XBP1\n",
      "YY1\n",
      "YY2\n",
      "ZBTB17\n",
      "ZBTB48\n",
      "ZFX\n",
      "ZFY\n",
      "CREBZF\n",
      "ZIC2\n",
      "ZNF473\n",
      "ZNF513\n",
      "ZNF561\n",
      "ZNF644\n",
      "ZNF654\n",
      "ZNF777\n",
      "ZNF143\n",
      "ZNF148\n",
      "ZNF16\n",
      "ZNF200\n",
      "ZNF280D\n",
      "ZNF326\n",
      "ZNF426\n",
      "ZNF516\n",
      "ZNF532\n",
      "ZNF592\n",
      "ZNF641\n",
      "ZNF687\n",
      "ZSCAN20\n",
      "ZXDA\n",
      "ZXDB\n",
      "ZXDC\n"
     ]
    }
   ],
   "source": [
    "for gene in cc_TFs[\"Gene\"]:\n",
    "    print(gene)\n",
    "    if gene in aligned_with_lambert:\n",
    "        alignment = AlignIO.read(\"../data/zoonomia_toga_mca/prot_alignments_for_vis/\" + gene + \".fasta\", \"fasta\")\n",
    "        percent_identities = []\n",
    "        summary_align = AlignInfo.SummaryInfo(alignment)\n",
    "        \n",
    "        for record in alignment:\n",
    "            if \"REFERENCE\" in record.id:\n",
    "                reference_seq = record.seq\n",
    "    \n",
    "            if \"LambertSeq\" in record.id:\n",
    "                lambert_TF_seq = record.seq\n",
    "        \n",
    "        percent_identities = []\n",
    "        positions = []\n",
    "        \n",
    "        # Iterate over each position in the alignment where ref_AA is not a gap\n",
    "        for i in range(alignment.get_alignment_length()):\n",
    "            ref_AA = reference_seq[i]\n",
    "            lam_AA = lambert_TF_seq[i]\n",
    "    \n",
    "            \n",
    "            # if ref_AA == \"X\" or ref_AA == \"-\":\n",
    "            #     print(gene)\n",
    "    \n",
    "            if lam_AA != \"-\":\n",
    "                # Get the column (position) for all sequences\n",
    "                column = alignment[:, i]\n",
    "            \n",
    "                # Calculating percent identity\n",
    "                count = column.count(ref_AA)\n",
    "                percent_identity = 100 * count / len(column)\n",
    "            \n",
    "                # AAs.append(ref_AA)\n",
    "                percent_identities.append(percent_identity)\n",
    "                positions.append(lam_AA)\n",
    "            \n",
    "        percent_identities_df = pd.DataFrame({\"percent_identity\" : percent_identities, \"pos\" : positions})\n",
    "        percent_identities_df.to_csv(\"../data/zoonomia_toga_mca/prot_alignment_percent_identities_for_vis/\" + gene)\n",
    "        #display(percent_identities_df)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ece93df7-b220-43c1-8dd0-f2dee9312557",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "03cb8566-4ffe-4657-87cd-949f47efe34a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a6aa1ebc-c429-48ec-a03b-1f2968d7cc4f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ecc17dc-0c48-4c32-b31c-da6326672251",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "551ce66a-6d52-4ebd-a505-acc8ed26218a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0e3be25-d82b-410b-b4a3-6d383fdedefd",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6acb15c6-3207-41f6-8487-c39ff1c93f7c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "517236c7-e7e4-4562-8627-e85070e639fc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f0798e0-459e-4876-b79d-ec235f6698cc",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7820c60-089b-4ae6-80c5-efa4b3aa2508",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1b879655-314f-4df8-8036-9139950ce528",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dac6fce6-c5f8-4d38-80c0-46b7cdfde794",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4dd47ee9-df9f-4e6c-9761-f617ba9dbe2c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "id": "24c8be20-d858-4835-9140-77317132f6b6",
   "metadata": {},
   "source": [
    "## A. Correct isoforms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "7a236ec1-d4a9-43df-a1e5-728d71bfe7b7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb9ea4fb-2c1c-4152-be6a-f466f4c0f61c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
