{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "f56b8101",
   "metadata": {},
   "outputs": [],
   "source": [
    "%load_ext autoreload"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "cb62e820",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib.pyplot import figure\n",
    "\n",
    "%autoreload 2\n",
    "%aimport AD_predictor_tools\n",
    "%aimport AD_comparison_tools\n",
    "%aimport PlottingTools\n",
    "%aimport uniprotBedTools"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "6385607c",
   "metadata": {},
   "source": [
    "1. Get genomic coordinates of all SFARI ADs, save as a bed file &rarr; SFARI_ADs.bed\n",
    "2. Get genomic coordinates of all SFARI TFs, save as a bed file &rarr; SFARI_TFs.bed \n",
    "3. Run bedtools subtract to get bed file of SFARI TF regions that are not ADs &rarr; SFARI_TFs_without_ADs.bed\n",
    "4. Save SFARI variants as bed file  &rarr; SFARI_variants.bed\n",
    "5. Use bedtools intersect to compare 1 and 4  (SFARI_ADs.bed vs. SFARI_variants.bed)\n",
    "6. Use bedtools intersect to compare 3 and 4  (SFARI_TFs_without_ADs.bed vs. SFARI_variants.bed)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2065f252",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>uniprotID</th>\n",
       "      <th>Reference</th>\n",
       "      <th>ProteinSeq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MEIS2</td>\n",
       "      <td>340</td>\n",
       "      <td>477</td>\n",
       "      <td>O14770</td>\n",
       "      <td>activation_regions.txt, GSL</td>\n",
       "      <td>DQSNRAGFLLDPSVSQGAAYSPEGQPMGSFVLDGQQHMGIRPAGLQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>KLF7</td>\n",
       "      <td>1</td>\n",
       "      <td>100</td>\n",
       "      <td>O75840</td>\n",
       "      <td>Stanford</td>\n",
       "      <td>MDVLASYSIFQELQLVHDTGYFSALPSLEETWQQTCLELERYLQTE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CAMTA2</td>\n",
       "      <td>471</td>\n",
       "      <td>580</td>\n",
       "      <td>O94983</td>\n",
       "      <td>Stanford</td>\n",
       "      <td>PSPAPLEPSSRVGRGEALFGGPVGASELEPFSLSSFPDLMGELISD...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CAMTA2</td>\n",
       "      <td>285</td>\n",
       "      <td>468</td>\n",
       "      <td>O94983</td>\n",
       "      <td>PMID: 16678093, Soto / Stanford</td>\n",
       "      <td>KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NKX2-2</td>\n",
       "      <td>220</td>\n",
       "      <td>273</td>\n",
       "      <td>O95096</td>\n",
       "      <td>PMID: 10944215, Soto</td>\n",
       "      <td>AQDLAAATFQAGIPFSAYSAQSLQHMQYNAQYSSASTPQYPTAHPL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>KDM5B</td>\n",
       "      <td>1301</td>\n",
       "      <td>1390</td>\n",
       "      <td>Q9UGL1</td>\n",
       "      <td>Stanford</td>\n",
       "      <td>TNKVSQPPGTTSFSLPDDWDNRTSYLHSPFSTGRSCIPLHGVSPEV...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>TCF20</td>\n",
       "      <td>1</td>\n",
       "      <td>327</td>\n",
       "      <td>Q9UGU0</td>\n",
       "      <td>PMID: 10995766, Soto</td>\n",
       "      <td>MQSFREQSSYHGNQQSYPQEVHGSSRLEEFSPRQAQMFQNFGGTGG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>MYT1L</td>\n",
       "      <td>151</td>\n",
       "      <td>425</td>\n",
       "      <td>Q9UL68</td>\n",
       "      <td>Stanford / Stanford / PMID: 29291346, Soto / S...</td>\n",
       "      <td>DEEEEEEEEEEEEEEEENEDHQMNCHNTRIMQDTEKDDNNNDEYDN...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>TBX22</td>\n",
       "      <td>401</td>\n",
       "      <td>480</td>\n",
       "      <td>Q9Y458</td>\n",
       "      <td>Stanford</td>\n",
       "      <td>QSLAPLMMEVPMLSSLGVTNSKSGSSEDSSDQYLQAPNSTNQMLYG...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>NFE2L3 / NF2L3</td>\n",
       "      <td>251</td>\n",
       "      <td>450</td>\n",
       "      <td>Q9Y4A8</td>\n",
       "      <td>Stanford / PMID: 15388789, Soto / Stanford / S...</td>\n",
       "      <td>RHLNGTDTSFSLEDLFQLLSSQPENSLEGISLGDIPLPGSISDGMN...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>63 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              Gene  Start   End uniprotID  \\\n",
       "0            MEIS2    340   477    O14770   \n",
       "1             KLF7      1   100    O75840   \n",
       "2           CAMTA2    471   580    O94983   \n",
       "3           CAMTA2    285   468    O94983   \n",
       "4           NKX2-2    220   273    O95096   \n",
       "..             ...    ...   ...       ...   \n",
       "58           KDM5B   1301  1390    Q9UGL1   \n",
       "59           TCF20      1   327    Q9UGU0   \n",
       "60           MYT1L    151   425    Q9UL68   \n",
       "61           TBX22    401   480    Q9Y458   \n",
       "62  NFE2L3 / NF2L3    251   450    Q9Y4A8   \n",
       "\n",
       "                                            Reference  \\\n",
       "0                         activation_regions.txt, GSL   \n",
       "1                                            Stanford   \n",
       "2                                            Stanford   \n",
       "3                     PMID: 16678093, Soto / Stanford   \n",
       "4                                PMID: 10944215, Soto   \n",
       "..                                                ...   \n",
       "58                                           Stanford   \n",
       "59                               PMID: 10995766, Soto   \n",
       "60  Stanford / Stanford / PMID: 29291346, Soto / S...   \n",
       "61                                           Stanford   \n",
       "62  Stanford / PMID: 15388789, Soto / Stanford / S...   \n",
       "\n",
       "                                           ProteinSeq  \n",
       "0   DQSNRAGFLLDPSVSQGAAYSPEGQPMGSFVLDGQQHMGIRPAGLQ...  \n",
       "1   MDVLASYSIFQELQLVHDTGYFSALPSLEETWQQTCLELERYLQTE...  \n",
       "2   PSPAPLEPSSRVGRGEALFGGPVGASELEPFSLSSFPDLMGELISD...  \n",
       "3   KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...  \n",
       "4   AQDLAAATFQAGIPFSAYSAQSLQHMQYNAQYSSASTPQYPTAHPL...  \n",
       "..                                                ...  \n",
       "58  TNKVSQPPGTTSFSLPDDWDNRTSYLHSPFSTGRSCIPLHGVSPEV...  \n",
       "59  MQSFREQSSYHGNQQSYPQEVHGSSRLEEFSPRQAQMFQNFGGTGG...  \n",
       "60  DEEEEEEEEEEEEEEEENEDHQMNCHNTRIMQDTEKDDNNNDEYDN...  \n",
       "61  QSLAPLMMEVPMLSSLGVTNSKSGSSEDSSDQYLQAPNSTNQMLYG...  \n",
       "62  RHLNGTDTSFSLEDLFQLLSSQPENSLEGISLGDIPLPGSISDGMN...  \n",
       "\n",
       "[63 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 1. Get genomic coordinates of all SFARI ADs, save as a bed file\n",
    "SFARI_TF_ADs = pd.read_csv(\"../output/SFARI_TF_known_ADs.csv\", index_col = 0)\n",
    "SFARI_TF_ADs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1a2687da",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: O95718 has no corresponding ENST code\n",
      "WARNING: O95718 has no corresponding ENST code\n",
      "WARNING: O95718 has no corresponding ENST code\n",
      "WARNING: O95718 has no corresponding ENST code\n"
     ]
    }
   ],
   "source": [
    "AD_coord_dfs = []\n",
    "for i in SFARI_TF_ADs.index:\n",
    "    uniprotID = SFARI_TF_ADs[\"uniprotID\"].iloc[i]\n",
    "    Start = SFARI_TF_ADs[\"Start\"].iloc[i]\n",
    "    End = SFARI_TF_ADs[\"End\"].iloc[i]\n",
    "    AD_coord_dfs.append(uniprotBedTools.return_domain_bed_df(uniprotID, Start, End, indexing = 0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "1c91f4b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15</td>\n",
       "      <td>36892175</td>\n",
       "      <td>36892459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>15</td>\n",
       "      <td>36895150</td>\n",
       "      <td>36895261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15</td>\n",
       "      <td>36896627</td>\n",
       "      <td>36896646</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>207124206</td>\n",
       "      <td>207124404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>207165466</td>\n",
       "      <td>207165568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163</th>\n",
       "      <td>2</td>\n",
       "      <td>1922493</td>\n",
       "      <td>1923263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>164</th>\n",
       "      <td>2</td>\n",
       "      <td>1942981</td>\n",
       "      <td>1943036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>165</th>\n",
       "      <td>X</td>\n",
       "      <td>80030748</td>\n",
       "      <td>80030988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>166</th>\n",
       "      <td>7</td>\n",
       "      <td>26183700</td>\n",
       "      <td>26183784</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>167</th>\n",
       "      <td>7</td>\n",
       "      <td>26184532</td>\n",
       "      <td>26185048</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>168 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    chr      start        end\n",
       "0    15   36892175   36892459\n",
       "1    15   36895150   36895261\n",
       "2    15   36896627   36896646\n",
       "3     2  207124206  207124404\n",
       "4     2  207165466  207165568\n",
       "..   ..        ...        ...\n",
       "163   2    1922493    1923263\n",
       "164   2    1942981    1943036\n",
       "165   X   80030748   80030988\n",
       "166   7   26183700   26183784\n",
       "167   7   26184532   26185048\n",
       "\n",
       "[168 rows x 3 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "AD_coord_df = pd.concat(AD_coord_dfs)\n",
    "AD_coord_df = AD_coord_df.reset_index(drop = True)\n",
    "AD_coord_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d689a36e",
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprotBedTools.save_bed_from_df(AD_coord_df, \"../data/SFARI_ADs.bed\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32a983f4",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "85c4dff9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 2. Get genomic coordinates of all SFARI TFs, save as a bed file → SFARI_ADs.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e86809f9",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>status</th>\n",
       "      <th>gene-symbol</th>\n",
       "      <th>gene-name</th>\n",
       "      <th>ensembl-id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>genetic-category</th>\n",
       "      <th>gene-score</th>\n",
       "      <th>syndromic</th>\n",
       "      <th>eagle</th>\n",
       "      <th>number-of-reports</th>\n",
       "      <th>GeneName</th>\n",
       "      <th>ProteinSeq</th>\n",
       "      <th>uniprotID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>ADNP</td>\n",
       "      <td>Activity-dependent neuroprotector homeobox</td>\n",
       "      <td>ENSG00000101126</td>\n",
       "      <td>20</td>\n",
       "      <td>Rare Single Gene Mutation, Syndromic, Functional</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>41.50</td>\n",
       "      <td>64</td>\n",
       "      <td>sp|Q9H2P0|ADNP_HUMAN</td>\n",
       "      <td>MFQLPVNNLGSLRKARKTVKKILSDIGLEYCKEHIEDFKQFEPNDF...</td>\n",
       "      <td>Q9H2P0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>AHDC1</td>\n",
       "      <td>AT-hook DNA binding motif containing 1</td>\n",
       "      <td>ENSG00000126705</td>\n",
       "      <td>1</td>\n",
       "      <td>Rare Single Gene Mutation, Syndromic</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>14.25</td>\n",
       "      <td>24</td>\n",
       "      <td>sp|Q5TGY3|AHDC1_HUMAN</td>\n",
       "      <td>MRVKPQGLVVTSSAVCSSPDYLREPKYYPGGPPTPRPLLPTRPPAS...</td>\n",
       "      <td>Q5TGY3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "      <td>ARID2</td>\n",
       "      <td>AT-rich interaction domain 2</td>\n",
       "      <td>ENSG00000189079</td>\n",
       "      <td>12</td>\n",
       "      <td>Rare Single Gene Mutation, Syndromic</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12</td>\n",
       "      <td>sp|Q68CP9|ARID2_HUMAN</td>\n",
       "      <td>MANSTGKAPPDERRKGLAFLDELRQFHHSRGSPFKKIPAVGGKELD...</td>\n",
       "      <td>Q68CP9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>ARNT2</td>\n",
       "      <td>aryl-hydrocarbon receptor nuclear translocator 2</td>\n",
       "      <td>ENSG00000172379</td>\n",
       "      <td>15</td>\n",
       "      <td>Rare Single Gene Mutation, Syndromic, Genetic ...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15</td>\n",
       "      <td>sp|Q9HBZ2|ARNT2_HUMAN</td>\n",
       "      <td>MATPAAVNPPEMASDIPGSVTLPVAPMAATGQVRMAGAMPARGGKR...</td>\n",
       "      <td>Q9HBZ2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>ARX</td>\n",
       "      <td>aristaless related homeobox</td>\n",
       "      <td>ENSG00000004848</td>\n",
       "      <td>X</td>\n",
       "      <td>Rare Single Gene Mutation, Syndromic</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>13.80</td>\n",
       "      <td>24</td>\n",
       "      <td>sp|Q96QS3|ARX_HUMAN</td>\n",
       "      <td>MSNQYQEEGCSERPECKSKSPTLLSSYCIDSILGRRSPCKMRLLGA...</td>\n",
       "      <td>Q96QS3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>122</td>\n",
       "      <td>9</td>\n",
       "      <td>ZNF711</td>\n",
       "      <td>zinc finger protein 711</td>\n",
       "      <td>ENSG00000147180</td>\n",
       "      <td>X</td>\n",
       "      <td>Rare Single Gene Mutation</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>sp|Q9Y462|ZN711_HUMAN</td>\n",
       "      <td>MDSGGGSLGLHTPDSRMAHTMIMQDFVAGMAGTAHIDGDHIVVSVP...</td>\n",
       "      <td>Q9Y462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>123</td>\n",
       "      <td>9</td>\n",
       "      <td>ZNF713</td>\n",
       "      <td>Zinc finger protein 713</td>\n",
       "      <td>ENSG00000178665</td>\n",
       "      <td>7</td>\n",
       "      <td>Rare Single Gene Mutation</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>sp|Q8N859|ZN713_HUMAN</td>\n",
       "      <td>MEEEEMNDGSQMVRSQESLTFQDVAVDFTREEWDQLYPAQKNLYRD...</td>\n",
       "      <td>Q8N859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>124</td>\n",
       "      <td>9</td>\n",
       "      <td>ZNF774</td>\n",
       "      <td>Zinc finger protein 774</td>\n",
       "      <td>ENSG00000196391</td>\n",
       "      <td>15</td>\n",
       "      <td>Rare Single Gene Mutation</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>sp|Q6NX45|ZN774_HUMAN</td>\n",
       "      <td>MWLGTSGKSGLPGHCLENPLQECHPAQLEEWALKGISRPSVISQPE...</td>\n",
       "      <td>Q6NX45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>125</td>\n",
       "      <td>9</td>\n",
       "      <td>ZNF804A</td>\n",
       "      <td>Zinc finger protein 804A</td>\n",
       "      <td>ENSG00000170396</td>\n",
       "      <td>2</td>\n",
       "      <td>Rare Single Gene Mutation, Genetic Association...</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16</td>\n",
       "      <td>sp|Q7Z570|Z804A_HUMAN</td>\n",
       "      <td>MECYYIVISSTHLSNGHFRNIKGVFRGPLSKNGNKTLDYAEKENTI...</td>\n",
       "      <td>Q7Z570</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>126</td>\n",
       "      <td>9</td>\n",
       "      <td>ZNF827</td>\n",
       "      <td>Zinc finger protein 827</td>\n",
       "      <td>ENSG00000151612</td>\n",
       "      <td>4</td>\n",
       "      <td>Rare Single Gene Mutation, Genetic Association</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>sp|Q17R98|ZN827_HUMAN</td>\n",
       "      <td>MPRRKQEQPKRLPSHVSRQEEAEGELSEGEHWYGNSSETPSEASYG...</td>\n",
       "      <td>Q17R98</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>127 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Unnamed: 0  status gene-symbol  \\\n",
       "0             0       9        ADNP   \n",
       "1             1       9       AHDC1   \n",
       "2             2       9       ARID2   \n",
       "3             3       9       ARNT2   \n",
       "4             4       9         ARX   \n",
       "..          ...     ...         ...   \n",
       "122         122       9      ZNF711   \n",
       "123         123       9      ZNF713   \n",
       "124         124       9      ZNF774   \n",
       "125         125       9     ZNF804A   \n",
       "126         126       9      ZNF827   \n",
       "\n",
       "                                            gene-name       ensembl-id  \\\n",
       "0          Activity-dependent neuroprotector homeobox  ENSG00000101126   \n",
       "1              AT-hook DNA binding motif containing 1  ENSG00000126705   \n",
       "2                        AT-rich interaction domain 2  ENSG00000189079   \n",
       "3    aryl-hydrocarbon receptor nuclear translocator 2  ENSG00000172379   \n",
       "4                         aristaless related homeobox  ENSG00000004848   \n",
       "..                                                ...              ...   \n",
       "122                           zinc finger protein 711  ENSG00000147180   \n",
       "123                           Zinc finger protein 713  ENSG00000178665   \n",
       "124                           Zinc finger protein 774  ENSG00000196391   \n",
       "125                          Zinc finger protein 804A  ENSG00000170396   \n",
       "126                           Zinc finger protein 827  ENSG00000151612   \n",
       "\n",
       "    chromosome                                   genetic-category  gene-score  \\\n",
       "0           20   Rare Single Gene Mutation, Syndromic, Functional         1.0   \n",
       "1            1               Rare Single Gene Mutation, Syndromic         1.0   \n",
       "2           12               Rare Single Gene Mutation, Syndromic         2.0   \n",
       "3           15  Rare Single Gene Mutation, Syndromic, Genetic ...         2.0   \n",
       "4            X               Rare Single Gene Mutation, Syndromic         1.0   \n",
       "..         ...                                                ...         ...   \n",
       "122          X                          Rare Single Gene Mutation         2.0   \n",
       "123          7                          Rare Single Gene Mutation         2.0   \n",
       "124         15                          Rare Single Gene Mutation         2.0   \n",
       "125          2  Rare Single Gene Mutation, Genetic Association...         2.0   \n",
       "126          4     Rare Single Gene Mutation, Genetic Association         2.0   \n",
       "\n",
       "     syndromic  eagle  number-of-reports               GeneName  \\\n",
       "0            1  41.50                 64   sp|Q9H2P0|ADNP_HUMAN   \n",
       "1            1  14.25                 24  sp|Q5TGY3|AHDC1_HUMAN   \n",
       "2            1    NaN                 12  sp|Q68CP9|ARID2_HUMAN   \n",
       "3            0    NaN                 15  sp|Q9HBZ2|ARNT2_HUMAN   \n",
       "4            1  13.80                 24    sp|Q96QS3|ARX_HUMAN   \n",
       "..         ...    ...                ...                    ...   \n",
       "122          0    NaN                  5  sp|Q9Y462|ZN711_HUMAN   \n",
       "123          0    NaN                  3  sp|Q8N859|ZN713_HUMAN   \n",
       "124          0    NaN                  4  sp|Q6NX45|ZN774_HUMAN   \n",
       "125          0    NaN                 16  sp|Q7Z570|Z804A_HUMAN   \n",
       "126          0    NaN                  3  sp|Q17R98|ZN827_HUMAN   \n",
       "\n",
       "                                            ProteinSeq uniprotID  \n",
       "0    MFQLPVNNLGSLRKARKTVKKILSDIGLEYCKEHIEDFKQFEPNDF...    Q9H2P0  \n",
       "1    MRVKPQGLVVTSSAVCSSPDYLREPKYYPGGPPTPRPLLPTRPPAS...    Q5TGY3  \n",
       "2    MANSTGKAPPDERRKGLAFLDELRQFHHSRGSPFKKIPAVGGKELD...    Q68CP9  \n",
       "3    MATPAAVNPPEMASDIPGSVTLPVAPMAATGQVRMAGAMPARGGKR...    Q9HBZ2  \n",
       "4    MSNQYQEEGCSERPECKSKSPTLLSSYCIDSILGRRSPCKMRLLGA...    Q96QS3  \n",
       "..                                                 ...       ...  \n",
       "122  MDSGGGSLGLHTPDSRMAHTMIMQDFVAGMAGTAHIDGDHIVVSVP...    Q9Y462  \n",
       "123  MEEEEMNDGSQMVRSQESLTFQDVAVDFTREEWDQLYPAQKNLYRD...    Q8N859  \n",
       "124  MWLGTSGKSGLPGHCLENPLQECHPAQLEEWALKGISRPSVISQPE...    Q6NX45  \n",
       "125  MECYYIVISSTHLSNGHFRNIKGVFRGPLSKNGNKTLDYAEKENTI...    Q7Z570  \n",
       "126  MPRRKQEQPKRLPSHVSRQEEAEGELSEGEHWYGNSSETPSEASYG...    Q17R98  \n",
       "\n",
       "[127 rows x 14 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SFARI_TFs = pd.read_csv(\"../data/SFARI_TFs.csv\")\n",
    "SFARI_TFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "65508ae1",
   "metadata": {},
   "outputs": [],
   "source": [
    "coordinate_dfs = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "27426bcd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING: P56179 has no corresponding ENST code\n",
      "WARNING: O95718 has no corresponding ENST code\n",
      "WARNING: P49639 has no corresponding ENST code\n",
      "WARNING: O43151 has no corresponding ENST code\n",
      "WARNING: Q569K4 has no corresponding ENST code\n"
     ]
    }
   ],
   "source": [
    "for uniprotID in SFARI_TFs[\"uniprotID\"]:\n",
    "    coordinate_dfs.append(uniprotBedTools.return_protein_bed_df(uniprotID, indexing = 0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "de97056c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>20</td>\n",
       "      <td>50903888</td>\n",
       "      <td>50903996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>20</td>\n",
       "      <td>50902016</td>\n",
       "      <td>50902109</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>592</th>\n",
       "      <td>20</td>\n",
       "      <td>50891404</td>\n",
       "      <td>50894512</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>1</td>\n",
       "      <td>27547303</td>\n",
       "      <td>27552115</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>434</th>\n",
       "      <td>12</td>\n",
       "      <td>45729836</td>\n",
       "      <td>45729928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1249</th>\n",
       "      <td>4</td>\n",
       "      <td>145775788</td>\n",
       "      <td>145775960</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1250</th>\n",
       "      <td>4</td>\n",
       "      <td>145774505</td>\n",
       "      <td>145774672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1251</th>\n",
       "      <td>4</td>\n",
       "      <td>145765546</td>\n",
       "      <td>145765738</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1252</th>\n",
       "      <td>4</td>\n",
       "      <td>145764987</td>\n",
       "      <td>145765165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1257</th>\n",
       "      <td>4</td>\n",
       "      <td>145763106</td>\n",
       "      <td>145763122</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1280 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chr      start        end\n",
       "590   20   50903888   50903996\n",
       "589   20   50902016   50902109\n",
       "592   20   50891404   50894512\n",
       "54     1   27547303   27552115\n",
       "434   12   45729836   45729928\n",
       "...   ..        ...        ...\n",
       "1249   4  145775788  145775960\n",
       "1250   4  145774505  145774672\n",
       "1251   4  145765546  145765738\n",
       "1252   4  145764987  145765165\n",
       "1257   4  145763106  145763122\n",
       "\n",
       "[1280 rows x 3 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SFARI_TF_coords = pd.concat(coordinate_dfs)\n",
    "SFARI_TF_coords"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2b70860c",
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprotBedTools.save_bed_from_df(SFARI_TF_coords, \"../data/SFARI_TFs.bed\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "348cb057",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "8adc0347",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 3. Run bedtools subtract to get bed file of SFARI TF regions that are not ADs \n",
    "# → SFARI_TFs_without_ADs.bed\n",
    "\n",
    "! bedtools subtract -a ../data/SFARI_TFs.bed -b ../data/SFARI_ADs.bed > ../output/SFARI_TFs_without_ADs.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "be9c4bc4",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d2b64b11",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>af</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69081</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69130</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>9e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69134</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>4.2e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69135</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>5e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69149</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>9e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12443148</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25044036</td>\n",
       "      <td>CTGG</td>\n",
       "      <td>CTGAG,C</td>\n",
       "      <td>1.9e-05,5e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12443149</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25044040</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>1.9e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12443150</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25044041</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>1.9e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12443151</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25044046</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>1.4e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12443152</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25044050</td>\n",
       "      <td>ATGG</td>\n",
       "      <td>A,ATAG,AAGG</td>\n",
       "      <td>1.4e-05,5e-06,5e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12443153 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         chrom       pos   ref          alt                   af\n",
       "0         chr1     69081     G            C             0.000187\n",
       "1         chr1     69130     C            G                9e-06\n",
       "2         chr1     69134     A            G              4.2e-05\n",
       "3         chr1     69135     A            G                5e-06\n",
       "4         chr1     69149     T            A                9e-06\n",
       "...        ...       ...   ...          ...                  ...\n",
       "12443148  chrY  25044036  CTGG      CTGAG,C        1.9e-05,5e-06\n",
       "12443149  chrY  25044040     T            A              1.9e-05\n",
       "12443150  chrY  25044041    CT            C              1.9e-05\n",
       "12443151  chrY  25044046     G            A              1.4e-05\n",
       "12443152  chrY  25044050  ATGG  A,ATAG,AAGG  1.4e-05,5e-06,5e-06\n",
       "\n",
       "[12443153 rows x 5 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 4. Save SFARI variants as bed file\n",
    "SFARI_variants = pd.read_csv('../data/iWES_v2.gatk.pvcf_variants.tsv', sep = '\\t')\n",
    "SFARI_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "48b43d3b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69080</td>\n",
       "      <td>69081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69129</td>\n",
       "      <td>69130</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69133</td>\n",
       "      <td>69134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69134</td>\n",
       "      <td>69135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69148</td>\n",
       "      <td>69149</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12443148</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25044035</td>\n",
       "      <td>25044036</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12443149</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25044039</td>\n",
       "      <td>25044040</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12443150</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25044040</td>\n",
       "      <td>25044041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12443151</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25044045</td>\n",
       "      <td>25044046</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12443152</th>\n",
       "      <td>chrY</td>\n",
       "      <td>25044049</td>\n",
       "      <td>25044050</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12443153 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           chr     start       end\n",
       "0         chr1     69080     69081\n",
       "1         chr1     69129     69130\n",
       "2         chr1     69133     69134\n",
       "3         chr1     69134     69135\n",
       "4         chr1     69148     69149\n",
       "...        ...       ...       ...\n",
       "12443148  chrY  25044035  25044036\n",
       "12443149  chrY  25044039  25044040\n",
       "12443150  chrY  25044040  25044041\n",
       "12443151  chrY  25044045  25044046\n",
       "12443152  chrY  25044049  25044050\n",
       "\n",
       "[12443153 rows x 3 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# VCF files use one-based indexing\n",
    "# So, to convert to bed, need to adjust\n",
    "# ex.) 1 would become: start at 0, end at 1\n",
    "# ex.) 3 would become: start at 2, end at 3\n",
    "\n",
    "SFARI_variants_bed = SFARI_variants\n",
    "SFARI_variants_bed = SFARI_variants_bed.rename(columns = {\"chrom\" : \"chr\"})\n",
    "SFARI_variants_bed[\"start\"] = SFARI_variants_bed[\"pos\"] - 1\n",
    "SFARI_variants_bed[\"end\"] = SFARI_variants_bed[\"pos\"]\n",
    "SFARI_variants_bed = SFARI_variants_bed[[\"chr\", \"start\", \"end\"]]\n",
    "SFARI_variants_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d865f7bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "uniprotBedTools.save_bed_from_df(SFARI_variants_bed, \"../data/SFARI_variants.bed\", add_chr = False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "45e65535",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "90d51db8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 4. Use bedtools intersect to compare 1 and 4 (SFARI_ADs.bed vs. SFARI_variants.bed)\n",
    "! bedtools intersect -a ../data/SFARI_ADs.bed -b ../data/SFARI_variants.bed > ../output/SFARI_variants_in_ADs.bed\n",
    "# Running on command line on savio instead of in notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "86c21da9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Need to sort because large\n",
    "# please presort your data by chromosome and then by start position"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "dee41dfa",
   "metadata": {},
   "outputs": [],
   "source": [
    "! sortBed -i ../data/SFARI_ADs.bed > ../data/sorted_SFARI_ADs.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "900413c6",
   "metadata": {},
   "outputs": [],
   "source": [
    "! sortBed -i ../data/SFARI_variants.bed > ../data/sorted_SFARI_variants.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "37c598c5",
   "metadata": {},
   "outputs": [],
   "source": [
    "! bedtools intersect -a ../data/sorted_SFARI_ADs.bed -b ../data/sorted_SFARI_variants.bed -sorted > ../output/SFARI_variants_in_ADs.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "26b5a716",
   "metadata": {},
   "source": [
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "40ca94e2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# 5. Use bedtools intersect to compare 3 and 4 \n",
    "# (SFARI_TFs_without_ADs.bed vs. SFARI_variants.bed)\n",
    "# Running on command line on savio instead of in notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "1c1d2bc5",
   "metadata": {},
   "outputs": [],
   "source": [
    "! sortBed -i ../output/SFARI_TFs_without_ADs.bed > ../output/sorted_SFARI_TFs_without_ADs.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "906f01fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "! bedtools intersect -a ../output/sorted_SFARI_TFs_without_ADs.bed -b ../data/sorted_SFARI_variants.bed -sorted > ../output/SFARI_variants_not_in_ADs.bed\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
