{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d8038c1c-0ca8-4fbb-ad77-d562a51c29f3",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "825a859d-3782-4a2a-9fb1-63be3d5f6ff3",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Function</th>\n",
       "      <th>Allele Frequency (AF)_LASI_DAD</th>\n",
       "      <th>AF_gnomAD_SAS</th>\n",
       "      <th>AF_gnomAD_others</th>\n",
       "      <th>HOM_ALT</th>\n",
       "      <th>GENEID_refseq</th>\n",
       "      <th>ClinVar_significance</th>\n",
       "      <th>GENEID_ClinVar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:100042515:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.001495</td>\n",
       "      <td>0.000829</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:100048756:A:C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LoF</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:100048762:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000197</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:100048771:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:100048810:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000560</td>\n",
       "      <td>0.000207</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406299</th>\n",
       "      <td>9:99968589:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>STX17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406300</th>\n",
       "      <td>9:99968599:A:C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>STX17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406301</th>\n",
       "      <td>9:99968635:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>STX17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406302</th>\n",
       "      <td>9:99968649:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>ERP44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406303</th>\n",
       "      <td>9:99985031:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>406304 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     #ID REF ALT  Function  Allele Frequency (AF)_LASI_DAD  \\\n",
       "0       10:100042515:C:T   C   T  missense                        0.001495   \n",
       "1       10:100048756:A:C   A   C       LoF                        0.000187   \n",
       "2       10:100048762:G:A   G   A  missense                        0.000187   \n",
       "3       10:100048771:G:A   G   A  missense                        0.000187   \n",
       "4       10:100048810:C:G   C   G  missense                        0.000560   \n",
       "...                  ...  ..  ..       ...                             ...   \n",
       "406299    9:99968589:A:G   A   G  missense                        0.000187   \n",
       "406300    9:99968599:A:C   A   C  missense                        0.000187   \n",
       "406301    9:99968635:C:G   C   G  missense                        0.000187   \n",
       "406302    9:99968649:C:A   C   A  missense                        0.000187   \n",
       "406303    9:99985031:T:C   T   C  missense                        0.000187   \n",
       "\n",
       "        AF_gnomAD_SAS  AF_gnomAD_others  HOM_ALT GENEID_refseq  \\\n",
       "0            0.000829          0.000000        0          CPN1   \n",
       "1            0.000000          0.000000        0          CPN1   \n",
       "2            0.000000          0.000197        0          CPN1   \n",
       "3            0.000000          0.000068        0          CPN1   \n",
       "4            0.000207          0.000000        0          CPN1   \n",
       "...               ...               ...      ...           ...   \n",
       "406299       0.000000          0.000007        0         STX17   \n",
       "406300       0.000000          0.000000        0         STX17   \n",
       "406301       0.000000          0.000000        0         STX17   \n",
       "406302       0.000000          0.000000        0         ERP44   \n",
       "406303       0.000000          0.000000        0           NaN   \n",
       "\n",
       "       ClinVar_significance GENEID_ClinVar  \n",
       "0                       NaN            NaN  \n",
       "1                       NaN            NaN  \n",
       "2                       NaN            NaN  \n",
       "3                       NaN            NaN  \n",
       "4                       NaN            NaN  \n",
       "...                     ...            ...  \n",
       "406299                  NaN            NaN  \n",
       "406300                  NaN            NaN  \n",
       "406301                  NaN            NaN  \n",
       "406302                  NaN            NaN  \n",
       "406303                  NaN            NaN  \n",
       "\n",
       "[406304 rows x 11 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "south_asian_vars = pd.read_excel(\"../data/kerdoncuff_2024/Extended Data Table S1 Summary of loss of function (LOF) and missense variants in LASI-DAD.xlsx\")\n",
    "south_asian_vars"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d6dd25b7-1be5-4044-be1c-77a8e2b22745",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Function</th>\n",
       "      <th>Allele Frequency (AF)_LASI_DAD</th>\n",
       "      <th>AF_gnomAD_SAS</th>\n",
       "      <th>AF_gnomAD_others</th>\n",
       "      <th>HOM_ALT</th>\n",
       "      <th>GENEID_refseq</th>\n",
       "      <th>ClinVar_significance</th>\n",
       "      <th>GENEID_ClinVar</th>\n",
       "      <th>chr</th>\n",
       "      <th>end</th>\n",
       "      <th>start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:100042515:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.001495</td>\n",
       "      <td>0.000829</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>100042515</td>\n",
       "      <td>100042514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>10:100048756:A:C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>LoF</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>100048756</td>\n",
       "      <td>100048755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:100048762:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000197</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>100048762</td>\n",
       "      <td>100048761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:100048771:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>100048771</td>\n",
       "      <td>100048770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:100048810:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000560</td>\n",
       "      <td>0.000207</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>100048810</td>\n",
       "      <td>100048809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406299</th>\n",
       "      <td>9:99968589:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>STX17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>99968589</td>\n",
       "      <td>99968588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406300</th>\n",
       "      <td>9:99968599:A:C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>STX17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>99968599</td>\n",
       "      <td>99968598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406301</th>\n",
       "      <td>9:99968635:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>STX17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>99968635</td>\n",
       "      <td>99968634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406302</th>\n",
       "      <td>9:99968649:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>ERP44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>99968649</td>\n",
       "      <td>99968648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406303</th>\n",
       "      <td>9:99985031:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>99985031</td>\n",
       "      <td>99985030</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>406304 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     #ID REF ALT  Function  Allele Frequency (AF)_LASI_DAD  \\\n",
       "0       10:100042515:C:T   C   T  missense                        0.001495   \n",
       "1       10:100048756:A:C   A   C       LoF                        0.000187   \n",
       "2       10:100048762:G:A   G   A  missense                        0.000187   \n",
       "3       10:100048771:G:A   G   A  missense                        0.000187   \n",
       "4       10:100048810:C:G   C   G  missense                        0.000560   \n",
       "...                  ...  ..  ..       ...                             ...   \n",
       "406299    9:99968589:A:G   A   G  missense                        0.000187   \n",
       "406300    9:99968599:A:C   A   C  missense                        0.000187   \n",
       "406301    9:99968635:C:G   C   G  missense                        0.000187   \n",
       "406302    9:99968649:C:A   C   A  missense                        0.000187   \n",
       "406303    9:99985031:T:C   T   C  missense                        0.000187   \n",
       "\n",
       "        AF_gnomAD_SAS  AF_gnomAD_others  HOM_ALT GENEID_refseq  \\\n",
       "0            0.000829          0.000000        0          CPN1   \n",
       "1            0.000000          0.000000        0          CPN1   \n",
       "2            0.000000          0.000197        0          CPN1   \n",
       "3            0.000000          0.000068        0          CPN1   \n",
       "4            0.000207          0.000000        0          CPN1   \n",
       "...               ...               ...      ...           ...   \n",
       "406299       0.000000          0.000007        0         STX17   \n",
       "406300       0.000000          0.000000        0         STX17   \n",
       "406301       0.000000          0.000000        0         STX17   \n",
       "406302       0.000000          0.000000        0         ERP44   \n",
       "406303       0.000000          0.000000        0           NaN   \n",
       "\n",
       "       ClinVar_significance GENEID_ClinVar chr        end      start  \n",
       "0                       NaN            NaN  10  100042515  100042514  \n",
       "1                       NaN            NaN  10  100048756  100048755  \n",
       "2                       NaN            NaN  10  100048762  100048761  \n",
       "3                       NaN            NaN  10  100048771  100048770  \n",
       "4                       NaN            NaN  10  100048810  100048809  \n",
       "...                     ...            ...  ..        ...        ...  \n",
       "406299                  NaN            NaN   9   99968589   99968588  \n",
       "406300                  NaN            NaN   9   99968599   99968598  \n",
       "406301                  NaN            NaN   9   99968635   99968634  \n",
       "406302                  NaN            NaN   9   99968649   99968648  \n",
       "406303                  NaN            NaN   9   99985031   99985030  \n",
       "\n",
       "[406304 rows x 14 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "south_asian_vars[\"chr\"] = south_asian_vars[\"#ID\"].str.split(\":\").str[0]\n",
    "south_asian_vars[\"end\"] = south_asian_vars[\"#ID\"].str.split(\":\").str[1].astype(int)\n",
    "south_asian_vars[\"start\"] = south_asian_vars[\"end\"] - 1\n",
    "south_asian_vars"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "d92f98d2-e800-4619-8931-7a160ad10e07",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>Function</th>\n",
       "      <th>Allele Frequency (AF)_LASI_DAD</th>\n",
       "      <th>AF_gnomAD_SAS</th>\n",
       "      <th>AF_gnomAD_others</th>\n",
       "      <th>HOM_ALT</th>\n",
       "      <th>GENEID_refseq</th>\n",
       "      <th>ClinVar_significance</th>\n",
       "      <th>GENEID_ClinVar</th>\n",
       "      <th>chr</th>\n",
       "      <th>end</th>\n",
       "      <th>start</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10:100042515:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.001495</td>\n",
       "      <td>0.000829</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>100042515</td>\n",
       "      <td>100042514</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10:100048762:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000197</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>100048762</td>\n",
       "      <td>100048761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10:100048771:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>100048771</td>\n",
       "      <td>100048770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10:100048810:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000560</td>\n",
       "      <td>0.000207</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>100048810</td>\n",
       "      <td>100048809</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10:100048843:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>CPN1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10</td>\n",
       "      <td>100048843</td>\n",
       "      <td>100048842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406299</th>\n",
       "      <td>9:99968589:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000007</td>\n",
       "      <td>0</td>\n",
       "      <td>STX17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>99968589</td>\n",
       "      <td>99968588</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406300</th>\n",
       "      <td>9:99968599:A:C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>STX17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>99968599</td>\n",
       "      <td>99968598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406301</th>\n",
       "      <td>9:99968635:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>STX17</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>99968635</td>\n",
       "      <td>99968634</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406302</th>\n",
       "      <td>9:99968649:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>ERP44</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>99968649</td>\n",
       "      <td>99968648</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406303</th>\n",
       "      <td>9:99985031:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>missense</td>\n",
       "      <td>0.000187</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>99985031</td>\n",
       "      <td>99985030</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>385985 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     #ID REF ALT  Function  Allele Frequency (AF)_LASI_DAD  \\\n",
       "0       10:100042515:C:T   C   T  missense                        0.001495   \n",
       "2       10:100048762:G:A   G   A  missense                        0.000187   \n",
       "3       10:100048771:G:A   G   A  missense                        0.000187   \n",
       "4       10:100048810:C:G   C   G  missense                        0.000560   \n",
       "5       10:100048843:G:A   G   A  missense                        0.000187   \n",
       "...                  ...  ..  ..       ...                             ...   \n",
       "406299    9:99968589:A:G   A   G  missense                        0.000187   \n",
       "406300    9:99968599:A:C   A   C  missense                        0.000187   \n",
       "406301    9:99968635:C:G   C   G  missense                        0.000187   \n",
       "406302    9:99968649:C:A   C   A  missense                        0.000187   \n",
       "406303    9:99985031:T:C   T   C  missense                        0.000187   \n",
       "\n",
       "        AF_gnomAD_SAS  AF_gnomAD_others  HOM_ALT GENEID_refseq  \\\n",
       "0            0.000829          0.000000        0          CPN1   \n",
       "2            0.000000          0.000197        0          CPN1   \n",
       "3            0.000000          0.000068        0          CPN1   \n",
       "4            0.000207          0.000000        0          CPN1   \n",
       "5            0.000000          0.000000        0          CPN1   \n",
       "...               ...               ...      ...           ...   \n",
       "406299       0.000000          0.000007        0         STX17   \n",
       "406300       0.000000          0.000000        0         STX17   \n",
       "406301       0.000000          0.000000        0         STX17   \n",
       "406302       0.000000          0.000000        0         ERP44   \n",
       "406303       0.000000          0.000000        0           NaN   \n",
       "\n",
       "       ClinVar_significance GENEID_ClinVar chr        end      start  \n",
       "0                       NaN            NaN  10  100042515  100042514  \n",
       "2                       NaN            NaN  10  100048762  100048761  \n",
       "3                       NaN            NaN  10  100048771  100048770  \n",
       "4                       NaN            NaN  10  100048810  100048809  \n",
       "5                       NaN            NaN  10  100048843  100048842  \n",
       "...                     ...            ...  ..        ...        ...  \n",
       "406299                  NaN            NaN   9   99968589   99968588  \n",
       "406300                  NaN            NaN   9   99968599   99968598  \n",
       "406301                  NaN            NaN   9   99968635   99968634  \n",
       "406302                  NaN            NaN   9   99968649   99968648  \n",
       "406303                  NaN            NaN   9   99985031   99985030  \n",
       "\n",
       "[385985 rows x 14 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missense_south_asian_vars = south_asian_vars[south_asian_vars[\"Function\"] == \"missense\"]\n",
    "missense_south_asian_vars"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "b333796f-a5dd-40d1-93df-2618a45f691a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>10</td>\n",
       "      <td>100042514</td>\n",
       "      <td>100042515</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>10</td>\n",
       "      <td>100048761</td>\n",
       "      <td>100048762</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>10</td>\n",
       "      <td>100048770</td>\n",
       "      <td>100048771</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>10</td>\n",
       "      <td>100048809</td>\n",
       "      <td>100048810</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>10</td>\n",
       "      <td>100048842</td>\n",
       "      <td>100048843</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406299</th>\n",
       "      <td>9</td>\n",
       "      <td>99968588</td>\n",
       "      <td>99968589</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406300</th>\n",
       "      <td>9</td>\n",
       "      <td>99968598</td>\n",
       "      <td>99968599</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406301</th>\n",
       "      <td>9</td>\n",
       "      <td>99968634</td>\n",
       "      <td>99968635</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406302</th>\n",
       "      <td>9</td>\n",
       "      <td>99968648</td>\n",
       "      <td>99968649</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>406303</th>\n",
       "      <td>9</td>\n",
       "      <td>99985030</td>\n",
       "      <td>99985031</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>385985 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chr      start        end REF ALT\n",
       "0       10  100042514  100042515   C   T\n",
       "2       10  100048761  100048762   G   A\n",
       "3       10  100048770  100048771   G   A\n",
       "4       10  100048809  100048810   C   G\n",
       "5       10  100048842  100048843   G   A\n",
       "...     ..        ...        ...  ..  ..\n",
       "406299   9   99968588   99968589   A   G\n",
       "406300   9   99968598   99968599   A   C\n",
       "406301   9   99968634   99968635   C   G\n",
       "406302   9   99968648   99968649   C   A\n",
       "406303   9   99985030   99985031   T   C\n",
       "\n",
       "[385985 rows x 5 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missense_south_asian_vars[[\"chr\", \"start\", \"end\", \"REF\", \"ALT\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "cc331e76-f8d1-45a0-9a9f-b778322a27e7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Mutant Sequence</th>\n",
       "      <th>Protein Change</th>\n",
       "      <th>Mutation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>NCOA1_AD3</td>\n",
       "      <td>VTSVTIKSEILPASLQSTTARPTSRLNRLPELELEAIDNQFGQPGT...</td>\n",
       "      <td>p.Ala857Thr</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>NCOA1_AD3</td>\n",
       "      <td>VTSVTIKSEILPASLQSATARPTSRLNRLPELELEAIDNQFGQPGT...</td>\n",
       "      <td>p.Lys959Arg</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>NCOA1_AD3</td>\n",
       "      <td>VTSVTIKSEILPASLQSATARPTSRLNRLPELELEAIDNQFGQPGT...</td>\n",
       "      <td>p.Val968Ile</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>NCOA1_AD3</td>\n",
       "      <td>VTSVTIKSEILPASLQSATARPTSRLNRLPELELEAIDNQFGQPGT...</td>\n",
       "      <td>p.Met984Ile</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>NCOA1_AD3</td>\n",
       "      <td>VTSVTIKSEILPASLQSATARPTSRLNRLPELELEAIDNQFGQPGT...</td>\n",
       "      <td>p.Arg987Ile</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>CAMTA2_AD2</td>\n",
       "      <td>SPAPLEPSSRVGRGEALFGGPVGASELEPFSLSSFPDLMGELISDE...</td>\n",
       "      <td>p.Pro580Ala</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>6</td>\n",
       "      <td>CAMTA2_AD2</td>\n",
       "      <td>SPAPLEPSSRVGRGEALFGGPVGASELEPFSLSSFPDLMGELISDE...</td>\n",
       "      <td>p.Pro524Leu</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>7</td>\n",
       "      <td>NKX2-2_AD</td>\n",
       "      <td>AQDLAAATFQAGIPFSAYSAQSLQHMQYNVQYSSASTPQYPTAHPL...</td>\n",
       "      <td>p.Ala249Val</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>8</td>\n",
       "      <td>NKX2-2_AD</td>\n",
       "      <td>AQDLAAATFQAGIPFSAYSVQSLQHMQYNAQYSSASTPQYPTAHPL...</td>\n",
       "      <td>p.Ala239Val</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>9</td>\n",
       "      <td>NKX2-2_AD</td>\n",
       "      <td>AQDLAAATFQAGIPFSAYRAQSLQHMQYNAQYSSASTPQYPTAHPL...</td>\n",
       "      <td>p.Ser238Arg</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>10</td>\n",
       "      <td>NKX2-2_AD</td>\n",
       "      <td>AQDLAAVTFQAGIPFSAYSAQSLQHMQYNAQYSSASTPQYPTAHPL...</td>\n",
       "      <td>p.Ala226Val</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>11</td>\n",
       "      <td>OTX1_AD</td>\n",
       "      <td>AASSIWSPASISPGSAPASVSVPEPLAAPSNTSCMQRSVAAGAATA...</td>\n",
       "      <td>p.Asp247Asn</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>12</td>\n",
       "      <td>OTX1_AD</td>\n",
       "      <td>AASSIWSPASISPGSAPASVSVPEPLAAPSNTSCMQRSVAAGAATA...</td>\n",
       "      <td>p.Pro254Ser</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>13</td>\n",
       "      <td>OTX1_AD</td>\n",
       "      <td>AASSIWSPASISPGSAPASVSVPEPLAAPSNTSCMQRSVAAGAATA...</td>\n",
       "      <td>p.Gln263His</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>14</td>\n",
       "      <td>OTX1_AD</td>\n",
       "      <td>AASSIWSPASISPGSAPASVSVPEPLAAPSNTSCMQRSVAAGAATA...</td>\n",
       "      <td>p.Leu333Phe</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>15</td>\n",
       "      <td>OTX1_AD</td>\n",
       "      <td>AASSIWSPASISPGSAPASVSVPEPLAAPSNTSCMQRSVAAGAATA...</td>\n",
       "      <td>p.Val353Leu</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>16</td>\n",
       "      <td>CAMTA2_AD1</td>\n",
       "      <td>KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...</td>\n",
       "      <td>p.Gly457Arg</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>17</td>\n",
       "      <td>CAMTA2_AD1</td>\n",
       "      <td>KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...</td>\n",
       "      <td>p.Arg437Lys</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>18</td>\n",
       "      <td>CAMTA2_AD1</td>\n",
       "      <td>KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...</td>\n",
       "      <td>p.Gly427Cys</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>19</td>\n",
       "      <td>CAMTA2_AD1</td>\n",
       "      <td>KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...</td>\n",
       "      <td>p.Arg386Lys</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>20</td>\n",
       "      <td>CAMTA2_AD1</td>\n",
       "      <td>KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...</td>\n",
       "      <td>p.Pro367Leu</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>21</td>\n",
       "      <td>CAMTA2_AD1</td>\n",
       "      <td>KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...</td>\n",
       "      <td>p.Val358Met</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>22</td>\n",
       "      <td>CAMTA2_AD1</td>\n",
       "      <td>KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSQGGSSRGGTAILLLTG...</td>\n",
       "      <td>p.Arg315Gln</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>23</td>\n",
       "      <td>CAMTA2_AD1</td>\n",
       "      <td>KAHTSPSSSSSSSLSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...</td>\n",
       "      <td>p.Ser298Leu</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>24</td>\n",
       "      <td>CAMTA2_AD1</td>\n",
       "      <td>KAHTSPSSSSCSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...</td>\n",
       "      <td>p.Ser295Cys</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>25</td>\n",
       "      <td>PITX1_AD</td>\n",
       "      <td>VPGMPNSGLNNINNLTGSSLNSAMSPGACPYGTPASPYGVYRDTCN...</td>\n",
       "      <td>p.Ser272Gly</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>26</td>\n",
       "      <td>NR4A2_AD1</td>\n",
       "      <td>MPCVQEQYGSSPQGASPASQSYSYHSSGEYSSDFLTPEFVKFSMDL...</td>\n",
       "      <td>p.Ala6Glu</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>27</td>\n",
       "      <td>NCOA1_AD2</td>\n",
       "      <td>GEPNFAPSLSPGSSMVPMPIPPPQSSLLQQTPPASGYQSPDMKAWQ...</td>\n",
       "      <td>p.Ala1243Pro</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>28</td>\n",
       "      <td>NCOA1_AD2</td>\n",
       "      <td>GEANFAPSLSPGSSMVPMPIPPPQSSLLQQTSPASGYQSPDMKAWQ...</td>\n",
       "      <td>p.Pro1272Ser</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>29</td>\n",
       "      <td>NCOA1_AD2</td>\n",
       "      <td>GEANFAPSLSPGSSMVPMPIPPPQSSLLQQTPPASRYQSPDMKAWQ...</td>\n",
       "      <td>p.Gly1276Arg</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>30</td>\n",
       "      <td>NCOA1_AD2</td>\n",
       "      <td>GEANFAPSLSPGSSMVPMPIPPPQSSLLQQTPPASGYQSPDMKAWQ...</td>\n",
       "      <td>p.Asn1332Ser</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>31</td>\n",
       "      <td>NCOA1_AD2</td>\n",
       "      <td>GEANFAPSLSPGSSMVPMPIPPPQSSLLQQTPPASGYQSPDMKAWQ...</td>\n",
       "      <td>p.Val1351Met</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>32</td>\n",
       "      <td>NCOA1_AD2</td>\n",
       "      <td>GEANFAPSLSPGSSMVPMPIPPPQSSLLQQTPPASGYQSPDMKAWQ...</td>\n",
       "      <td>p.Ser1373Pro</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>33</td>\n",
       "      <td>ERG_AD2</td>\n",
       "      <td>MTTNERRVIVPADPTLWSTDHVRQWLEWAVKEYGLPDVNILLFQNI...</td>\n",
       "      <td>p.Asp164Asn</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>34</td>\n",
       "      <td>ERG_AD2</td>\n",
       "      <td>MTTNERRVIVPADPTLWSTDHVRQWLEWAVKEYGVPDVNILLFQNI...</td>\n",
       "      <td>p.Leu152Val</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>35</td>\n",
       "      <td>ERG_AD2</td>\n",
       "      <td>MTTNERRVIVPADPMLWSTDHVRQWLEWAVKEYGLPDVNILLFQNI...</td>\n",
       "      <td>p.Thr132Met</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>36</td>\n",
       "      <td>PAX6_AD</td>\n",
       "      <td>LRNQRRQASNTPSHIPISSSFSTSVYQPIPQPTTPVSSFTSGSMLG...</td>\n",
       "      <td>p.Pro375Gln</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>37</td>\n",
       "      <td>PAX6_AD</td>\n",
       "      <td>LRNQRRQASNTPSHIPISSSFSTSVYQPIPQPTTPVSSFTSGSMLG...</td>\n",
       "      <td>p.Ala321Gly</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>38</td>\n",
       "      <td>ERG_AD1</td>\n",
       "      <td>PHPPALPVTSSSFLAAPNPYWNSPTGGIYPNTRLPTSHMPSHLGTYY</td>\n",
       "      <td>p.Phe446Leu</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>39</td>\n",
       "      <td>ERG_AD1</td>\n",
       "      <td>PHPSALPVTSSSFFAAPNPYWNSPTGGIYPNTRLPTSHMPSHLGTYY</td>\n",
       "      <td>p.Pro436Ser</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>40</td>\n",
       "      <td>PAX5_AD</td>\n",
       "      <td>GRDLASTTLPGYPPHVPPTGQGSYSAPTLTGMVPGSEFSGSPYSHP...</td>\n",
       "      <td>p.Ala322Thr</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>41</td>\n",
       "      <td>PAX5_AD</td>\n",
       "      <td>GRDLASTTLPGYPPHDPPAGQGSYSAPTLTGMVPGSEFSGSPYSHP...</td>\n",
       "      <td>p.Val319Asp</td>\n",
       "      <td>south_asian</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Unnamed: 0   Gene Name                                    Mutant Sequence  \\\n",
       "0            0   NCOA1_AD3  VTSVTIKSEILPASLQSTTARPTSRLNRLPELELEAIDNQFGQPGT...   \n",
       "1            1   NCOA1_AD3  VTSVTIKSEILPASLQSATARPTSRLNRLPELELEAIDNQFGQPGT...   \n",
       "2            2   NCOA1_AD3  VTSVTIKSEILPASLQSATARPTSRLNRLPELELEAIDNQFGQPGT...   \n",
       "3            3   NCOA1_AD3  VTSVTIKSEILPASLQSATARPTSRLNRLPELELEAIDNQFGQPGT...   \n",
       "4            4   NCOA1_AD3  VTSVTIKSEILPASLQSATARPTSRLNRLPELELEAIDNQFGQPGT...   \n",
       "5            5  CAMTA2_AD2  SPAPLEPSSRVGRGEALFGGPVGASELEPFSLSSFPDLMGELISDE...   \n",
       "6            6  CAMTA2_AD2  SPAPLEPSSRVGRGEALFGGPVGASELEPFSLSSFPDLMGELISDE...   \n",
       "7            7   NKX2-2_AD  AQDLAAATFQAGIPFSAYSAQSLQHMQYNVQYSSASTPQYPTAHPL...   \n",
       "8            8   NKX2-2_AD  AQDLAAATFQAGIPFSAYSVQSLQHMQYNAQYSSASTPQYPTAHPL...   \n",
       "9            9   NKX2-2_AD  AQDLAAATFQAGIPFSAYRAQSLQHMQYNAQYSSASTPQYPTAHPL...   \n",
       "10          10   NKX2-2_AD  AQDLAAVTFQAGIPFSAYSAQSLQHMQYNAQYSSASTPQYPTAHPL...   \n",
       "11          11     OTX1_AD  AASSIWSPASISPGSAPASVSVPEPLAAPSNTSCMQRSVAAGAATA...   \n",
       "12          12     OTX1_AD  AASSIWSPASISPGSAPASVSVPEPLAAPSNTSCMQRSVAAGAATA...   \n",
       "13          13     OTX1_AD  AASSIWSPASISPGSAPASVSVPEPLAAPSNTSCMQRSVAAGAATA...   \n",
       "14          14     OTX1_AD  AASSIWSPASISPGSAPASVSVPEPLAAPSNTSCMQRSVAAGAATA...   \n",
       "15          15     OTX1_AD  AASSIWSPASISPGSAPASVSVPEPLAAPSNTSCMQRSVAAGAATA...   \n",
       "16          16  CAMTA2_AD1  KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...   \n",
       "17          17  CAMTA2_AD1  KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...   \n",
       "18          18  CAMTA2_AD1  KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...   \n",
       "19          19  CAMTA2_AD1  KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...   \n",
       "20          20  CAMTA2_AD1  KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...   \n",
       "21          21  CAMTA2_AD1  KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...   \n",
       "22          22  CAMTA2_AD1  KAHTSPSSSSSSSSSGFAEPLEIRPSPPTSQGGSSRGGTAILLLTG...   \n",
       "23          23  CAMTA2_AD1  KAHTSPSSSSSSSLSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...   \n",
       "24          24  CAMTA2_AD1  KAHTSPSSSSCSSSSGFAEPLEIRPSPPTSRGGSSRGGTAILLLTG...   \n",
       "25          25    PITX1_AD  VPGMPNSGLNNINNLTGSSLNSAMSPGACPYGTPASPYGVYRDTCN...   \n",
       "26          26   NR4A2_AD1  MPCVQEQYGSSPQGASPASQSYSYHSSGEYSSDFLTPEFVKFSMDL...   \n",
       "27          27   NCOA1_AD2  GEPNFAPSLSPGSSMVPMPIPPPQSSLLQQTPPASGYQSPDMKAWQ...   \n",
       "28          28   NCOA1_AD2  GEANFAPSLSPGSSMVPMPIPPPQSSLLQQTSPASGYQSPDMKAWQ...   \n",
       "29          29   NCOA1_AD2  GEANFAPSLSPGSSMVPMPIPPPQSSLLQQTPPASRYQSPDMKAWQ...   \n",
       "30          30   NCOA1_AD2  GEANFAPSLSPGSSMVPMPIPPPQSSLLQQTPPASGYQSPDMKAWQ...   \n",
       "31          31   NCOA1_AD2  GEANFAPSLSPGSSMVPMPIPPPQSSLLQQTPPASGYQSPDMKAWQ...   \n",
       "32          32   NCOA1_AD2  GEANFAPSLSPGSSMVPMPIPPPQSSLLQQTPPASGYQSPDMKAWQ...   \n",
       "33          33     ERG_AD2  MTTNERRVIVPADPTLWSTDHVRQWLEWAVKEYGLPDVNILLFQNI...   \n",
       "34          34     ERG_AD2  MTTNERRVIVPADPTLWSTDHVRQWLEWAVKEYGVPDVNILLFQNI...   \n",
       "35          35     ERG_AD2  MTTNERRVIVPADPMLWSTDHVRQWLEWAVKEYGLPDVNILLFQNI...   \n",
       "36          36     PAX6_AD  LRNQRRQASNTPSHIPISSSFSTSVYQPIPQPTTPVSSFTSGSMLG...   \n",
       "37          37     PAX6_AD  LRNQRRQASNTPSHIPISSSFSTSVYQPIPQPTTPVSSFTSGSMLG...   \n",
       "38          38     ERG_AD1    PHPPALPVTSSSFLAAPNPYWNSPTGGIYPNTRLPTSHMPSHLGTYY   \n",
       "39          39     ERG_AD1    PHPSALPVTSSSFFAAPNPYWNSPTGGIYPNTRLPTSHMPSHLGTYY   \n",
       "40          40     PAX5_AD  GRDLASTTLPGYPPHVPPTGQGSYSAPTLTGMVPGSEFSGSPYSHP...   \n",
       "41          41     PAX5_AD  GRDLASTTLPGYPPHDPPAGQGSYSAPTLTGMVPGSEFSGSPYSHP...   \n",
       "\n",
       "   Protein Change     Mutation  \n",
       "0     p.Ala857Thr  south_asian  \n",
       "1     p.Lys959Arg  south_asian  \n",
       "2     p.Val968Ile  south_asian  \n",
       "3     p.Met984Ile  south_asian  \n",
       "4     p.Arg987Ile  south_asian  \n",
       "5     p.Pro580Ala  south_asian  \n",
       "6     p.Pro524Leu  south_asian  \n",
       "7     p.Ala249Val  south_asian  \n",
       "8     p.Ala239Val  south_asian  \n",
       "9     p.Ser238Arg  south_asian  \n",
       "10    p.Ala226Val  south_asian  \n",
       "11    p.Asp247Asn  south_asian  \n",
       "12    p.Pro254Ser  south_asian  \n",
       "13    p.Gln263His  south_asian  \n",
       "14    p.Leu333Phe  south_asian  \n",
       "15    p.Val353Leu  south_asian  \n",
       "16    p.Gly457Arg  south_asian  \n",
       "17    p.Arg437Lys  south_asian  \n",
       "18    p.Gly427Cys  south_asian  \n",
       "19    p.Arg386Lys  south_asian  \n",
       "20    p.Pro367Leu  south_asian  \n",
       "21    p.Val358Met  south_asian  \n",
       "22    p.Arg315Gln  south_asian  \n",
       "23    p.Ser298Leu  south_asian  \n",
       "24    p.Ser295Cys  south_asian  \n",
       "25    p.Ser272Gly  south_asian  \n",
       "26      p.Ala6Glu  south_asian  \n",
       "27   p.Ala1243Pro  south_asian  \n",
       "28   p.Pro1272Ser  south_asian  \n",
       "29   p.Gly1276Arg  south_asian  \n",
       "30   p.Asn1332Ser  south_asian  \n",
       "31   p.Val1351Met  south_asian  \n",
       "32   p.Ser1373Pro  south_asian  \n",
       "33    p.Asp164Asn  south_asian  \n",
       "34    p.Leu152Val  south_asian  \n",
       "35    p.Thr132Met  south_asian  \n",
       "36    p.Pro375Gln  south_asian  \n",
       "37    p.Ala321Gly  south_asian  \n",
       "38    p.Phe446Leu  south_asian  \n",
       "39    p.Pro436Ser  south_asian  \n",
       "40    p.Ala322Thr  south_asian  \n",
       "41    p.Val319Asp  south_asian  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output = pd.read_csv(\"../output/caitlin_experiment/south_asian_vars.csv\")\n",
    "output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "d885ffda-c4c3-4023-9cbd-0212bf91e373",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Gene Name</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CAMTA2_AD1</th>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCOA1_AD2</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NCOA1_AD3</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>OTX1_AD</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NKX2-2_AD</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERG_AD2</th>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CAMTA2_AD2</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAX6_AD</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ERG_AD1</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PAX5_AD</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PITX1_AD</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NR4A2_AD1</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            count\n",
       "Gene Name        \n",
       "CAMTA2_AD1      9\n",
       "NCOA1_AD2       6\n",
       "NCOA1_AD3       5\n",
       "OTX1_AD         5\n",
       "NKX2-2_AD       4\n",
       "ERG_AD2         3\n",
       "CAMTA2_AD2      2\n",
       "PAX6_AD         2\n",
       "ERG_AD1         2\n",
       "PAX5_AD         2\n",
       "PITX1_AD        1\n",
       "NR4A2_AD1       1"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(output[\"Gene Name\"].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b2360929-e3f8-48a9-9191-dd2cc3b01b63",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
