{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "1cd69781",
   "metadata": {},
   "outputs": [],
   "source": [
    "from IPython.display import HTML\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "c0836b55",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Files in the directory:\n",
      "ENST00000359486\n",
      "ENST00000379044\n",
      "ENST00000437473\n",
      "ENST00000339562\n",
      "ENST00000348332\n",
      "ENST00000367265\n",
      "ENST00000342988\n",
      "ENST00000372583\n",
      "ENST00000262238\n",
      "ENST00000303329\n",
      "ENST00000315869\n",
      "ENST00000241001\n",
      "ENST00000348066\n",
      "ENST00000428368\n",
      "ENST00000318003\n",
      "ENST00000341099\n",
      "ENST00000056233\n",
      "ENST00000331340\n",
      "ENST00000356073\n",
      "ENST00000265340\n",
      "ENST00000377142\n",
      "ENST00000262518\n",
      "ENST00000239243\n",
      "ENST00000398919\n",
      "ENST00000373294\n",
      "ENST00000309446\n",
      "ENST00000561208\n",
      "ENST00000355995\n",
      "ENST00000395324\n",
      "ENST00000335670\n",
      "ENST00000312233\n",
      "ENST00000373036\n",
      "ENST00000434704\n",
      "ENST00000380013\n",
      "ENST00000355311\n",
      "ENST00000377022\n",
      "ENST00000250916\n",
      "ENST00000389506\n",
      "ENST00000246672\n",
      "ENST00000317216\n",
      "ENST00000403491\n",
      "ENST00000592199\n",
      "ENST00000358127\n",
      "ENST00000282549\n",
      "ENST00000264637\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Directory path\n",
    "directory = '../outputs/mutations/cds_bed_format/'\n",
    "\n",
    "# Get all files in the directory\n",
    "files = [file for file in os.listdir(directory) if os.path.isfile(os.path.join(directory, file))]\n",
    "\n",
    "# Print the list of files\n",
    "print(\"Files in the directory:\")\n",
    "for file in files:\n",
    "    print(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "27582b14",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<a href='https://gnomad.broadinstitute.org/transcript/ENST00000359486?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000359486?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000379044?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000379044?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000437473?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000437473?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000339562?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000339562?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000348332?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000348332?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000367265?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000367265?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000342988?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000342988?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000372583?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000372583?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000262238?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000262238?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000303329?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000303329?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000315869?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000315869?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000241001?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000241001?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000348066?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000348066?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000428368?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000428368?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000318003?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000318003?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000341099?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000341099?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000056233?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000056233?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000331340?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000331340?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000356073?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000356073?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000265340?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000265340?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000377142?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000377142?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000262518?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000262518?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000239243?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000239243?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000398919?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000398919?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000373294?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000373294?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000309446?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000309446?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000561208?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000561208?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000355995?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000355995?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000395324?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000395324?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000335670?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000335670?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000312233?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000312233?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000373036?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000373036?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000434704?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000434704?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000380013?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000380013?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000355311?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000355311?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000377022?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000377022?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000250916?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000250916?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000389506?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000389506?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000246672?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000246672?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000317216?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000317216?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000403491?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000403491?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000592199?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000592199?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000358127?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000358127?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000282549?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000282549?dataset=gnomad_r4</a><br><a href='https://gnomad.broadinstitute.org/transcript/ENST00000264637?dataset=gnomad_r4'>https://gnomad.broadinstitute.org/transcript/ENST00000264637?dataset=gnomad_r4</a><br>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# List of links\n",
    "links = [\"https://gnomad.broadinstitute.org/transcript/\" + ENST + \"?dataset=gnomad_r4\" for ENST in files]\n",
    "\n",
    "# Generate HTML code for hyperlinks\n",
    "html_code = \"\"\n",
    "for link in links:\n",
    "    html_code += f\"<a href='{link}'>{link}</a><br>\"\n",
    "\n",
    "# Display the HTML code\n",
    "display(HTML(html_code))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2c9f8149",
   "metadata": {},
   "source": [
    "Clicking on link, downloading CSV of variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "9cbd03fa",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Files in the directory:\n",
      "gnomAD_v4.0.0_ENST00000309446_2024_02_26_00_26_03.csv\n",
      "gnomAD_v4.0.0_ENST00000335670_2024_02_26_00_26_31.csv\n",
      "gnomAD_v4.0.0_ENST00000264637_2024_02_26_00_27_51.csv\n",
      "gnomAD_v4.0.0_ENST00000428368_2024_02_26_00_23_33.csv\n",
      "gnomAD_v4.0.0_ENST00000356073_2024_02_26_00_24_07.csv\n",
      "gnomAD_v4.0.0_ENST00000395324_2024_02_26_00_26_25.csv\n",
      "gnomAD_v4.0.0_ENST00000379044_2024_02_26_00_19_01.csv\n",
      "gnomAD_v4.0.0_ENST00000339562_2024_02_26_00_19_11.csv\n",
      "gnomAD_v4.0.0_ENST00000398919_2024_02_26_00_24_50.csv\n",
      "gnomAD_v4.0.0_ENST00000389506_2024_02_26_00_27_22.csv\n",
      ".DS_Store\n",
      "gnomAD_v4.0.0_ENST00000317216_2024_02_26_00_20_49.csv\n",
      "gnomAD_v4.0.0_ENST00000373294_2024_02_26_00_25_02.csv\n",
      "gnomAD_v4.0.0_ENST00000348066_2024_02_26_00_23_50.csv\n",
      "gnomAD_v4.0.0_ENST00000056233_2024_02_26_00_24_00.csv\n",
      "gnomAD_v4.0.0_ENST00000315869_2024_02_26_00_19_45.csv\n",
      "gnomAD_v4.0.0_ENST00000341099_2024_02_26_00_20_07.csv\n",
      "gnomAD_v4.0.0_ENST00000331340_2024_02_26_00_24_20.csv\n",
      "gnomAD_v4.0.0_ENST00000437473_2024_02_26_00_19_06.csv\n",
      "gnomAD_v4.0.0_ENST00000312233_2024_02_26_00_26_35.csv\n",
      "gnomAD_v4.0.0_ENST00000318003_2024_02_26_00_23_41.csv\n",
      "gnomAD_v4.0.0_ENST00000241001_2024_02_26_00_19_49.csv\n",
      "gnomAD_v4.0.0_ENST00000367265_2024_02_26_00_19_21.csv\n",
      "gnomAD_v4.0.0_ENST00000358127_2024_02_26_00_27_42.csv\n",
      "gnomAD_v4.0.0_ENST00000373036_2024_02_26_00_26_51.csv\n",
      "gnomAD_v4.0.0_ENST00000561208_2024_02_26_00_21_20.csv\n",
      "gnomAD_v4.0.0_ENST00000355995_2024_02_26_00_26_15.csv\n",
      "gnomAD_v4.0.0_ENST00000348332_2024_02_26_00_19_16.csv\n",
      "gnomAD_v4.0.0_ENST00000250916_2024_02_26_00_27_17.csv\n",
      "gnomAD_v4.0.0_ENST00000359486_2024_02_26_00_18_30.csv\n",
      "gnomAD_v4.0.0_ENST00000282549_2024_02_26_00_27_46.csv\n",
      "gnomAD_v4.0.0_ENST00000262518_2024_02_26_00_24_40.csv\n",
      "gnomAD_v4.0.0_ENST00000403491_2024_02_26_00_27_37.csv\n",
      "gnomAD_v4.0.0_ENST00000342988_2024_02_26_00_19_26.csv\n",
      "gnomAD_v4.0.0_ENST00000380013_2024_02_26_00_27_07.csv\n",
      "gnomAD_v4.0.0_ENST00000377022_2024_02_26_00_27_12.csv\n",
      "gnomAD_v4.0.0_ENST00000246672_2024_02_26_00_27_26.csv\n",
      "gnomAD_v4.0.0_ENST00000239243_2024_02_26_00_24_45.csv\n",
      "gnomAD_v4.0.0_ENST00000262238_2024_02_26_00_19_36.csv\n",
      "gnomAD_v4.0.0_ENST00000377142_2024_02_26_00_24_35.csv\n",
      "gnomAD_v4.0.0_ENST00000355311_2024_02_26_00_21_03.csv\n",
      "gnomAD_v4.0.0_ENST00000265340_2024_02_26_00_24_13.csv\n",
      "gnomAD_v4.0.0_ENST00000303329_2024_02_26_00_21_42.csv\n",
      "gnomAD_v4.0.0_ENST00000434704_2024_02_26_00_27_01.csv\n",
      "gnomAD_v4.0.0_ENST00000372583_2024_02_26_00_19_31.csv\n",
      "gnomAD_v4.0.0_ENST00000592199_2024_02_26_00_20_39.csv\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "\n",
    "# Directory path\n",
    "directory = '../raw_files/gnomAD/'\n",
    "\n",
    "# Get all files in the directory\n",
    "gnomAD_files = [file for file in os.listdir(directory) if os.path.isfile(os.path.join(directory, file))]\n",
    "\n",
    "# Print the list of files\n",
    "print(\"Files in the directory:\")\n",
    "for file in gnomAD_files:\n",
    "    print(file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "2f429ded",
   "metadata": {},
   "outputs": [],
   "source": [
    "dfs = []\n",
    "\n",
    "for file in gnomAD_files:\n",
    "    if file[0] == \"g\":\n",
    "        output_df = pd.read_csv(\"../raw_files/gnomAD/\" + str(file), low_memory = False)\n",
    "        output_df[\"start\"] = output_df[\"Position\"] - 1\n",
    "        output_df = output_df[[\"Chromosome\", \"start\", \"Position\", \"Reference\", \"Alternate\", \"Allele Frequency\"]]\n",
    "        dfs.append(output_df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "id": "b5e97c11",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>Position</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternate</th>\n",
       "      <th>Allele Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>207081140</td>\n",
       "      <td>207081141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>3.216086e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>207081143</td>\n",
       "      <td>207081144</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>6.373689e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>207081144</td>\n",
       "      <td>207081145</td>\n",
       "      <td>G</td>\n",
       "      <td>GC</td>\n",
       "      <td>7.777598e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>207081144</td>\n",
       "      <td>207081145</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "      <td>2.333279e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>207081144</td>\n",
       "      <td>207081145</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>7.777610e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775</th>\n",
       "      <td>19</td>\n",
       "      <td>13094705</td>\n",
       "      <td>13094706</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>7.334088e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776</th>\n",
       "      <td>19</td>\n",
       "      <td>13094708</td>\n",
       "      <td>13094709</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1.997268e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1777</th>\n",
       "      <td>19</td>\n",
       "      <td>13094710</td>\n",
       "      <td>13094711</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.250769e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1778</th>\n",
       "      <td>19</td>\n",
       "      <td>13094717</td>\n",
       "      <td>13094718</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3.763805e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1779</th>\n",
       "      <td>19</td>\n",
       "      <td>13094719</td>\n",
       "      <td>13094720</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>6.577476e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>117303 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chromosome      start   Position Reference Alternate  Allele Frequency\n",
       "0             2  207081140  207081141         T         C      3.216086e-06\n",
       "1             2  207081143  207081144        AG         A      6.373689e-06\n",
       "2             2  207081144  207081145         G        GC      7.777598e-07\n",
       "3             2  207081144  207081145        GC         G      2.333279e-06\n",
       "4             2  207081144  207081145         G         C      7.777610e-07\n",
       "...         ...        ...        ...       ...       ...               ...\n",
       "1775         19   13094705   13094706         G         A      7.334088e-07\n",
       "1776         19   13094708   13094709         G         T      1.997268e-06\n",
       "1777         19   13094710   13094711         A         G      1.250769e-06\n",
       "1778         19   13094717   13094718         A         G      3.763805e-05\n",
       "1779         19   13094719   13094720         A         G      6.577476e-06\n",
       "\n",
       "[117303 rows x 6 columns]"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "all_gnomad_variants = pd.concat(dfs)\n",
    "all_gnomad_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "id": "7ef7ffe9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# all_gnomad_variants[(all_gnomad_variants[\"Filters - exomes\"] == \"PASS\") | (all_gnomad_variants[\"Filters - genomes\"] == \"PASS\")]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "1918ef2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# all_gnomad_variants.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "id": "481a8dc7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# all_gnomad_variants[[\"Allele\"]].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "id": "b0cb3c73",
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_len = all_gnomad_variants[\"Reference\"].str.len() == 1\n",
    "alt_len = all_gnomad_variants[\"Alternate\"].str.len() == 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "2c7e22fa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>Position</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Alternate</th>\n",
       "      <th>Allele Frequency</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2</td>\n",
       "      <td>207081140</td>\n",
       "      <td>207081141</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>3.216086e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2</td>\n",
       "      <td>207081144</td>\n",
       "      <td>207081145</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>7.777610e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "      <td>207081145</td>\n",
       "      <td>207081146</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>7.714145e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2</td>\n",
       "      <td>207081146</td>\n",
       "      <td>207081147</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>1.529487e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2</td>\n",
       "      <td>207081147</td>\n",
       "      <td>207081148</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>5.777024e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1775</th>\n",
       "      <td>19</td>\n",
       "      <td>13094705</td>\n",
       "      <td>13094706</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>7.334088e-07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1776</th>\n",
       "      <td>19</td>\n",
       "      <td>13094708</td>\n",
       "      <td>13094709</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>1.997268e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1777</th>\n",
       "      <td>19</td>\n",
       "      <td>13094710</td>\n",
       "      <td>13094711</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>1.250769e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1778</th>\n",
       "      <td>19</td>\n",
       "      <td>13094717</td>\n",
       "      <td>13094718</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>3.763805e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1779</th>\n",
       "      <td>19</td>\n",
       "      <td>13094719</td>\n",
       "      <td>13094720</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>6.577476e-06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>107572 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chromosome      start   Position Reference Alternate  Allele Frequency\n",
       "0             2  207081140  207081141         T         C      3.216086e-06\n",
       "4             2  207081144  207081145         G         C      7.777610e-07\n",
       "5             2  207081145  207081146         C         G      7.714145e-07\n",
       "6             2  207081146  207081147         C         T      1.529487e-06\n",
       "7             2  207081147  207081148         C         T      5.777024e-06\n",
       "...         ...        ...        ...       ...       ...               ...\n",
       "1775         19   13094705   13094706         G         A      7.334088e-07\n",
       "1776         19   13094708   13094709         G         T      1.997268e-06\n",
       "1777         19   13094710   13094711         A         G      1.250769e-06\n",
       "1778         19   13094717   13094718         A         G      3.763805e-05\n",
       "1779         19   13094719   13094720         A         G      6.577476e-06\n",
       "\n",
       "[107572 rows x 6 columns]"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gnomad_snps = all_gnomad_variants[ref_len & alt_len]\n",
    "gnomad_snps"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "55ea4cc4",
   "metadata": {},
   "outputs": [],
   "source": [
    "gnomad_snps.to_csv(\"../raw_files/gnomad.bed\", index=False, sep='\\t', header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e52ace6",
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "sns.histplot(gnomad_snps[\"Allele Frequency\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8e42b5b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
