{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "2fd034ab-f3f0-4514-9088-576654c8c9c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import glob\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e67fcf29-6cb4-47dc-ada6-b037014593e4",
   "metadata": {},
   "source": [
    "## Gathering variants\n",
    "Gathering information on other variant types:\n",
    "- Here: **AlphaMissense 10 most + least pathogenic variants per AD**\n",
    "- All gnomAD variants in AD\n",
    "- All Clinvar variants in AD"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8f3064ba-fdbd-40d6-a4e5-7b9a9728aa60",
   "metadata": {},
   "source": [
    "### Loading in the AD names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "dbb44f7f-a4e7-4d47-a1f3-e3eb26743612",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['../output/caitlin_experiment/variant fastas/ERG_P11308_AD_118-261_seq_adj',\n",
       " '../output/caitlin_experiment/variant fastas/NCOA1_Q15788_AD_840-1011',\n",
       " '../output/caitlin_experiment/variant fastas/CAMTA2_O94983_AD_472-581',\n",
       " '../output/caitlin_experiment/variant fastas/OTX1_P32242_AD_172-354',\n",
       " '../output/caitlin_experiment/variant fastas/NCOA1_Q15788_AD_1241-1385',\n",
       " '../output/caitlin_experiment/variant fastas/CAMTA2_O94983_AD_285-468',\n",
       " '../output/caitlin_experiment/variant fastas/IKZF1_Q13422_AD_284-365',\n",
       " '../output/caitlin_experiment/variant fastas/NKX2-2_O95096_AD_220-273',\n",
       " '../output/caitlin_experiment/variant fastas/PAX6_P26367_AD_271-422',\n",
       " '../output/caitlin_experiment/variant fastas/NR4A2_P43354_AD_584-598',\n",
       " '../output/caitlin_experiment/variant fastas/MEIS2_O14770_AD_340-477',\n",
       " '../output/caitlin_experiment/variant fastas/PITX1_P78337_AD_234-283',\n",
       " '../output/caitlin_experiment/variant fastas/ERG_P11308_AD_433-479_seq_adj',\n",
       " '../output/caitlin_experiment/variant fastas/NR4A2_P43354_AD_1-91',\n",
       " '../output/caitlin_experiment/variant fastas/PAX5_Q02548_AD_304-358']"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variant_fasta_filepaths = glob.glob('../output/caitlin_experiment/variant fastas/*')\n",
    "variant_fasta_filepaths"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "f3f66ffc-6bb5-4363-bfc3-22354dd4d29d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filepath</th>\n",
       "      <th>gene</th>\n",
       "      <th>uniprotID</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/ER...</td>\n",
       "      <td>ERG</td>\n",
       "      <td>P11308</td>\n",
       "      <td>118</td>\n",
       "      <td>261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/NC...</td>\n",
       "      <td>NCOA1</td>\n",
       "      <td>Q15788</td>\n",
       "      <td>840</td>\n",
       "      <td>1011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/CA...</td>\n",
       "      <td>CAMTA2</td>\n",
       "      <td>O94983</td>\n",
       "      <td>472</td>\n",
       "      <td>581</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/OT...</td>\n",
       "      <td>OTX1</td>\n",
       "      <td>P32242</td>\n",
       "      <td>172</td>\n",
       "      <td>354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/NC...</td>\n",
       "      <td>NCOA1</td>\n",
       "      <td>Q15788</td>\n",
       "      <td>1241</td>\n",
       "      <td>1385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/CA...</td>\n",
       "      <td>CAMTA2</td>\n",
       "      <td>O94983</td>\n",
       "      <td>285</td>\n",
       "      <td>468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/IK...</td>\n",
       "      <td>IKZF1</td>\n",
       "      <td>Q13422</td>\n",
       "      <td>284</td>\n",
       "      <td>365</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/NK...</td>\n",
       "      <td>NKX2-2</td>\n",
       "      <td>O95096</td>\n",
       "      <td>220</td>\n",
       "      <td>273</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/PA...</td>\n",
       "      <td>PAX6</td>\n",
       "      <td>P26367</td>\n",
       "      <td>271</td>\n",
       "      <td>422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/NR...</td>\n",
       "      <td>NR4A2</td>\n",
       "      <td>P43354</td>\n",
       "      <td>584</td>\n",
       "      <td>598</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/ME...</td>\n",
       "      <td>MEIS2</td>\n",
       "      <td>O14770</td>\n",
       "      <td>340</td>\n",
       "      <td>477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/PI...</td>\n",
       "      <td>PITX1</td>\n",
       "      <td>P78337</td>\n",
       "      <td>234</td>\n",
       "      <td>283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/ER...</td>\n",
       "      <td>ERG</td>\n",
       "      <td>P11308</td>\n",
       "      <td>433</td>\n",
       "      <td>479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/NR...</td>\n",
       "      <td>NR4A2</td>\n",
       "      <td>P43354</td>\n",
       "      <td>1</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/PA...</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>Q02548</td>\n",
       "      <td>304</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             filepath    gene uniprotID  \\\n",
       "0   ../output/caitlin_experiment/variant fastas/ER...     ERG    P11308   \n",
       "1   ../output/caitlin_experiment/variant fastas/NC...   NCOA1    Q15788   \n",
       "2   ../output/caitlin_experiment/variant fastas/CA...  CAMTA2    O94983   \n",
       "3   ../output/caitlin_experiment/variant fastas/OT...    OTX1    P32242   \n",
       "4   ../output/caitlin_experiment/variant fastas/NC...   NCOA1    Q15788   \n",
       "5   ../output/caitlin_experiment/variant fastas/CA...  CAMTA2    O94983   \n",
       "6   ../output/caitlin_experiment/variant fastas/IK...   IKZF1    Q13422   \n",
       "7   ../output/caitlin_experiment/variant fastas/NK...  NKX2-2    O95096   \n",
       "8   ../output/caitlin_experiment/variant fastas/PA...    PAX6    P26367   \n",
       "9   ../output/caitlin_experiment/variant fastas/NR...   NR4A2    P43354   \n",
       "10  ../output/caitlin_experiment/variant fastas/ME...   MEIS2    O14770   \n",
       "11  ../output/caitlin_experiment/variant fastas/PI...   PITX1    P78337   \n",
       "12  ../output/caitlin_experiment/variant fastas/ER...     ERG    P11308   \n",
       "13  ../output/caitlin_experiment/variant fastas/NR...   NR4A2    P43354   \n",
       "14  ../output/caitlin_experiment/variant fastas/PA...    PAX5    Q02548   \n",
       "\n",
       "    start   end  \n",
       "0     118   261  \n",
       "1     840  1011  \n",
       "2     472   581  \n",
       "3     172   354  \n",
       "4    1241  1385  \n",
       "5     285   468  \n",
       "6     284   365  \n",
       "7     220   273  \n",
       "8     271   422  \n",
       "9     584   598  \n",
       "10    340   477  \n",
       "11    234   283  \n",
       "12    433   479  \n",
       "13      1    91  \n",
       "14    304   358  "
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ADs = pd.DataFrame({\"filepath\" : variant_fasta_filepaths})\n",
    "ADs[\"gene\"] = ADs[\"filepath\"].str.split(\"fastas/\").str[1].str.split(\"_\").str[0]\n",
    "ADs[\"uniprotID\"] = ADs[\"filepath\"].str.split(\"_\").str[2]\n",
    "ADs[\"start\"] = ADs[\"filepath\"].str.split(\"_AD_\").str[1].str.split(\"-\").str[0].astype(int)\n",
    "ADs[\"end\"] = ADs[\"filepath\"].str.split(\"_AD_\").str[1].str.split(\"-\").str[1].str.split(\"_\").str[0].astype(int)\n",
    "ADs"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b0eee897-0c0b-446a-b372-f3509668cbe2",
   "metadata": {},
   "source": [
    "### AlphaMissense"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "0352a1c4-2a54-4eb0-8827-1a2bbbf21716",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>genome</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>protein_variant</th>\n",
       "      <th>am_pathogenicity</th>\n",
       "      <th>am_class</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69094</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>V2L</td>\n",
       "      <td>0.2937</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69094</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>V2L</td>\n",
       "      <td>0.2937</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69094</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>V2M</td>\n",
       "      <td>0.3296</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69095</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>V2A</td>\n",
       "      <td>0.2609</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69095</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>V2E</td>\n",
       "      <td>0.2922</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4995</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942896</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q96NU1</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>D468N</td>\n",
       "      <td>0.2016</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4996</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942896</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q96NU1</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>D468H</td>\n",
       "      <td>0.4723</td>\n",
       "      <td>ambiguous</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4997</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942896</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q96NU1</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>D468Y</td>\n",
       "      <td>0.2192</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4998</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942897</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q96NU1</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>D468A</td>\n",
       "      <td>0.2714</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4999</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942897</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q96NU1</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>D468G</td>\n",
       "      <td>0.2125</td>\n",
       "      <td>likely_benign</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5000 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #CHROM     POS REF ALT genome uniprot_id      transcript_id  \\\n",
       "0      chr1   69094   G   T   hg38     Q8NH21  ENST00000335137.4   \n",
       "1      chr1   69094   G   C   hg38     Q8NH21  ENST00000335137.4   \n",
       "2      chr1   69094   G   A   hg38     Q8NH21  ENST00000335137.4   \n",
       "3      chr1   69095   T   C   hg38     Q8NH21  ENST00000335137.4   \n",
       "4      chr1   69095   T   A   hg38     Q8NH21  ENST00000335137.4   \n",
       "...     ...     ...  ..  ..    ...        ...                ...   \n",
       "4995   chr1  942896   G   A   hg38     Q96NU1  ENST00000342066.8   \n",
       "4996   chr1  942896   G   C   hg38     Q96NU1  ENST00000342066.8   \n",
       "4997   chr1  942896   G   T   hg38     Q96NU1  ENST00000342066.8   \n",
       "4998   chr1  942897   A   C   hg38     Q96NU1  ENST00000342066.8   \n",
       "4999   chr1  942897   A   G   hg38     Q96NU1  ENST00000342066.8   \n",
       "\n",
       "     protein_variant  am_pathogenicity       am_class  \n",
       "0                V2L            0.2937  likely_benign  \n",
       "1                V2L            0.2937  likely_benign  \n",
       "2                V2M            0.3296  likely_benign  \n",
       "3                V2A            0.2609  likely_benign  \n",
       "4                V2E            0.2922  likely_benign  \n",
       "...              ...               ...            ...  \n",
       "4995           D468N            0.2016  likely_benign  \n",
       "4996           D468H            0.4723      ambiguous  \n",
       "4997           D468Y            0.2192  likely_benign  \n",
       "4998           D468A            0.2714  likely_benign  \n",
       "4999           D468G            0.2125  likely_benign  \n",
       "\n",
       "[5000 rows x 10 columns]"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# What does table look like\n",
    "test_chunk = pd.read_csv(\"../data/AlphaMissense_hg38.tsv\", sep = \"\\t\", header = 3, nrows = 5000)\n",
    "test_chunk"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e7ff8638-1528-4e7f-96e6-071ba940a71e",
   "metadata": {},
   "source": [
    "Will use the protein variant column and uniprotID to merge with our ADs. \n",
    "\n",
    "Plan: \n",
    "- AlphaMissense table: create a foreign_key column on the uniprot_id + \"_\" + position\n",
    "- ADs table: expand to create a foreign_key column on every possible uniprotID + \"_\" + position combo using start and end\n",
    "- Then, search file for matches on foreign_key column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "f7fbdace-de79-42e0-9d97-fc1324a4256c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>genome</th>\n",
       "      <th>uniprot_id</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>protein_variant</th>\n",
       "      <th>am_pathogenicity</th>\n",
       "      <th>am_class</th>\n",
       "      <th>foreign_key</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69094</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>V2L</td>\n",
       "      <td>0.2937</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>Q8NH21_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69094</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>V2L</td>\n",
       "      <td>0.2937</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>Q8NH21_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69094</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>V2M</td>\n",
       "      <td>0.3296</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>Q8NH21_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69095</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>V2A</td>\n",
       "      <td>0.2609</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>Q8NH21_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>69095</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q8NH21</td>\n",
       "      <td>ENST00000335137.4</td>\n",
       "      <td>V2E</td>\n",
       "      <td>0.2922</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>Q8NH21_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4995</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942896</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q96NU1</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>D468N</td>\n",
       "      <td>0.2016</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>Q96NU1_468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4996</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942896</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q96NU1</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>D468H</td>\n",
       "      <td>0.4723</td>\n",
       "      <td>ambiguous</td>\n",
       "      <td>Q96NU1_468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4997</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942896</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q96NU1</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>D468Y</td>\n",
       "      <td>0.2192</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>Q96NU1_468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4998</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942897</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q96NU1</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>D468A</td>\n",
       "      <td>0.2714</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>Q96NU1_468</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4999</th>\n",
       "      <td>chr1</td>\n",
       "      <td>942897</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>hg38</td>\n",
       "      <td>Q96NU1</td>\n",
       "      <td>ENST00000342066.8</td>\n",
       "      <td>D468G</td>\n",
       "      <td>0.2125</td>\n",
       "      <td>likely_benign</td>\n",
       "      <td>Q96NU1_468</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5000 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     #CHROM     POS REF ALT genome uniprot_id      transcript_id  \\\n",
       "0      chr1   69094   G   T   hg38     Q8NH21  ENST00000335137.4   \n",
       "1      chr1   69094   G   C   hg38     Q8NH21  ENST00000335137.4   \n",
       "2      chr1   69094   G   A   hg38     Q8NH21  ENST00000335137.4   \n",
       "3      chr1   69095   T   C   hg38     Q8NH21  ENST00000335137.4   \n",
       "4      chr1   69095   T   A   hg38     Q8NH21  ENST00000335137.4   \n",
       "...     ...     ...  ..  ..    ...        ...                ...   \n",
       "4995   chr1  942896   G   A   hg38     Q96NU1  ENST00000342066.8   \n",
       "4996   chr1  942896   G   C   hg38     Q96NU1  ENST00000342066.8   \n",
       "4997   chr1  942896   G   T   hg38     Q96NU1  ENST00000342066.8   \n",
       "4998   chr1  942897   A   C   hg38     Q96NU1  ENST00000342066.8   \n",
       "4999   chr1  942897   A   G   hg38     Q96NU1  ENST00000342066.8   \n",
       "\n",
       "     protein_variant  am_pathogenicity       am_class foreign_key  \n",
       "0                V2L            0.2937  likely_benign    Q8NH21_2  \n",
       "1                V2L            0.2937  likely_benign    Q8NH21_2  \n",
       "2                V2M            0.3296  likely_benign    Q8NH21_2  \n",
       "3                V2A            0.2609  likely_benign    Q8NH21_2  \n",
       "4                V2E            0.2922  likely_benign    Q8NH21_2  \n",
       "...              ...               ...            ...         ...  \n",
       "4995           D468N            0.2016  likely_benign  Q96NU1_468  \n",
       "4996           D468H            0.4723      ambiguous  Q96NU1_468  \n",
       "4997           D468Y            0.2192  likely_benign  Q96NU1_468  \n",
       "4998           D468A            0.2714  likely_benign  Q96NU1_468  \n",
       "4999           D468G            0.2125  likely_benign  Q96NU1_468  \n",
       "\n",
       "[5000 rows x 11 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# AlphaMissense table: create an index on the uniprot_id + \"_\" + position\n",
    "\n",
    "test_chunk[\"prot_pos\"] = test_chunk[\"protein_variant\"].str[1:-1]\n",
    "test_chunk[\"foreign_key\"] = test_chunk[\"uniprot_id\"] + \"_\" + test_chunk[\"prot_pos\"]\n",
    "test_chunk = test_chunk.drop(columns = \"prot_pos\")\n",
    "test_chunk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "31d2ea8b-79d1-4b36-a390-786a12c37c40",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ADs table: expand to create a foreign_key column on every possible uniprotID + \"_\" + position combo using start and end\n",
    "# ADs = ADs[[\"gene\", \"uniprotID\", \"start\", \"end\"]]\n",
    "\n",
    "# Adding one row per position\n",
    "ADs[\"range\"] = ADs.apply(lambda row: np.arange(row[\"start\"], row[\"end\"] + 1), axis=1)\n",
    "ADs = ADs.explode('range')\n",
    "\n",
    "# Creating and saving foreign key\n",
    "ADs[\"foreign_key\"] = ADs[\"uniprotID\"] + \"_\" + ADs[\"range\"].astype(str)\n",
    "AD_foreign_key_set = set(ADs[\"foreign_key\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "61263dca-9278-4636-b3dd-cad38b5bf9ce",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 993 variants\n",
      "Found 1920 variants\n",
      "Found 3781 variants\n",
      "Found 4892 variants\n",
      "Found 5869 variants\n",
      "Found 7072 variants\n",
      "Found 7770 variants\n",
      "Found 8123 variants\n",
      "Found 9373 variants\n",
      "Found 9694 variants\n",
      "Found 10240 variants\n",
      "Found 10594 variants\n"
     ]
    }
   ],
   "source": [
    "# Now, search for matches using foreign key column\n",
    "# Code adapted from http://localhost:8888/notebooks/Desktop/Staller_Lab/SFARI/notebooks/AlphaMissense%20Pathogenicity%20Preds%20-%20V2.ipynb\n",
    "\n",
    "chunksize = 100000 \n",
    "am_output = pd.DataFrame()\n",
    "num_vars = 0\n",
    "alpha_m = \"../data/AlphaMissense_hg38.tsv\"\n",
    "\n",
    "\n",
    "# Reading in chunks, keeping rows overlapping a variant\n",
    "for chunk in pd.read_csv(alpha_m, sep='\\t', chunksize=chunksize, header = 3):\n",
    "\n",
    "    \n",
    "    chunk[\"prot_pos\"] = chunk[\"protein_variant\"].str[1:-1]\n",
    "    chunk[\"foreign_key\"] = chunk[\"uniprot_id\"] + \"_\" + chunk[\"prot_pos\"]\n",
    "    chunk = chunk.drop(columns = \"prot_pos\")\n",
    "\n",
    "    # display(chunk)\n",
    "\n",
    "    keep = chunk[chunk[\"foreign_key\"].isin(AD_foreign_key_set)]\n",
    "    am_output = pd.concat([am_output, keep])\n",
    "    if len(keep):\n",
    "        print(\"Found \" + str(len(set(am_output.index))) + \" variants\")\n",
    "\n",
    "        # display(joined)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "8af1a87a-e963-4149-82e3-6de810668bb1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "32440"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Does this match how many variants I expected?\n",
    "# Expect number of positions * 20 residues\n",
    "\n",
    "len(AD_foreign_key_set) * 20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "b789abdf-426d-4ac6-aeda-d94dd5a22db3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11\n",
      "11\n"
     ]
    }
   ],
   "source": [
    "# Represents all uniprotIDs ✅\n",
    "print(len(set(am_output[\"uniprot_id\"])))\n",
    "print(len(set(ADs[\"uniprotID\"])))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fddd2d8a-36fb-46af-be6e-95282033f4db",
   "metadata": {},
   "source": [
    "### Now, for each AD- save the top 10 and bottom 10 most pathogenic variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "56f4d143-76c3-4fa6-86f1-a238994ffa9e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>filepath</th>\n",
       "      <th>gene</th>\n",
       "      <th>uniprotID</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>range</th>\n",
       "      <th>foreign_key</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/ER...</td>\n",
       "      <td>ERG</td>\n",
       "      <td>P11308</td>\n",
       "      <td>118</td>\n",
       "      <td>261</td>\n",
       "      <td>118</td>\n",
       "      <td>P11308_118</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/ER...</td>\n",
       "      <td>ERG</td>\n",
       "      <td>P11308</td>\n",
       "      <td>118</td>\n",
       "      <td>261</td>\n",
       "      <td>119</td>\n",
       "      <td>P11308_119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/ER...</td>\n",
       "      <td>ERG</td>\n",
       "      <td>P11308</td>\n",
       "      <td>118</td>\n",
       "      <td>261</td>\n",
       "      <td>120</td>\n",
       "      <td>P11308_120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/ER...</td>\n",
       "      <td>ERG</td>\n",
       "      <td>P11308</td>\n",
       "      <td>118</td>\n",
       "      <td>261</td>\n",
       "      <td>121</td>\n",
       "      <td>P11308_121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/ER...</td>\n",
       "      <td>ERG</td>\n",
       "      <td>P11308</td>\n",
       "      <td>118</td>\n",
       "      <td>261</td>\n",
       "      <td>122</td>\n",
       "      <td>P11308_122</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/PA...</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>Q02548</td>\n",
       "      <td>304</td>\n",
       "      <td>358</td>\n",
       "      <td>354</td>\n",
       "      <td>Q02548_354</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/PA...</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>Q02548</td>\n",
       "      <td>304</td>\n",
       "      <td>358</td>\n",
       "      <td>355</td>\n",
       "      <td>Q02548_355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/PA...</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>Q02548</td>\n",
       "      <td>304</td>\n",
       "      <td>358</td>\n",
       "      <td>356</td>\n",
       "      <td>Q02548_356</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/PA...</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>Q02548</td>\n",
       "      <td>304</td>\n",
       "      <td>358</td>\n",
       "      <td>357</td>\n",
       "      <td>Q02548_357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>../output/caitlin_experiment/variant fastas/PA...</td>\n",
       "      <td>PAX5</td>\n",
       "      <td>Q02548</td>\n",
       "      <td>304</td>\n",
       "      <td>358</td>\n",
       "      <td>358</td>\n",
       "      <td>Q02548_358</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1622 rows × 7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                             filepath  gene uniprotID  start  \\\n",
       "0   ../output/caitlin_experiment/variant fastas/ER...   ERG    P11308    118   \n",
       "0   ../output/caitlin_experiment/variant fastas/ER...   ERG    P11308    118   \n",
       "0   ../output/caitlin_experiment/variant fastas/ER...   ERG    P11308    118   \n",
       "0   ../output/caitlin_experiment/variant fastas/ER...   ERG    P11308    118   \n",
       "0   ../output/caitlin_experiment/variant fastas/ER...   ERG    P11308    118   \n",
       "..                                                ...   ...       ...    ...   \n",
       "14  ../output/caitlin_experiment/variant fastas/PA...  PAX5    Q02548    304   \n",
       "14  ../output/caitlin_experiment/variant fastas/PA...  PAX5    Q02548    304   \n",
       "14  ../output/caitlin_experiment/variant fastas/PA...  PAX5    Q02548    304   \n",
       "14  ../output/caitlin_experiment/variant fastas/PA...  PAX5    Q02548    304   \n",
       "14  ../output/caitlin_experiment/variant fastas/PA...  PAX5    Q02548    304   \n",
       "\n",
       "    end range foreign_key  \n",
       "0   261   118  P11308_118  \n",
       "0   261   119  P11308_119  \n",
       "0   261   120  P11308_120  \n",
       "0   261   121  P11308_121  \n",
       "0   261   122  P11308_122  \n",
       "..  ...   ...         ...  \n",
       "14  358   354  Q02548_354  \n",
       "14  358   355  Q02548_355  \n",
       "14  358   356  Q02548_356  \n",
       "14  358   357  Q02548_357  \n",
       "14  358   358  Q02548_358  \n",
       "\n",
       "[1622 rows x 7 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ADs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20972021-aed8-40bf-a2d9-5d5ebf1a927c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "id": "bfcc0154-f6aa-4fe7-bd02-d4135c37e356",
   "metadata": {},
   "outputs": [],
   "source": [
    "most_pathogenic = pd.DataFrame()\n",
    "least_pathogenic = pd.DataFrame()\n",
    "\n",
    "for filepath in set(ADs[\"filepath\"]):\n",
    "    AD_pos_rows = ADs[ADs[\"filepath\"] == filepath]\n",
    "\n",
    "    # Filter am output to that AD's variants only\n",
    "    AD_am_output = pd.merge(AD_pos_rows, am_output, on = \"foreign_key\", how = \"left\")\n",
    "    AD_am_output = AD_am_output.sort_values(by = \"am_pathogenicity\", ascending = False)\n",
    "\n",
    "    # Sort by pathogenicity to get top 10 and bottom 10 for each AD\n",
    "    most_pathogenic = pd.concat([most_pathogenic, AD_am_output.iloc[:10]])\n",
    "    least_pathogenic = pd.concat([least_pathogenic, AD_am_output.iloc[:10]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "be33e3c7-d042-49c4-a511-4702b3d1aeb4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filepath</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/NCOA1_Q15788_AD_840-1011</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/ERG_P11308_AD_118-261_seq_adj</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/IKZF1_Q13422_AD_284-365</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/ERG_P11308_AD_433-479_seq_adj</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/NR4A2_P43354_AD_1-91</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/PAX6_P26367_AD_271-422</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/CAMTA2_O94983_AD_472-581</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/PITX1_P78337_AD_234-283</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/NKX2-2_O95096_AD_220-273</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/PAX5_Q02548_AD_304-358</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/OTX1_P32242_AD_172-354</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/NCOA1_Q15788_AD_1241-1385</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/NR4A2_P43354_AD_584-598</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/CAMTA2_O94983_AD_285-468</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/MEIS2_O14770_AD_340-477</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    count\n",
       "filepath                                                 \n",
       "../output/caitlin_experiment/variant fastas/NCO...     10\n",
       "../output/caitlin_experiment/variant fastas/ERG...     10\n",
       "../output/caitlin_experiment/variant fastas/IKZ...     10\n",
       "../output/caitlin_experiment/variant fastas/ERG...     10\n",
       "../output/caitlin_experiment/variant fastas/NR4...     10\n",
       "../output/caitlin_experiment/variant fastas/PAX...     10\n",
       "../output/caitlin_experiment/variant fastas/CAM...     10\n",
       "../output/caitlin_experiment/variant fastas/PIT...     10\n",
       "../output/caitlin_experiment/variant fastas/NKX...     10\n",
       "../output/caitlin_experiment/variant fastas/PAX...     10\n",
       "../output/caitlin_experiment/variant fastas/OTX...     10\n",
       "../output/caitlin_experiment/variant fastas/NCO...     10\n",
       "../output/caitlin_experiment/variant fastas/NR4...     10\n",
       "../output/caitlin_experiment/variant fastas/CAM...     10\n",
       "../output/caitlin_experiment/variant fastas/MEI...     10"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(most_pathogenic[\"filepath\"].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "3e5c229d-419f-45f9-b212-0eb1ea551405",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>filepath</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/NCOA1_Q15788_AD_840-1011</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/ERG_P11308_AD_118-261_seq_adj</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/IKZF1_Q13422_AD_284-365</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/ERG_P11308_AD_433-479_seq_adj</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/NR4A2_P43354_AD_1-91</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/PAX6_P26367_AD_271-422</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/CAMTA2_O94983_AD_472-581</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/PITX1_P78337_AD_234-283</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/NKX2-2_O95096_AD_220-273</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/PAX5_Q02548_AD_304-358</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/OTX1_P32242_AD_172-354</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/NCOA1_Q15788_AD_1241-1385</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/NR4A2_P43354_AD_584-598</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/CAMTA2_O94983_AD_285-468</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>../output/caitlin_experiment/variant fastas/MEIS2_O14770_AD_340-477</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    count\n",
       "filepath                                                 \n",
       "../output/caitlin_experiment/variant fastas/NCO...     10\n",
       "../output/caitlin_experiment/variant fastas/ERG...     10\n",
       "../output/caitlin_experiment/variant fastas/IKZ...     10\n",
       "../output/caitlin_experiment/variant fastas/ERG...     10\n",
       "../output/caitlin_experiment/variant fastas/NR4...     10\n",
       "../output/caitlin_experiment/variant fastas/PAX...     10\n",
       "../output/caitlin_experiment/variant fastas/CAM...     10\n",
       "../output/caitlin_experiment/variant fastas/PIT...     10\n",
       "../output/caitlin_experiment/variant fastas/NKX...     10\n",
       "../output/caitlin_experiment/variant fastas/PAX...     10\n",
       "../output/caitlin_experiment/variant fastas/OTX...     10\n",
       "../output/caitlin_experiment/variant fastas/NCO...     10\n",
       "../output/caitlin_experiment/variant fastas/NR4...     10\n",
       "../output/caitlin_experiment/variant fastas/CAM...     10\n",
       "../output/caitlin_experiment/variant fastas/MEI...     10"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame(least_pathogenic[\"filepath\"].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2a7e8f06-1cb5-4e84-aa35-9f9c6906748c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
