{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "57427b84",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "a67a7866",
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Seq import Seq\n",
    "from Bio import SeqIO\n",
    "#from Bio.Alphabet import IUPAC\n",
    "import sys\n",
    "# import pyensembl\n",
    "import os\n",
    "#https://towardsdatascience.com/a-simple-guide-to-command-line-arguments-with-argparse-6824c30ab1c3\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d352d2e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# SK: Dictionary of protein names to sequences\n",
    "proteins = {}\n",
    "for record in SeqIO.parse(\"../raw_files/gencode.v36.pc_translations.fa\", \"fasta\"):\n",
    "    name = record.id.split(\"|\")[1].split(\".\")[0]\n",
    "    proteins[name] = str(record.seq)\n",
    "    \n",
    "\n",
    "# SK: Dictionary of protein names to CDS dna transcript\n",
    "\n",
    "dna_transcripts = {}\n",
    "for record in SeqIO.parse(\"../raw_files/gencode.v36.pc_transcripts.fa\", \"fasta\"):\n",
    "\tname = record.id.split(\"|\")[0].split(\".\")[0]\n",
    "\trecord_c = record.id.split(\"|\")\n",
    "\tfor i in record_c:\n",
    "\t\tif \"CDS\" in i:\n",
    "\t\t\tcoords = i.replace(\"CDS:\",\"\")\n",
    "\tstart = int(coords.split(\"-\")[0])\n",
    "\tend = int(coords.split(\"-\")[1])\n",
    "\tdna_seq = str(record.seq)[start-1:end]\n",
    "\tdna_transcripts[name] = dna_seq"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e65ea99",
   "metadata": {},
   "source": [
    "# Plan:\n",
    "1. Use cds coordinates, create df with genomic coordinates as one column and corresponding nt as second column\n",
    "2. Use position in variant file to resplace the value in the nt column\n",
    "3. Translate the nt column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "id": "681bfa4e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "      <th>22</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>37020708</td>\n",
       "      <td>37020709</td>\n",
       "      <td>AD</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000358127</td>\n",
       "      <td>9</td>\n",
       "      <td>37020708</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>9</td>\n",
       "      <td>37020708</td>\n",
       "      <td>37020709</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>Q</td>\n",
       "      <td>K</td>\n",
       "      <td>No-Syn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>37020741</td>\n",
       "      <td>37020742</td>\n",
       "      <td>AD</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000358127</td>\n",
       "      <td>9</td>\n",
       "      <td>37020741</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>9</td>\n",
       "      <td>37020741</td>\n",
       "      <td>37020742</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>V</td>\n",
       "      <td>I</td>\n",
       "      <td>No-Syn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9</td>\n",
       "      <td>37015012</td>\n",
       "      <td>37015013</td>\n",
       "      <td>AD</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000358127</td>\n",
       "      <td>9</td>\n",
       "      <td>37015012</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>9</td>\n",
       "      <td>37015012</td>\n",
       "      <td>37015013</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>V</td>\n",
       "      <td>I</td>\n",
       "      <td>No-Syn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>9</td>\n",
       "      <td>37015048</td>\n",
       "      <td>37015049</td>\n",
       "      <td>AD</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000358127</td>\n",
       "      <td>9</td>\n",
       "      <td>37015048</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>9</td>\n",
       "      <td>37015048</td>\n",
       "      <td>37015049</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>No-Syn</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    0         1         2   3                4  5  6                7   8   \\\n",
       "4    9  37020708  37020709  AD  ENSG00000196092  .  -  ENST00000358127   9   \n",
       "8    9  37020741  37020742  AD  ENSG00000196092  .  -  ENST00000358127   9   \n",
       "16   9  37015012  37015013  AD  ENSG00000196092  .  -  ENST00000358127   9   \n",
       "23   9  37015048  37015049  AD  ENSG00000196092  .  -  ENST00000358127   9   \n",
       "\n",
       "          9   ...  13 14        15        16  17  18        19 20 21      22  \n",
       "4   37020708  ...   -  9  37020708  37020709   G   T  0.000005  Q  K  No-Syn  \n",
       "8   37020741  ...   -  9  37020741  37020742   C   T  0.000009  V  I  No-Syn  \n",
       "16  37015012  ...   -  9  37015012  37015013   C   T  0.000009  V  I  No-Syn  \n",
       "23  37015048  ...   -  9  37015048  37015049   C   T  0.000009  A  T  No-Syn  \n",
       "\n",
       "[4 rows x 23 columns]"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants = pd.read_csv(\"../outputs/mutations/domains_expanded_iWES_v2_variants_snv_classified/ENST00000358127.bed\", sep = \"\\t\", header = None)\n",
    "no_syn_variants = variants[variants[22] == \"No-Syn\"]\n",
    "no_syn_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "id": "61dc0ba2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'-'"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strand = no_syn_variants[13].iloc[1]\n",
    "strand"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "36b86936",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ATGGATTTAGAGAAAAATTATCCGACTCCTCGGACCAGCAGGACAGGACATGGAGGAGTGAATCAGCTTGGGGGGGTTTTTGTGAATGGACGGCCACTCCCGGATGTAGTCCGCCAGAGGATAGTGGAACTTGCTCATCAAGGTGTCAGGCCCTGCGACATCTCCAGGCAGCTTCGGGTCAGCCATGGTTGTGTCAGCAAAATTCTTGGCAGGTATTATGAGACAGGAAGCATCAAGCCTGGGGTAATTGGAGGATCCAAACCAAAGGTCGCCACACCCAAAGTGGTGGAAAAAATCGCTGAATATAAACGCCAAAATCCCACCATGTTTGCCTGGGAGATCAGGGACCGGCTGCTGGCAGAGCGGGTGTGTGACAATGACACCGTGCCTAGCGTCAGTTCCATCAACAGGATCATCCGGACAAAAGTACAGCAGCCACCCAACCAACCAGTCCCAGCTTCCAGTCACAGCATAGTGTCCACTGGCTCCGTGACGCAGGTGTCCTCGGTGAGCACGGATTCGGCCGGCTCGTCGTACTCCATCAGCGGCATCCTGGGCATCACGTCCCCCAGCGCCGACACCAACAAGCGCAAGAGAGACGAAGGTATTCAGGAGTCTCCGGTGCCGAACGGCCACTCGCTTCCGGGCAGAGACTTCCTCCGGAAGCAGATGCGGGGAGACTTGTTCACACAGCAGCAGCTGGAGGTGCTGGACCGCGTGTTTGAGAGGCAGCACTACTCAGACATCTTCACCACCACAGAGCCCATCAAGCCCGAGCAGACCACAGAGTATTCAGCCATGGCCTCGCTGGCTGGTGGGCTGGACGACATGAAGGCCAATCTGGCCAGCCCCACCCCTGCTGACATCGGGAGCAGTGTGCCAGGCCCGCAGTCCTACCCCATTGTGACAGGCCGTGACTTGGCGAGCACGACCCTCCCCGGGTACCCTCCACACGTCCCCCCCGCTGGACAGGGCAGCTACTCAGCACCGACGCTGACAGGGATGGTGCCTGGGAGTGAGTTTTCCGGGAGTCCCTACAGCCACCCTCAGTATTCCTCGTACAACGACTCCTGGAGGTTCCCCAACCCGGGGCTGCTTGGCTCCCCCTACTATTATAGCGCTGCCGCCCGAGGAGCCGCCCCACCTGCAGCCGCCACTGCCTATGACCGTCACTGA'"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wt_nt = dna_transcripts['ENST00000358127']\n",
    "wt_nt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "7749355f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'TCAGTGACGGTCATAGGCAGTGGCGGCTGCAGGTGGGGCGGCTCCTCGGGCGGCAGCGCTATAATAGTAGGGGGAGCCAAGCAGCCCCGGGTTGGGGAACCTCCAGGAGTCGTTGTACGAGGAATACTGAGGGTGGCTGTAGGGACTCCCGGAAAACTCACTCCCAGGCACCATCCCTGTCAGCGTCGGTGCTGAGTAGCTGCCCTGTCCAGCGGGGGGGACGTGTGGAGGGTACCCGGGGAGGGTCGTGCTCGCCAAGTCACGGCCTGTCACAATGGGGTAGGACTGCGGGCCTGGCACACTGCTCCCGATGTCAGCAGGGGTGGGGCTGGCCAGATTGGCCTTCATGTCGTCCAGCCCACCAGCCAGCGAGGCCATGGCTGAATACTCTGTGGTCTGCTCGGGCTTGATGGGCTCTGTGGTGGTGAAGATGTCTGAGTAGTGCTGCCTCTCAAACACGCGGTCCAGCACCTCCAGCTGCTGCTGTGTGAACAAGTCTCCCCGCATCTGCTTCCGGAGGAAGTCTCTGCCCGGAAGCGAGTGGCCGTTCGGCACCGGAGACTCCTGAATACCTTCGTCTCTCTTGCGCTTGTTGGTGTCGGCGCTGGGGGACGTGATGCCCAGGATGCCGCTGATGGAGTACGACGAGCCGGCCGAATCCGTGCTCACCGAGGACACCTGCGTCACGGAGCCAGTGGACACTATGCTGTGACTGGAAGCTGGGACTGGTTGGTTGGGTGGCTGCTGTACTTTTGTCCGGATGATCCTGTTGATGGAACTGACGCTAGGCACGGTGTCATTGTCACACACCCGCTCTGCCAGCAGCCGGTCCCTGATCTCCCAGGCAAACATGGTGGGATTTTGGCGTTTATATTCAGCGATTTTTTCCACCACTTTGGGTGTGGCGACCTTTGGTTTGGATCCTCCAATTACCCCAGGCTTGATGCTTCCTGTCTCATAATACCTGCCAAGAATTTTGCTGACACAACCATGGCTGACCCGAAGCTGCCTGGAGATGTCGCAGGGCCTGACACCTTGATGAGCAAGTTCCACTATCCTCTGGCGGACTACATCCGGGAGTGGCCGTCCATTCACAAAAACCCCCCCAAGCTGATTCACTCCTCCATGTCCTGTCCTGCTGGTCCGAGGAGTCGGATAATTTTTCTCTAAATCCAT'"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "if strand == \"-\":\n",
    "    wt_nt = str(Seq(wt_nt).reverse_complement())\n",
    "wt_nt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "ae250574",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1171</th>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1172</th>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1173</th>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1174</th>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1175</th>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1176 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     nt\n",
       "0     T\n",
       "1     C\n",
       "2     A\n",
       "3     G\n",
       "4     T\n",
       "...  ..\n",
       "1171  T\n",
       "1172  C\n",
       "1173  C\n",
       "1174  A\n",
       "1175  T\n",
       "\n",
       "[1176 rows x 1 columns]"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nt_df = pd.DataFrame({\"nt\" : [*wt_nt]})\n",
    "nt_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "id": "20403765",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>37033986</td>\n",
       "      <td>37034031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37020636</td>\n",
       "      <td>37020801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>37014997</td>\n",
       "      <td>37015194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>37006473</td>\n",
       "      <td>37006537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>37002648</td>\n",
       "      <td>37002776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>36966549</td>\n",
       "      <td>36966724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>36923355</td>\n",
       "      <td>36923484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>36882004</td>\n",
       "      <td>36882105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>36846843</td>\n",
       "      <td>36846929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>36840560</td>\n",
       "      <td>36840636</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          1         2\n",
       "0  37033986  37034031\n",
       "1  37020636  37020801\n",
       "2  37014997  37015194\n",
       "3  37006473  37006537\n",
       "4  37002648  37002776\n",
       "5  36966549  36966724\n",
       "6  36923355  36923484\n",
       "7  36882004  36882105\n",
       "8  36846843  36846929\n",
       "9  36840560  36840636"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cds_bed = pd.read_csv(\"../outputs/mutations/cds_bed_format/ENST00000358127\", sep = \"\\t\", header = None)\n",
    "cds_bed = cds_bed[[1, 2]]\n",
    "cds_bed[1] += 1\n",
    "cds_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "7fccf151",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>37033986</td>\n",
       "      <td>37034031</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>37020636</td>\n",
       "      <td>37020801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>37014997</td>\n",
       "      <td>37015194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>37006473</td>\n",
       "      <td>37006537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>37002648</td>\n",
       "      <td>37002776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>36966549</td>\n",
       "      <td>36966724</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>36923355</td>\n",
       "      <td>36923484</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>36882004</td>\n",
       "      <td>36882105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>36846843</td>\n",
       "      <td>36846929</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>36840560</td>\n",
       "      <td>36840636</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          1         2\n",
       "0  37033986  37034031\n",
       "1  37020636  37020801\n",
       "2  37014997  37015194\n",
       "3  37006473  37006537\n",
       "4  37002648  37002776\n",
       "5  36966549  36966724\n",
       "6  36923355  36923484\n",
       "7  36882004  36882105\n",
       "8  36846843  36846929\n",
       "9  36840560  36840636"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cds_bed = cds_bed.sort_values(by = 1, ascending = False)\n",
    "cds_bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "id": "28a9c99b",
   "metadata": {},
   "outputs": [],
   "source": [
    "range_col = []\n",
    "for start, end in zip(cds_bed[1], cds_bed[2]):\n",
    "    if strand == \"-\":\n",
    "        range_col += list(range(end, start - 1, -1))\n",
    "    else:\n",
    "        range_col += list(range(start, end + 1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "6591f59c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1176"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(range_col)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "id": "12917a57",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nt</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gen_pos</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>37034031</th>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37034030</th>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37034029</th>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37034028</th>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37034027</th>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36840564</th>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36840563</th>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36840562</th>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36840561</th>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36840560</th>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1176 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         nt\n",
       "gen_pos    \n",
       "37034031  T\n",
       "37034030  C\n",
       "37034029  A\n",
       "37034028  G\n",
       "37034027  T\n",
       "...      ..\n",
       "36840564  T\n",
       "36840563  C\n",
       "36840562  C\n",
       "36840561  A\n",
       "36840560  T\n",
       "\n",
       "[1176 rows x 1 columns]"
      ]
     },
     "execution_count": 136,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nt_df[\"gen_pos\"] = range_col\n",
    "nt_df = nt_df.set_index(\"gen_pos\")\n",
    "nt_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "id": "a0b1f306",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "nt    G\n",
       "Name: 37020709, dtype: object"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nt_df.loc[37020709]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "id": "017d3a41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "nt    G\n",
       "Name: 37020742, dtype: object"
      ]
     },
     "execution_count": 144,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nt_df.loc[37020742]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "id": "ccca6134",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "nt    G\n",
       "Name: 37015013, dtype: object"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nt_df.loc[37015013]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "id": "ed546af0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "nt    C\n",
       "Name: 37015049, dtype: object"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nt_df.loc[37015049]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "68a99abe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "      <th>22</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>37020708</td>\n",
       "      <td>37020709</td>\n",
       "      <td>AD</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000358127</td>\n",
       "      <td>9</td>\n",
       "      <td>37020708</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>9</td>\n",
       "      <td>37020708</td>\n",
       "      <td>37020709</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>Q</td>\n",
       "      <td>K</td>\n",
       "      <td>No-Syn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>37020741</td>\n",
       "      <td>37020742</td>\n",
       "      <td>AD</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000358127</td>\n",
       "      <td>9</td>\n",
       "      <td>37020741</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>9</td>\n",
       "      <td>37020741</td>\n",
       "      <td>37020742</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>V</td>\n",
       "      <td>I</td>\n",
       "      <td>No-Syn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>9</td>\n",
       "      <td>37015012</td>\n",
       "      <td>37015013</td>\n",
       "      <td>AD</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000358127</td>\n",
       "      <td>9</td>\n",
       "      <td>37015012</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>9</td>\n",
       "      <td>37015012</td>\n",
       "      <td>37015013</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>V</td>\n",
       "      <td>I</td>\n",
       "      <td>No-Syn</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>9</td>\n",
       "      <td>37015048</td>\n",
       "      <td>37015049</td>\n",
       "      <td>AD</td>\n",
       "      <td>ENSG00000196092</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000358127</td>\n",
       "      <td>9</td>\n",
       "      <td>37015048</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>9</td>\n",
       "      <td>37015048</td>\n",
       "      <td>37015049</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>No-Syn</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows × 23 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    0         1         2   3                4  5  6                7   8   \\\n",
       "4    9  37020708  37020709  AD  ENSG00000196092  .  -  ENST00000358127   9   \n",
       "8    9  37020741  37020742  AD  ENSG00000196092  .  -  ENST00000358127   9   \n",
       "16   9  37015012  37015013  AD  ENSG00000196092  .  -  ENST00000358127   9   \n",
       "23   9  37015048  37015049  AD  ENSG00000196092  .  -  ENST00000358127   9   \n",
       "\n",
       "          9   ...  13 14        15        16  17  18        19 20 21      22  \n",
       "4   37020708  ...   -  9  37020708  37020709   G   T  0.000005  Q  K  No-Syn  \n",
       "8   37020741  ...   -  9  37020741  37020742   C   T  0.000009  V  I  No-Syn  \n",
       "16  37015012  ...   -  9  37015012  37015013   C   T  0.000009  V  I  No-Syn  \n",
       "23  37015048  ...   -  9  37015048  37015049   C   T  0.000009  A  T  No-Syn  \n",
       "\n",
       "[4 rows x 23 columns]"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "no_syn_variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23c9caff",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1436df64",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa1a5209",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b471dc21",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c5db880e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f6b3f42",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "42ba9dda",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "35e758c9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0d864628",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "70809044",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "88c1f487",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>uniprotID</th>\n",
       "      <th>Matching Isoforms</th>\n",
       "      <th>Canonical Transcript ID</th>\n",
       "      <th>orig_uniprotID</th>\n",
       "      <th>Reference</th>\n",
       "      <th>TileType</th>\n",
       "      <th>Sequence</th>\n",
       "      <th>matching_isoforms</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>PAX5</td>\n",
       "      <td>304</td>\n",
       "      <td>358</td>\n",
       "      <td>Q02548</td>\n",
       "      <td>[['Q02548'], ['Q02548']]</td>\n",
       "      <td>nan / nan</td>\n",
       "      <td>Q02548 / Q02548</td>\n",
       "      <td>PMID: 8617244, Soto / Choi 2000 list, GSL</td>\n",
       "      <td>TF</td>\n",
       "      <td>GRDLASTTLPGYPPHVPPAGQGSYSAPTLTGMVPGSEFSGSPYSHP...</td>\n",
       "      <td>['Q02548']</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Gene  Start  End uniprotID         Matching Isoforms  \\\n",
       "473  PAX5    304  358    Q02548  [['Q02548'], ['Q02548']]   \n",
       "\n",
       "    Canonical Transcript ID   orig_uniprotID  \\\n",
       "473               nan / nan  Q02548 / Q02548   \n",
       "\n",
       "                                     Reference TileType  \\\n",
       "473  PMID: 8617244, Soto / Choi 2000 list, GSL       TF   \n",
       "\n",
       "                                              Sequence matching_isoforms  \n",
       "473  GRDLASTTLPGYPPHVPPAGQGSYSAPTLTGMVPGSEFSGSPYSHP...        ['Q02548']  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "SFARI_TF_known_ADs[SFARI_TF_known_ADs[\"uniprotID\"] == \"Q02548\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "id": "969d1a74",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Seq('GRDLASTTLPGYPPHVPPAGQGSYSAPTLTGMVPGSEFSGSPYSHPQYSSYNDSW')"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Seq(wt_nt).translate()[303:358]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "41bef2aa",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4fb43be1",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5b9260b0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "26a685f9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d16bfcc7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1920633f",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a7f16415",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3649a940",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "52928a85",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['9', '37020691', '37020801', 'AD', 'ENSG00000196092', '.', '-', 'ENST00000358127']\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "invalid literal for int() with base 10: 'ENSG00000196092'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-19-0a17a1d168f3>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     25\u001b[0m         \u001b[0;31m# SK: Not sure if this should be zero or one indexed, assuming one\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     26\u001b[0m         \u001b[0;31m# SK 10/25: running into an indexing issue.. not sure why\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 27\u001b[0;31m         \u001b[0mmut_pos\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     28\u001b[0m         \u001b[0menst\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     29\u001b[0m         \u001b[0mname\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: invalid literal for int() with base 10: 'ENSG00000196092'"
     ]
    }
   ],
   "source": [
    "reverse = {\"C\":\"G\", \"A\":\"T\", \"T\":\"A\", \"G\":\"C\"}\n",
    "\n",
    "with open(\"../outputs/mutations/domains_expanded_iWES_v2_variants_snv_classified//Q02548\") as f1:\n",
    "        \n",
    "    print_count = 0 # SK\n",
    "\n",
    "    # SK: Reading in the variants from the bed file\n",
    "    for line in f1:\n",
    "        # SK: Read in strand and mutated nucleotide\n",
    "        line = line.rstrip().split(\"\\t\")\n",
    "        print(line)\n",
    "        strand = line[6]\n",
    "\n",
    "        # mut_nt = line[-4]\n",
    "        mut_nt = line[-2]\n",
    "\n",
    "        # SK: Check that SNP\n",
    "        if len(mut_nt) != 1:\n",
    "            continue\n",
    "        if len(line[-3]) != 1:\n",
    "            continue\n",
    "        if mut_nt == \".\":\n",
    "            continue\n",
    "\n",
    "        # SK: Not sure if this should be zero or one indexed, assuming one\n",
    "        # SK 10/25: running into an indexing issue.. not sure why \n",
    "        mut_pos = int(line[-4])\n",
    "        enst = line[4]\n",
    "        name = line[4]\n",
    "\n",
    "        # SK: Reading in the sorted cds genomic coordinates\n",
    "        list_coords = []\n",
    "        with open(\"../outputs/mutations/cds_bed_format/sorted/\" + name + \".bed\") as f2:\n",
    "\n",
    "            # SK: Iterate through exons in bed file, append tuples of coords to list_coords\n",
    "            for linex in f2:\n",
    "                linex = linex.rstrip().split(\"\\t\")\n",
    "                start, end = int(linex[1]), int(linex[2])\n",
    "                list_coords.append((start+1, end)) # SK: Bed file -> One-indexed coordinates now\n",
    "\n",
    "        # SK: Reverse list_coords if on negative strand and reverse translate mutant nucleotide\n",
    "        if strand == \"-\":\n",
    "            list_coords.reverse()\n",
    "            mut_nt = reverse[mut_nt]\n",
    "\n",
    "        # SK: Get DNA transcript and protein sequence of protein\n",
    "        nt_seq, translation = dna_transcripts[enst], proteins[enst]\n",
    "\n",
    "        # Create a list of coords\n",
    "        total_coords = []\n",
    "        # SK: Append all individual coordinates in order to total_coords\n",
    "        for start, end in list_coords:\n",
    "            if strand == \"-\":\n",
    "                total_coords += list(range(end, start - 1, -1))\n",
    "\n",
    "            else:\n",
    "                total_coords += list(range(start, end + 1))\n",
    "\n",
    "        # SK: dictionary of ith total coordinate to the ith nt_seq\n",
    "        pos_nt = {} # 1500: A-T-C-G\n",
    "        for i in range(len(total_coords)):\n",
    "            if i >= len(nt_seq):\n",
    "                print(i)\n",
    "                print(nt_seq)\n",
    "                print(total_coords[i])\n",
    "                print(list_coords)\n",
    "            pos_nt[total_coords[i]] = nt_seq[i]\n",
    "\n",
    "        # SK: Breaking total nt coordinates into total codon coordinates\n",
    "        start = 0\n",
    "        aa_codon = {} # 300 : [500, 520, 521]\n",
    "        for i in range(int(len(total_coords)/3)):\n",
    "            aa_codon[i+1] = total_coords[start:start+3]\n",
    "            start += 3\n",
    "\n",
    "        # SK: \n",
    "        codon = []\n",
    "        for k, v in aa_codon.items():\n",
    "            if mut_pos in v:\n",
    "                aa_pos = k\n",
    "                codon = v\n",
    "                break\n",
    "\n",
    "        # WT codon\n",
    "        wt_dict = {} # pos1:nt1, pos2:nt2, pos3:nt3\n",
    "        for pos in codon:\n",
    "            wt_dict[pos] = pos_nt[pos]\n",
    "\n",
    "        # Mutated codon\n",
    "        mt_dict = dict(wt_dict)\n",
    "        mt_dict[mut_pos] = mut_nt # pos1:nt1, pos2:nt2, pos3:nt3\n",
    "\n",
    "        #print(wt_dict, mt_dict)\n",
    "\n",
    "        # To String\n",
    "        wt_str = \"\"\n",
    "        for nt in wt_dict.values():\n",
    "            wt_str += nt\n",
    "        mt_str = \"\"\n",
    "        for nt in mt_dict.values():\n",
    "            mt_str += nt\n",
    "\n",
    "\n",
    "        wt_aa, mt_aa = Seq(wt_str).translate(), Seq(mt_str).translate()\n",
    "        print(wt_str)\n",
    "        print(mt_str)\n",
    "        print(wt_aa)\n",
    "        print(mt_aa)\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e8c963b9",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "233f4e0c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "adf56e0c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dfae0916",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "204acab0",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34778bdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "count = 1\n",
    "for file in files:\n",
    "    print(count, file)\n",
    "    count += 1\n",
    "    with open(directory +  \"/\" + file) as f1, open(o_directory + \"/\" + file, \"w\") as o1:\n",
    "        \n",
    "        print_count = 0 # SK\n",
    "        \n",
    "        # SK: Reading in the variants from the bed file\n",
    "        for line in f1:\n",
    "            # SK: Read in strand and mutated nucleotide\n",
    "            line = line.rstrip().split(\"\\t\")\n",
    "            strand = line[5]\n",
    "                \n",
    "            # mut_nt = line[-4]\n",
    "            mut_nt = line[-2]\n",
    "            \n",
    "            # SK: Check that SNP\n",
    "            if len(mut_nt) != 1:\n",
    "                continue\n",
    "            if len(line[-3]) != 1:\n",
    "                continue\n",
    "            if mut_nt == \".\":\n",
    "                continue\n",
    "\n",
    "            # SK: Not sure if this should be zero or one indexed, assuming one\n",
    "            # SK 10/25: running into an indexing issue.. not sure why \n",
    "            mut_pos = int(line[-4])\n",
    "            enst = line[4]\n",
    "            name = line[4]\n",
    "            \n",
    "            # SK: Reading in the sorted cds genomic coordinates\n",
    "            list_coords = []\n",
    "            with open(\"../outputs/mutations/cds_bed_format/sorted/\" + name + \".bed\") as f2:\n",
    "                \n",
    "                # SK: Iterate through exons in bed file, append tuples of coords to list_coords\n",
    "                for linex in f2:\n",
    "                    linex = linex.rstrip().split(\"\\t\")\n",
    "                    start, end = int(linex[1]), int(linex[2])\n",
    "                    list_coords.append((start+1, end)) # SK: Bed file -> One-indexed coordinates now\n",
    "                    \n",
    "            # SK: Reverse list_coords if on negative strand and reverse translate mutant nucleotide\n",
    "            if strand == \"-\":\n",
    "                list_coords.reverse()\n",
    "                mut_nt = reverse[mut_nt]\n",
    "            \n",
    "            # SK: Get DNA transcript and protein sequence of protein\n",
    "            nt_seq, translation = dna_transcripts[enst], proteins[enst]\n",
    "\n",
    "            # Create a list of coords\n",
    "            total_coords = []\n",
    "            # SK: Append all individual coordinates in order to total_coords\n",
    "            for start, end in list_coords:\n",
    "                if strand == \"-\":\n",
    "                    total_coords += list(range(end, start - 1, -1))\n",
    "         \n",
    "                else:\n",
    "                    total_coords += list(range(start, end + 1))\n",
    "\n",
    "            # SK: dictionary of ith total coordinate to the ith nt_seq\n",
    "            pos_nt = {} # 1500: A-T-C-G\n",
    "            for i in range(len(total_coords)):\n",
    "                if i >= len(nt_seq):\n",
    "                    print(i)\n",
    "                    print(nt_seq)\n",
    "                    print(total_coords[i])\n",
    "                    print(list_coords)\n",
    "                pos_nt[total_coords[i]] = nt_seq[i]\n",
    "\n",
    "            # SK: Breaking total nt coordinates into total codon coordinates\n",
    "            start = 0\n",
    "            aa_codon = {} # 300 : [500, 520, 521]\n",
    "            for i in range(int(len(total_coords)/3)):\n",
    "                aa_codon[i+1] = total_coords[start:start+3]\n",
    "                start += 3\n",
    "\n",
    "            # SK: \n",
    "            codon = []\n",
    "            for k, v in aa_codon.items():\n",
    "                if mut_pos in v:\n",
    "                    aa_pos = k\n",
    "                    codon = v\n",
    "                    break\n",
    "                    \n",
    "            # WT codon\n",
    "            wt_dict = {} # pos1:nt1, pos2:nt2, pos3:nt3\n",
    "            for pos in codon:\n",
    "                wt_dict[pos] = pos_nt[pos]\n",
    "\n",
    "            # Mutated codon\n",
    "            mt_dict = dict(wt_dict)\n",
    "            mt_dict[mut_pos] = mut_nt # pos1:nt1, pos2:nt2, pos3:nt3\n",
    "\n",
    "            #print(wt_dict, mt_dict)\n",
    "\n",
    "            # To String\n",
    "            wt_str = \"\"\n",
    "            for nt in wt_dict.values():\n",
    "                wt_str += nt\n",
    "            mt_str = \"\"\n",
    "            for nt in mt_dict.values():\n",
    "                mt_str += nt\n",
    "\n",
    "\n",
    "            wt_aa, mt_aa = Seq(wt_str).translate(), Seq(mt_str).translate()\n",
    "            line.append(str(wt_aa))\n",
    "            line.append(str(mt_aa))\n",
    "            if wt_aa == mt_aa:\n",
    "                line.append(\"Syn\")\n",
    "            elif wt_aa != mt_aa and mt_aa != \"*\":\n",
    "                line.append(\"No-Syn\")\n",
    "            elif wt_aa != mt_aa and mt_aa == \"*\":\n",
    "                line.append(\"NoSense\")\n",
    "            o1.write(\"\\t\".join(line) + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "b62cbf18",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'SFARI_TFs' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-149-a9097f546d03>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mSFARI_TFs\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'SFARI_TFs' is not defined"
     ]
    }
   ],
   "source": [
    "SFARI_TFs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be142f7b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
